---
title: "Results Aging"
output:
  html_document:
    toc: true
    toc_depth: 3
    toc_float: 
      collapsed: false
    includes:
      in_header: ../include/in_header.html
      before_body: ../include/before_body.html
      after_body: ../include/after_body.html
runtime: shiny_prerendered
---
<!-- # (c)  Juozas Gordevičius -->

```{r, include=FALSE}
require(broom)
require(knitr)
require(RobustRankAggreg)
require(cowplot)
source("../code/common.R")
source("../code/pformat.R")
source("../code/genomicEnrichment.R")
annotations <- cache(foo=loadHumanAnnotations,
                     fname = "annotations.RDS")
pathways <- readRDS("../Discover_Pathways/humanPathways.RDS")


wmean <- function(x, w) {
  i <- is.na(x) | is.na(w)
  x <- x[!i]; w <- w[!i]
  weighted.mean(x, w, na.rm=TRUE)
}

fdr <- function(x) { p.adjust(x, "fdr")}

```

# Appendix, padlocks

```{r}
apax <- fread("../Appendix_AgeAcceleration_Padlock/www/m6_Appendix_AgeAcceleration_Padlock.csv")
apx <- fread("../Appendix_PDvsControls_Padlock/www/m6_Appendix_PDvsControls_Padlock.csv")
apad <- readRDS("../Appendix_AgeAcceleration_Padlock/www/m6_methylation.RDS")
```

```{r}
url <- "../Appendix_AgeAcceleration_Padlock/www/m6_Appendix_AgeAcceleration_Padlock.csv"
stopifnot(file.exists(url))
```

- **[Additional file 11](`r url`) DNA methylation changes in the aging appendix of healthy individuals and PD patients.**


```{r, include=TRUE, results='asis'}
n <- nrow(apax)
glue("- Number of loci {n}")
t <- apad$key[, table(Diagnosis)]
glue("- Number of PD cases: {t[2]} and controls: {t[1]}")
```

Aging among controls and dominant direction of methylation change

```{r, include=TRUE, results='asis'}
n <- apax[, sum(p.adjust(P.DiagnosisControl.Age, "fdr") < 0.05, na.rm=TRUE)]
glue("- Number of significant loci {n}")
n <- apax[p.adjust(P.DiagnosisControl.Age, "fdr") < 0.05, length(unique(Gene))]
glue("- Number of genes affected {n}")
r <- apad$key[Diagnosis == "Control", range(Age) %>% paste(collapse=":")]
glue("- Age range of healthy controls {r}")
```

```{r, include=TRUE}
t <-
	apax[, list(
						Significant = ifelse(p.adjust(P.DiagnosisControl.Age, "fdr") < 0.05, "Significant", "NS"), 
					 	Direction = ifelse(C.DiagnosisControl.Age < 0, "Hypo-M", "Hyper-M")
					 )] %>%
	.[, table(Significant, Direction)] %>%
	.[c("NS", "Significant"), c("Hypo-M", "Hyper-M")]

# Show the table
knitr::kable(t) %>%
kable_styling(bootstrap_options = c("hover"),
							full_width = FALSE)

# Show fisher test result
t %>% fisher.test %>% 
tidy() %>%
kable() %>%
kable_styling(bootstrap_options = c("hover"))
```

Aging among PD samples and dominant direction of methylation change

```{r, include=TRUE, results='asis'}
n <- apax[, sum(p.adjust(P.DiagnosisPD.LBD.Age, "fdr") < 0.05, na.rm=TRUE)]
glue("- Number of significant loci {n}")
n <- apax[p.adjust(P.DiagnosisPD.LBD.Age, "fdr") < 0.05, length(unique(Gene))]
glue("- Number of genes affected {n}")
```

```{r, include=TRUE}
t <-
  apax[, list(
            Significant = ifelse(p.adjust(P.DiagnosisPD.LBD.Age, "fdr") < 0.05, "Significant", "NS"), 
            Direction = ifelse(C.DiagnosisPD.LBD.Age < 0, "Hypo-M", "Hyper-M")
           )] %>%
  .[, table(Significant, Direction)] %>%
  .[c("NS", "Significant"), c("Hypo-M", "Hyper-M")]

# Show the table
knitr::kable(t) %>%
kable_styling(bootstrap_options = c("hover"),
              full_width = FALSE)

# Show fisher test result
t %>% fisher.test %>% 
tidy() %>%
kable() %>%
kable_styling(bootstrap_options = c("hover"))
```
 
## Overlap of affected cytosines

Overlap of cytosines affected by PD or aging in control and PD samples

```{r, include=TRUE}
dt <-
  merge(
    apx[, list(ID, PDAffected = ifelse(adj.P.Val < 0.05, "Significant", "NS"))],
    apax[,
      list(
        ID,
        ControlAging = ifelse(p.adjust(P.DiagnosisControl.Age, "fdr") < 0.05, "Significant", "NS"),
        PDAging      = ifelse(p.adjust(P.DiagnosisPD.LBD.Age, "fdr") < 0.05, "Significant", "NS")
        )],
    by = "ID")
foo <- function(dt, A, B, title) {
  dt[, table(get(A), get(B))] %>%
  fisher.test %>%
  broom::tidy() %>% 
  mutate(Test = title) %>%
  select(Test, estimate, p.value, conf.low, conf.high)
}

list(
  foo(dt, "PDAffected", "ControlAging", title = "PD affected vs aging in controls"),
  foo(dt, "PDAffected", "PDAging", title = "PD affected vs aging in PDs"),
  foo(dt, "ControlAging", "PDAging", title = "Aging in controls vs aging in PDs")
  ) %>%
purrr::map(as.data.table) %>%
rbindlist %>%
kable() %>%
kable_styling(bootstrap_options = c("hover"))
```


## Overlap of affected genes

Overlap of ALP genes aging in control and PD samples

```{r, include=TRUE}
A <- apax %>%
  .[, list(ID, Chr, SNP, Gene, 
    FC = C.DiagnosisControl.Age, 
    P = P.DiagnosisControl.Age, 
    Q = p.adjust(P.DiagnosisControl.Age, "fdr"))
  ] %>%
  .[, list(HealthyAging = wmean(FC, -log(P)), IsHealthyAging = any(Q < 0.05, na.rm=TRUE)), Gene]

B <- apax %>%
  .[, list(ID, Chr, SNP, Gene, 
    FC = C.DiagnosisPD.LBD.Age, 
    P = P.DiagnosisPD.LBD.Age, 
    Q = p.adjust(P.DiagnosisPD.LBD.Age, "fdr"))
  ] %>%
  .[, list(PDAging = wmean(FC, -log(P)), IsPDAging = any(Q < 0.05, na.rm=TRUE)), Gene]

C  <- apx %>%
  .[, list(PD = wmean(logFC, -log(P.Value)), IsPD = any(adj.P.Val < 0.05, na.rm=TRUE)), Gene]
dt <- merge(merge(A, B, by = "Gene"), C, by = "Gene")

list(
  foo(dt, "IsPD", "IsHealthyAging", title = "PD affected vs aging in controls"),
  foo(dt, "IsPD", "IsPDAging", title = "PD affected vs aging in PDs"),
  foo(dt, "IsHealthyAging", "IsPDAging", title = "Aging in controls vs aging in PDs")
  ) %>%
purrr::map(as.data.table) %>%
rbindlist %>%
kable() %>%
kable_styling(bootstrap_options = c("hover"))
```

## Enrichment of genomic regions

```{r}
# touch()
foo <- function(analysis, prefix, annotations, t) {
  fits <- fread(glue("../{analysis}/www/{prefix}_{analysis}.csv"))
  ccol <- glue("C.DiagnosisControl.Age")
  pcol <- glue("P.DiagnosisControl.Age")
  fits <- fits[, list(ID, Chr, SNP, Str, 
                  adj.P.Val = p.adjust(get(pcol), "fdr"),
                  logFC = get(ccol)
                  )]
  dt1 <- getAnnotatrEnrichments(fits, annotations, qThresh = t)
  dt2 <- getAppendixChipEnrichments(fits, qThresh = t)
  dt3 <- getEpiRoadmapEnrichment(fits, qThresh = t)
  dt4 <- getPsychEncodeEnrichments(fits, qThresh = t)
  return(list(dt1, dt2, dt3, dt4))
}

# touch()
analysis <- "Appendix_AgeAcceleration_Padlock"
prefix <- "m6"
t <- 0.05

dtlist <- cache(
  foo = foo, fname = glue("enrichment_{analysis}.RDS"), verbose = TRUE,
  analysis = analysis, prefix= prefix, annotations = annotations,
  t = t
  )
```


General annotation of genomic elements:

```{r, include=TRUE}
dtlist[[1]] %>%
.[Type == "All"] %>%
.[, `P < 0.05` := texPstars(`P < 0.05`)] %>%
dplyr::select(estimate=OR, 
       p.value=P, 
       conf.low=Lo, 
       conf.high=Hi, 
       category=Category, 
       p_stars = `P < 0.05`) %>%
kable() %>%
kable_styling(bootstrap_options = c("hover"))
```

Genomic elements obtained from ChIP-seq:

```{r, include=TRUE}
dtlist[[2]] %>%
.[, `P < 0.05` := texPstars(`P < 0.05`)] %>%
dplyr::select(estimate=OR, 
       p.value=P, 
       conf.low=Lo, 
       conf.high=Hi, 
       category=Category, 
       p_stars = `P < 0.05`) %>%
kable() %>%
kable_styling(bootstrap_options = c("hover"))
```

```{r}
myplots <- plotAnnotatrEnrichment(dtlist[[1]])
myplots[[1]] <- myplots[[1]] + facet_grid(Type~.)
myplots[[2]] <- myplots[[2]] + facet_grid(Type~.)

myplots[[3]] <- plotAppendixChipEnrichment(dtlist[[2]], wrap = "vertical")
myplots[[4]] <- plotEpiRoadmapEnrichment(  dtlist[[3]], wrap = "vertical")
```

```{r, include=TRUE, fig.cap="**Figure S7. Genomic elements enriched with differentially methylated cytosines in the healthy aging appendix.**"}
ggarrange(myplots[[1]], myplots[[2]], myplots[[3]],
  ncol = 3, nrow = 1, labels = LETTERS[1:3],
  widths = c(4, 7, 3),
  common.legend = TRUE)
```



# Appendix, RNA-seq

```{r}
apar <- fread("../Appendix_AgeAcceleration_RNAseq/www/r1_Appendix_AgeAcceleration_RNAseq.csv")
apard <- readRDS("../Appendix_AgeAcceleration_RNAseq/www/r1_rnaseq.RDS")
```

```{r, include=TRUE, results='asis'}
n <- nrow(apar)
glue("- Number of transcripts {n}")
t <- apard$key[, table(Diagnosis)]
glue("- Number of PD cases: {t[2]} and controls: {t[1]}")
n <- apar[, sum(p.adjust(P.DiagnosisControl.Age, "fdr") < 0.05, na.rm=TRUE)]
glue("- Number of significant transcripts {n}")
```


Dominant direction of transcription change:

```{r, include=TRUE}
t <-
	apar[, list(
						Significant = ifelse(p.adjust(P.DiagnosisControl.Age, "fdr") < 0.05, "Significant", "NS"), 
					 	Direction = ifelse(C.DiagnosisControl.Age < 0, "Down", "Up")
					 )] %>%
	.[, table(Significant, Direction)] %>%
	.[c("NS", "Significant"), c("Down", "Up")]

# Show the table
knitr::kable(t) %>%
kable_styling(bootstrap_options = c("hover"),
							full_width = FALSE)

# Show fisher test result
t %>% fisher.test %>% 
broom::tidy() %>%
knitr::kable() %>%
kable_styling(bootstrap_options = c("hover"))
```

# Prefrontal cortex neurons, padlock

```{r}
pfcax <- fread("../Brain_AgeAcceleration_Padlock/www/m6_Brain_AgeAcceleration_Padlock.csv")
pfcx <- fread("../Brain_PFCRep_Padlock_withGLU/www/m6_Brain_PFCRep_Padlock_withGLU.csv")
pfcad <- readRDS("../Brain_AgeAcceleration_Padlock/www/m6_methylation.RDS")
```

```{r}
url <- "../Brain_AgeAcceleration_Padlock/www/m6_Brain_AgeAcceleration_Padlock.csv"
stopifnot(file.exists(url))
```

- **[Additional file 12](`r url`) DNA methylation changes in aging prefrontal cortex neurons of healthy individuals and PD patients.**


```{r, include=TRUE, results='asis'}
n <- nrow(pfcax[Type == "CG"])
glue("- Number of CpG loci {n}")
n <- nrow(pfcax[Type != "CG"])
glue("- Number of CpH loci {n}")
t <- pfcad$key[, table(Group)]
glue("- Number of PD cases: {t[2]} and controls: {t[1]}")
```

Aging among controls and dominant direction of methylation change

```{r, include=TRUE, results='asis'}
n <- pfcax[, sum(p.adjust(P.GroupCTRL.Age, "fdr") < 0.05, na.rm=TRUE)]
glue("- Number of significant loci {n}")
n <- pfcax[p.adjust(P.GroupCTRL.Age, "fdr") < 0.05, length(unique(Gene))]
glue("- Number of genes affected {n}")
```

```{r, include=TRUE}
t <-
	pfcax[, list(
						Significant = ifelse(p.adjust(P.GroupCTRL.Age, "fdr") < 0.05, "Significant", "NS"), 
					 	Direction = ifelse(C.GroupCTRL.Age < 0, "Hypo-M", "Hyper-M")
					 )] %>%
	.[, table(Significant, Direction)] %>%
	.[c("NS", "Significant"), c("Hypo-M", "Hyper-M")]

# Show the table
knitr::kable(t) %>%
kable_styling(bootstrap_options = c("hover"),
							full_width = FALSE)

# Show fisher test result
t %>% fisher.test %>% 
tidy() %>%
kable() %>%
kable_styling(bootstrap_options = c("hover"))
```

Aging among PD samples and dominant direction of methylation change

```{r, include=TRUE, results='asis'}
n <- pfcax[, sum(p.adjust(P.GroupPD.Age, "fdr") < 0.05, na.rm=TRUE)]
glue("- Number of significant loci {n}")
n <- pfcax[p.adjust(P.GroupPD.Age, "fdr") < 0.05, length(unique(Gene))]
glue("- Number of genes affected {n}")
```

```{r, include=TRUE}
t <-
  pfcax[, list(
            Significant = ifelse(p.adjust(P.GroupPD.Age, "fdr") < 0.05, "Significant", "NS"), 
            Direction = ifelse(C.GroupPD.Age < 0, "Hypo-M", "Hyper-M")
           )] %>%
  .[, table(Significant, Direction)] %>%
  .[c("NS", "Significant"), c("Hypo-M", "Hyper-M")]

# Show the table
knitr::kable(t) %>%
kable_styling(bootstrap_options = c("hover"),
              full_width = FALSE)

# Show fisher test result
t %>% fisher.test %>% 
tidy() %>%
kable() %>%
kable_styling(bootstrap_options = c("hover"))
```

## Overlap of affected cytosines

Overlap of cytosines affected by PD or aging in control and PD samples

```{r, include=TRUE}
dt <-
  merge(
    pfcx[, list(ID, PDAffected = ifelse(adj.P.Val < 0.05, "Significant", "NS"))],
    pfcax[,
      list(
        ID,
        ControlAging = ifelse(p.adjust(P.GroupCTRL.Age, "fdr") < 0.05, "Significant", "NS"),
        PDAging      = ifelse(p.adjust(P.GroupPD.Age, "fdr") < 0.05, "Significant", "NS")
        )],
    by = "ID")
foo <- function(dt, A, B, title) {
  dt[, table(get(A), get(B))] %>%
  fisher.test %>%
  broom::tidy() %>% 
  mutate(Test = title) %>%
  select(Test, estimate, p.value, conf.low, conf.high)
}

list(
  foo(dt, "PDAffected", "ControlAging", title = "PD affected vs aging in controls"),
  foo(dt, "PDAffected", "PDAging", title = "PD affected vs aging in PDs"),
  foo(dt, "ControlAging", "PDAging", title = "Aging in controls vs aging in PDs")
  ) %>%
purrr::map(as.data.table) %>%
rbindlist %>%
kable() %>%
kable_styling(bootstrap_options = c("hover"))
```


## Overlap of affected genes

Overlap of ALP genes aging in control and PD samples

```{r, include=TRUE}
A <- pfcax %>%
  .[, list(ID, Chr, SNP, Gene, 
    FC = C.GroupCTRL.Age, 
    P = P.GroupCTRL.Age, 
    Q = p.adjust(P.GroupCTRL.Age, "fdr"))
  ] %>%
  .[, list(HealthyAging = wmean(FC, -log(P)), IsHealthyAging = any(Q < 0.05, na.rm=TRUE)), Gene]

B <- pfcax %>%
  .[, list(ID, Chr, SNP, Gene, 
    FC = C.GroupPD.Age, 
    P = P.GroupPD.Age, 
    Q = p.adjust(P.GroupPD.Age, "fdr"))
  ] %>%
  .[, list(PDAging = wmean(FC, -log(P)), IsPDAging = any(Q < 0.05, na.rm=TRUE)), Gene]

C  <- pfcx %>%
  .[, list(PD = wmean(logFC, -log(P.Value)), IsPD = any(adj.P.Val < 0.05, na.rm=TRUE)), Gene]
dt <- merge(merge(A, B, by = "Gene"), C, by = "Gene")

list(
  foo(dt, "IsPD", "IsHealthyAging", title = "PD affected vs aging in controls"),
  foo(dt, "IsPD", "IsPDAging", title = "PD affected vs aging in PDs"),
  foo(dt, "IsHealthyAging", "IsPDAging", title = "Aging in controls vs aging in PDs")
  ) %>%
purrr::map(as.data.table) %>%
rbindlist %>%
kable() %>%
kable_styling(bootstrap_options = c("hover"))
```


# Olfactory bulb, padlock

```{r}
ofbax <- fread("../Brain_AgeAcceleration_OFB_Padlock_CGonly/www/m6_Brain_AgeAcceleration_OFB_Padlock_CGonly.csv")
ofbad <- readRDS("../Brain_AgeAcceleration_OFB_Padlock_CGonly/www/m6_methylation.RDS")
```

```{r, include=TRUE, results='asis'}
n <- nrow(ofbax[Type == "CG"])
glue("- Number of CpG loci {n}")
n <- nrow(ofbax[Type != "CG"])
glue("- Number of CpH loci {n}")
t <- ofbad$key[, table(Group)]
glue("- Number of PD cases: {t[2]} and controls: {t[1]}")
n <- ofbax[, sum(p.adjust(P.GroupCTRL.Age, "fdr") < 0.05, na.rm=TRUE)]
glue("- Number of significant loci {n}")
n <- ofbax[p.adjust(P.GroupCTRL.Age, "fdr") < 0.05, length(unique(Gene))]
glue("- Number of genes affected {n}")
```

Dominant direction of methylation

```{r, include=TRUE}
t <-
  ofbax[, list(
            Significant = ifelse(p.adjust(P.GroupCTRL.Age, "fdr") < 0.05, "Significant", "NS"), 
            Direction = ifelse(C.GroupCTRL.Age < 0, "Hypo-M", "Hyper-M")
           )] %>%
  .[, table(Significant, Direction)] %>%
  .[c("NS", "Significant"), c("Hypo-M", "Hyper-M")]

# Show the table
knitr::kable(t) %>%
kable_styling(bootstrap_options = c("hover"),
              full_width = FALSE)

# Show fisher test result
t %>% fisher.test %>% 
tidy() %>%
kable() %>%
kable_styling(bootstrap_options = c("hover"))
```

```{r}
url <- "../Brain_AgeAcceleration_OFB_Padlock_CGonly/www/m6_Brain_AgeAcceleration_OFB_Padlock_CGonly.csv"
stopifnot(file.exists(url))
```

- **[Additional file 13](`r url`) DNA methylation changes in aging olfactory bulb of healthy individuals and PD patients.**



# Enrichment of ALP pathways in the epigenome

```{r}
pd <- foreach(p = pathways, .combine = rbind) %do% {
  genes <- p$genes
  colP  <- "P.DiagnosisControl.Age"
  colFC <- "C.DiagnosisControl.Age"
  dt <- apax[, 
          list(
            Q = get(colP) %>% p.adjust("fdr"),
            FC= get(colFC),
            Gene)]
  res1 <- 
        dt[, table(Q < 0.05, Gene %in% genes)] %>%
          fisher.test %>%
          broom::tidy() %>%
          setDT %>%
          .[, Term := p$term] %>%
          .[, Name := p$name] %>%
          .[, Type := "Appendix"] %>%
          .[, Dir  := "All"]
  res1up <- 
        dt[, table(Q < 0.05 & sign(FC) > 0, Gene %in% genes)] %>%
          fisher.test %>%
          broom::tidy() %>%
          setDT %>%
          .[, Term := p$term] %>%
          .[, Name := p$name] %>%
          .[, Type := "Appendix"] %>%
          .[, Dir  := "Hyper-"]
  res1down <- 
        dt[, table(Q < 0.05 & sign(FC) < 0, Gene %in% genes)] %>%
          fisher.test %>%
          broom::tidy() %>%
          setDT %>%
          .[, Term := p$term] %>%
          .[, Name := p$name] %>%
          .[, Type := "Appendix"] %>%
          .[, Dir  := "Hypo-"]

  colP  <- "P.GroupCTRL.Age"
  colFC <- "C.GroupCTRL.Age"
  dt <- pfcax[, 
          list(
            Q = get(colP) %>% p.adjust("fdr"),
            FC= get(colFC),
            Gene)]
  res2 <- 
        dt[, table( Q < 0.05, Gene %in% genes)] %>% 
          fisher.test %>%
          broom::tidy() %>%
          setDT %>%
          .[, Term := p$term] %>%
          .[, Name := p$name] %>%
          .[, Type := "PFC"] %>%
          .[, Dir  := "All"]
  res2up <- 
        dt[, table( Q < 0.05 & sign(FC) > 0, Gene %in% genes)] %>% 
          fisher.test %>%
          broom::tidy() %>%
          setDT %>%
          .[, Term := p$term] %>%
          .[, Name := p$name] %>%
          .[, Type := "PFC"] %>%
          .[, Dir  := "Hyper-"]

  res2down <- 
        dt[, table( Q < 0.05 & sign(FC) < 0, Gene %in% genes)] %>% 
          fisher.test %>%
          broom::tidy() %>%
          setDT %>%
          .[, Term := p$term] %>%
          .[, Name := p$name] %>%
          .[, Type := "PFC"] %>%
          .[, Dir  := "Hypo-"]

  rbindlist(list(res1, res1up, res1down, res2, res2up, res2down))
}
if (!dir.exists("www")) dir.create("www")
fwrite(pd, file = "www/enrichment_Pathways_DMCs.csv")

```

```{r, include=TRUE}
pd %>%
.[, `P < 0.05` := gtools::stars.pval(p.value)] %>% 
.[order(Type, Dir), ] %>%
.[, list(estimate, p.value, conf.low, conf.high, Type,
				 term = sprintf("%s (%s)", Name, Term),
				 direction = Dir, 
				 p_stars = texPstars(`P < 0.05`))] %>%
kable() %>%
kable_styling(bootstrap_options = c("hover"))
```

```{r, include=TRUE, fig.cap="**Figure S8. ALP pathway changes in DNA methylation in the healthy aging appendix and prefrontal cortex neurons.**"}
ggplot(pd, aes(Name, log(estimate), color = p.value < 0.05)) +
  geom_point(size = 4) + 
  geom_errorbar(aes(ymin = log(conf.low), ymax = log(conf.high)), width = 0.25, size = 0.25, color = "black") +
  geom_hline(yintercept = 0) + 
  ylab("OR, log") +
  scale_color_brewer("P < 0.05", palette="Dark2") +
  facet_grid(Type ~ Dir) + 
  theme_bw(base_size = 14) + 
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 55, hjust = 1),
        legend.position = "top") 
```


Correlation of odds ratios in PFC and appendix

```{r, include=TRUE}
pd[Dir == "Hyper-", list(Name, Type, estimate)] %>%
dcast(Name ~ Type, value.var = 'estimate') %>%
.[, cor.test(Appendix, PFC)] %>%
tidy() %>%
kable %>% 
kable_styling(bootstrap_options="hover")
```

# Aging hallmarks

```{r}
computeORs <- function(dt) {
	require(doParallel)
	registerDoParallel(cores = parallel::detectCores())
  dt <- 
    foreach (gene = unique(dt$Gene), .combine = rbind) %dopar% {
      require(data.table)
      tryCatch({
        t <- dt[, table(Gene == gene, Significant == TRUE)]
        f <- fisher.test(t, alternative = "greater")
        data.table(OR = f$estimate, P = f$p.value, Gene = gene)    
      }, error = function(e) {
        message(e)
        data.table(OR = NA, P = NA, Gene = gene)
      })
    }
  dt
}

epfc <- cache(foo=computeORs, fname="epfc.RDS",
              dt = pfcax[, list(Gene, Significant = p.adjust(P.GroupCTRL.Age, "fdr") < 0.05)])
eapx  <- cache(foo=computeORs, fname="eapx.RDS",
               dt = apax[, list(Gene, Significant = p.adjust(P.DiagnosisControl.Age, "fdr") < 0.05)])
hallmarks <- aggregateRanks(
  list(
    epfc[order(P), Gene],
    eapx[order(P), Gene]
  )
)  
setDT(hallmarks)
setnames(hallmarks, "Name", "Gene")
hallmarks <- 
merge(hallmarks, epfc[, list(Gene, PFC.OR=OR, PFC.P=P)], by = "Gene") %>% 
		merge(., eapx[, list(Gene, APX.OR=OR, APX.P=P)], by = "Gene")

if (!dir.exists("www")) dir.create("www")
write.table(hallmarks[order(Score)], file = "www/Aging_Hallmarks.csv", sep = ",", row.names = FALSE)

```

```{r, include=TRUE}
hallmarks[order(Score)][1:15] %>%
kable %>%
kable_styling(bootstrap_options="hover")
```

```{r}
url <- "./www/Aging_Hallmarks.csv"
stopifnot(file.exists(url))
```

- **[Additional file 14](`r url`) ALP genes consistently epigenetically altered with aging in the healthy appendix and healthy prefrontal cortex neurons.**


# Aging in PD and control individuals


Correlation of absolute aging and PD effect sizes

```{r}
# Appendix
path <- "../Appendix_AgeAcceleration_Padlock/www/m6_Appendix_AgeAcceleration_Padlock.csv"
apxfit <- fread(path)

path <- "../Appendix_PDvsControls_Padlock/www/m6_Appendix_PDvsControls_Padlock.csv"
apxpdfit <- fread(path)

X <- 
  merge(
    apxfit[P.DiagnosisControl.Age < 0.05, list(ID, C.DiagnosisControl.Age)],
    apxpdfit[P.Value < 0.05, list(ID, logFC)], 
    by = "ID") %>%
  setnames(c("ID", "Age", "PD")) %>%
  mutate(Aging="Control aging", Tissue = "Appendix")

X2 <- 
  merge(
    apxfit[P.DiagnosisPD.LBD.Age < 0.05, list(ID, C.DiagnosisPD.LBD.Age)],
    apxpdfit[P.Value < 0.05, list(ID, logFC)], 
    by = "ID") %>%
  setnames(c("ID", "Age", "PD")) %>%
  mutate(Aging="PD aging", Tissue = "Appendix")

pd <- rbind(X, X2)

# PFC (rep)
path <- "../Brain_AgeAcceleration_Padlock/www/m6_Brain_AgeAcceleration_Padlock.csv"
apxfit <- fread(path)

path <- "../Brain_PFCRep_Padlock_withGLU/www/m6_Brain_PFCRep_Padlock_withGLU.csv"
apxpdfit <- fread(path)

X <- 
  merge(
    apxfit[P.GroupCTRL.Age < 0.05, list(ID, C.GroupCTRL.Age)],
    apxpdfit[P.Value < 0.05, list(ID, logFC)], 
    by = "ID") %>%
  setnames(c("ID", "Age", "PD")) %>%
  mutate(Aging="Control aging", Tissue = "PFC")

X2 <- 
  merge(
    apxfit[P.GroupPD.Age < 0.05, list(ID, C.GroupPD.Age)],
    apxpdfit[P.Value < 0.05, list(ID, logFC)], 
    by = "ID") %>%
  setnames(c("ID", "Age", "PD")) %>%
  mutate(Aging="PD aging", Tissue = "PFC")

pd <- rbindlist(list(pd, X, X2))


# OFB
path <- "../Brain_AgeAcceleration_OFB_Padlock_CGonly/www/m6_Brain_AgeAcceleration_OFB_Padlock_CGonly.csv"
apxfit <- fread(path)

path <- "../Brain_OFB_Padlock_CGonly/www/m6_Brain_OFB_Padlock_CGonly.csv"
apxpdfit <- fread(path)

X <- 
  merge(
    apxfit[P.GroupCTRL.Age < 0.05, list(ID, C.GroupCTRL.Age)],
    apxpdfit[P.Value < 0.05, list(ID, logFC)], 
    by = "ID") %>%
  setnames(c("ID", "Age", "PD")) %>%
  mutate(Aging="Control aging", Tissue = "OFB")

X2 <- 
  merge(
    apxfit[P.GroupPD.Age < 0.05, list(ID, C.GroupPD.Age)],
    apxpdfit[P.Value < 0.05, list(ID, logFC)], 
    by = "ID") %>%
  setnames(c("ID", "Age", "PD")) %>%
  mutate(Aging="PD aging", Tissue = "OFB")

pd <- rbindlist(list(pd, X, X2))

textLayer <-
            pd[, cor.test(abs(Age), abs(PD)) %>% broom::tidy(), list(Tissue, Aging)] %>%
            select(Tissue, Aging, estimate, p.value) %>%
            setDT() %>%
            .[, label := 
                  sprintf("r = $%s$, p %s %s", 
                          format(estimate, digits=1), 
                          ifelse(p.value == 0, "<", "="),
                          texP(p.value)
                         )
            ]
```

```{r, include=TRUE, fig.cap="**Figure S9. The relationship between epigenetic changes occurring with aging and epigenetic changes occurring in PD, examining the appendix, olfactory bulb and prefrontal cortex neurons. Concordance of absolute aging effects and absolute PD effects signifies that the cytosine sites differentially methylated in aging are the same sites that are disrupted in PD. Pearson's correlation coefficient r with p-value. ** "}
pd %>%
  ggplot(., aes(abs(Age), abs(PD))) + 
    geom_point(shape = 1, color = "grey") +
    geom_smooth(method = "lm", color = "black", size = 0.5) + 
    xlab("Healthy aging effect, absolute FC") + 
    ylab("PD diagnosis effect, absolute FC") +
    geom_text(data = textLayer, aes(label = TeX(label, output="character")), inherit.aes=FALSE, parse = TRUE, 
      x = -Inf, y = Inf, hjust = -0.1, vjust = 1) +
    facet_grid(Aging ~ Tissue) +
    scale_x_continuous(
      labels = scales::number_format(accuracy = 0.01)) + 
    theme_bw(base_size = 14)
```


Aging rate in appendix transcriptome

```{r, include=TRUE, fig.cap="**Figure S10. The aging rate in healthy and PD samples of appendix transcritome. **"}
fdr <- function(x) { p.adjust(x, "fdr")}
pd <- 
  apar[, list(
    Gene=SYMBOL, 
    Healthy = C.DiagnosisControl.Age,
    HealthySLP = -sign(C.DiagnosisControl.Age) * log(P.DiagnosisControl.Age),
    HealthyQ = fdr(P.DiagnosisControl.Age),
    PD = C.DiagnosisPD.LBD.Age,
    PDSLP = -sign(C.DiagnosisPD.LBD.Age) * log(P.DiagnosisPD.LBD.Age),
    PDQ = fdr(P.DiagnosisPD.LBD.Age)
    )]
pd <- 
  pd[, list(
    Healthy,
    PD,
    IsHealthy = HealthyQ < 0.05,
    IsPD = PDQ < 0.05
  ), Gene]
pd <- pd[IsHealthy | IsPD]
pd[IsHealthy == FALSE, Healthy := 0]
pd[IsPD == FALSE, PD := 0]

pd[, Label := paste(IsHealthy, IsPD)]
pd[, Label := factor(Label, 
                    levels = c("TRUE FALSE", "TRUE TRUE", "FALSE TRUE"),
                    labels = c("Control aging", "Control & PD aging", "PD aging"))]
pd <- pd[order(Label, PD+Healthy)]
pd[, X := 1:.N]

geneList <- c("GBA","NUCKS1","SLC41A1","SIPA1L2","TMEM163","CCNT2","STK39","CHMP2B","MCCC1","TMEM175","DGKQ","FAM200B","CD38","FAM47E","SNCA","HLA-DRB6","HLA-DQA1","KLHL7","NUPL2","GPNMB","MICU3","BAG3","DLG2","MIR4697","LRRK2","OGFOD2","GCH1","TMEM229B","VPS13C","ZNF646","KAT8","ARHGAP27","CRHR1","SPPL2C","MAPT","STH","KANSL1","SYT4","LSM7","DDRGK1","ITPKB","IL1R2","SCN3A","SATB1","NCKIPSD","CDC71","ALAS1","TLR9","DNAH1","BAP1","PHF7","NISCH","STAB1","ITIH3","ITIH4","ANK2","CAMK2D","ELOVL7","ZNF184","CTSB","SORBS3","PDLIM2","C8orf58","BIN3","SH3GL2","FAM171A1","GALC","COQ7","TOX3","ATP6V0A1","PSMC3IP","TUBG2")

require(ggrepel)
pd %>%
  ggplot(aes(x=X, xend=X, y=Healthy, yend=PD)) + 
  geom_segment(size = 0.25, color = "grey85") + 
  geom_segment(
      data = function(x) x[Gene %in% geneList],
      size = 0.5, color = "orange") +   
  geom_point(
    aes(x = X, y = Healthy), color = '#0080FF', size = 0.25) +
  geom_point(
    aes(x = X, y = PD), color = '#FF00FF', size = 0.25) +
  geom_text_repel(
    data = function(x) x[Gene %in% geneList],
    aes(label = Gene)
    ) + 
  ylab("Aging rate, logCPM/year") + 
  xlab("Gene") + 
  theme_bw(base_size=14) + 
  theme(
      axis.line.x = element_blank(),
      axis.ticks.x = element_blank(),
      axis.text.x = element_blank()
    ) 
```
