---
title: "Results Brain"
output:
  html_document:
    toc: true
    toc_depth: 3
    toc_float: 
      collapsed: false
    includes:
      in_header: ../include/in_header.html
      before_body: ../include/before_body.html
      after_body: ../include/after_body.html
runtime: shiny_prerendered
---
<!-- # (c)  Juozas Gordevičius -->

```{r, include=FALSE}
require(broom)
require(knitr)
require(RobustRankAggreg)
require("annotatr")
require("TxDb.Hsapiens.UCSC.hg19.knownGene")
require("org.Hs.eg.db")
source("../code/common.R")
```

# Olfactory bulb

```{r}
ofbx <- fread("../Brain_OFB_Padlock_CGonly/www/m6_Brain_OFB_Padlock_CGonly.csv")
ofbd <- readRDS("../Brain_OFB_Padlock_CGonly/www/m6_methylation.RDS")
```

```{r, include=TRUE, results='asis'}
n <- nrow(ofbx)
glue("- Number of loci {n}")
t <- ofbd$key[, table(Group)]
glue("- Number of PD cases: {t[2]} and controls: {t[1]}")
n <- ofbx[, sum(adj.P.Val < 0.05, na.rm=TRUE)]
glue("- Number of significant loci {n}")
n <- ofbx[adj.P.Val < 0.05, length(unique(Gene))]
glue("- Number of genes affected {n}")
```

```{r, include=TRUE, results='asis'}
value1 <- ofbx[adj.P.Val < 0.05, mean(abs(BetaFC))] %>%
  `*`(100) %>% round
value2 <- ofbx[adj.P.Val < 0.05, sum(adj.P.Val < 0.05, na.rm=TRUE), Gene] %>% .[, mean(V1)] %>% 
  round(digits = 1)
glue("- {value2} differentially methylated sites per affected ALP gene with {value1}% average methylation change")
```


Dominant direction of methylation

```{r, include=TRUE}
t <-
	ofbx[, list(
						Significant = ifelse(adj.P.Val < 0.05, "Significant", "NS"), 
					 	Direction = ifelse(logFC < 0, "Hypo-M", "Hyper-M")
					 )] %>%
	.[, table(Significant, Direction)] %>%
	.[c("NS", "Significant"), c("Hypo-M", "Hyper-M")]

# Show the table
knitr::kable(t) %>%
kable_styling(bootstrap_options = c("hover"),
							full_width = FALSE)

# Show fisher test result
t %>% fisher.test %>% 
tidy() %>%
kable() %>%
kable_styling(bootstrap_options = c("hover"))
```

```{r}
url <- "../Brain_OFB_Padlock_CGonly/www/m6_Brain_OFB_Padlock_CGonly.csv"
stopifnot(file.exists(url))
```

- **[Additional File 4](`r url`) DNA methylation changes in the olfactory bulb of PD patients relative to controls.**


```{r, include=TRUE, fig.cap="**Figure S5. DNA methylation changes in olfactory bulb of PD patients and controls.**"}
knitr::include_graphics(
	"../Brain_OFB_Padlock_CGonly/index_files/figure-html/unnamed-chunk-31-1.png")
```


# Prefrontal cortex neurons (primary cohort)

```{r}
pfcx <- fread("../Brain_PFCRep_Padlock_withGLU/www/m6_Brain_PFCRep_Padlock_withGLU.csv")
pfcd <- readRDS("../Brain_PFCRep_Padlock_withGLU/www/m6_methylation.RDS")
```

```{r, include=TRUE, results='asis'}
n <- nrow(pfcx[Type == "CG"])
glue("- Number of CpG loci {n}")
n <- nrow(pfcx[Type != "CG"])
glue("- Number of CpH loci {n}")
t <- pfcd$key[, table(Group)]
glue("- Number of PD cases: {t[2]} and controls: {t[1]}")
n <- pfcx[, sum(adj.P.Val < 0.05, na.rm=TRUE)]
glue("- Number of significant loci {n}")
n <- pfcx[adj.P.Val < 0.05, length(unique(Gene))]
glue("- Number of genes affected {n}")
```

```{r, include=TRUE, results='asis'}
value0 <- pfcx[adj.P.Val < 0.05 & Type != "CG", mean(abs(BetaFC))] %>%
  `*`(100) %>% round
value1 <- pfcx[adj.P.Val < 0.05 & Type == "CG", mean(abs(BetaFC))] %>%
  `*`(100) %>% round

value2 <- pfcx[adj.P.Val < 0.05, sum(adj.P.Val < 0.05, na.rm=TRUE), Gene] %>% .[, mean(V1)] %>% 
  round(digits = 1)
glue("- {value2} differentially methylated sites per affected ALP gene with average methylation change {value1}% in CpG and {value0}% in CpH sites")
```


Dominant direction of methylation

```{r, include=TRUE}
t <-
	pfcx[, list(
						Significant = ifelse(adj.P.Val < 0.05, "Significant", "NS"), 
					 	Direction = ifelse(logFC < 0, "Hypo-M", "Hyper-M")
					 )] %>%
	.[, table(Significant, Direction)] %>%
	.[c("NS", "Significant"), c("Hypo-M", "Hyper-M")]

# Show the table
knitr::kable(t) %>%
kable_styling(bootstrap_options = c("hover"),
							full_width = FALSE)

# Show fisher test result
t %>% fisher.test %>% 
tidy() %>%
kable() %>%
kable_styling(bootstrap_options = c("hover"))
```

```{r}
url <- "../Brain_PFCRep_Padlock_withGLU_Braak/www/m6_Brain_PFCRep_Padlock_withGLU_Braak.csv"
stopifnot(file.exists(url))
```

- **[Additional File 5](`r url`) DNA methylation changes in the prefrontal cortex neurons of PD patients relative to controls, primary cohort.**

```{r, include=TRUE, fig.cap="**Figure S6. DNA methylation changes in prefrontal cortex neurons of PD patients and controls.**"}
knitr::include_graphics(
	"../Brain_PFCRep_Padlock_withGLU/index_files/figure-html/unnamed-chunk-32-1.png")
```


# Prefrontal cortex neurons (large) stage-wise analysis

```{r, include=TRUE}
stagex <- fread("../Brain_PFCRep_Padlock_withGLU_Braak/www/m6_Brain_PFCRep_Padlock_withGLU_Braak.csv")
staged <- readRDS("../Brain_PFCRep_Padlock_withGLU_Braak/www/m6_methylation.RDS")
contrasts <- c(
  "StageEarly - StageControl",
  "StageLate - StageControl"
  )
```

```{r, include=TRUE, results='asis'}
t <- staged$key[PD_Braak_Stage_Group != "PD_Braak_5_6", table(Group)]
glue("- Number of PD cases: {t[2]} and controls: {t[1]}")
p <- "P.StageEarly - StageControl"
c <- "C.StageEarly - StageControl"
n <- stagex[, sum(p.adjust(get(p), "fdr") < 0.05, na.rm=TRUE)]
glue("- Number of significant loci {n}")
n <- stagex[p.adjust(get(p), "fdr") < 0.05, length(unique(Gene))]
glue("- Number of genes affected {n}")
```

Overlap of each contrast with the general result

```{r}
dt <- 
foreach(contrast = contrasts, .combine = rbind) %do% {
  colP <- paste0("P.", contrast)
  A <- pfcx[, list(ID, Significant = adj.P.Val < 0.05)]
  B <- stagex[, list(ID, Significant = p.adjust(get(colP), "fdr") < 0.05)]
  merge(A, B, by = "ID") %>%
  .[, table(Significant.x, Significant.y)] %>%
  fisher.test() %>%
  broom::tidy() %>% 
  setDT %>% 
  .[, Contrast := contrast]
}
```

```{r, include=TRUE}
dt %>%
kable %>%
kable_styling(bootstrap_options="hover")
```


```{r}
url <- "../Brain_PFCRep_Padlock_withGLU_Braak/www/m6_Brain_PFCRep_Padlock_withGLU_Braak.csv"
stopifnot(file.exists(url))
```

- **[Additional file 6](`r url`) DNA methylation changes in prefrontal cortex neurons with PD Braak stage.**



# Prefrontal cortex neurons (replication cohort)

```{r}
pfc2x <- fread("../Brain_PFC_Padlock_CGonly/www/m6_Brain_PFC_Padlock_CGonly.csv")
pfc2d <- readRDS("../Brain_PFC_Padlock_CGonly/www/m6_methylation.RDS")
```


```{r, include=TRUE, results='asis'}
n <- nrow(pfc2x)
glue("- Number of loci {n}")
t <- pfc2d$key[, table(Group)]
glue("- Number of PD cases: {t[2]} and controls: {t[1]}")
n <- pfc2x[, sum(adj.P.Val < 0.05, na.rm=TRUE)]
glue("- Number of significant loci {n}")
n <- pfc2x[adj.P.Val < 0.05, length(unique(Gene))]
glue("- Number of genes affected {n}")
```

```{r, include=TRUE, results='asis'}
value1 <- pfc2x[adj.P.Val < 0.05, mean(abs(BetaFC))] %>%
  `*`(100) %>% round
value2 <- pfc2x[adj.P.Val < 0.05, sum(adj.P.Val < 0.05, na.rm=TRUE), Gene] %>% .[, mean(V1)] %>% 
  round(digits = 1)
glue("- {value2} differentially methylated sites per affected ALP gene with {value1}% average methylation change")
```


Dominant direction of methylation

```{r, include=TRUE}
t <-
  pfc2x[, list(
            Significant = ifelse(adj.P.Val < 0.05, "Significant", "NS"), 
            Direction = ifelse(logFC < 0, "Hypo-M", "Hyper-M")
           )] %>%
  .[, table(Significant, Direction)] %>%
  .[c("NS", "Significant"), c("Hypo-M", "Hyper-M")]

# Show the table
knitr::kable(t) %>%
kable_styling(bootstrap_options = c("hover"),
              full_width = FALSE)

# Show fisher test result
t %>% fisher.test %>% 
tidy() %>%
kable() %>%
kable_styling(bootstrap_options = c("hover"))
```

```{r}
url <- "../Brain_PFC_Padlock_CGonly/www/m6_Brain_PFC_Padlock_CGonly.csv"
stopifnot(file.exists(url))
```

- **[Additional file 7](`r url`) DNA methylation changes in prefrontal cortex neurons replication cohort.**

# Overlap of appendix and brain

## At the level of individual cytosines

```{r}
apx <- fread("../Appendix_PDvsControls_Padlock/www/m6_Appendix_PDvsControls_Padlock.csv")
```

Appendix - PFC 

```{r, include=TRUE}
t <- merge( 
          apx[, list(ID, InApx=adj.P.Val < 0.05)],
          pfcx[, list(ID, InPFC=adj.P.Val < 0.05)], 
          by = "ID"
  ) %>% 
  .[, table(InApx, InPFC)]
f <- fisher.test(t)

# Show the table
knitr::kable(t) %>%
kable_styling(bootstrap_options = c("hover"),
              full_width = FALSE)

# Show fisher test result
f %>%
tidy() %>%
kable() %>%
kable_styling(bootstrap_options = c("hover"))
```

Appendix - PFC 2

```{r, include=TRUE}
t <- merge( 
          apx[, list(ID, InApx=adj.P.Val < 0.05)],
          pfc2x[, list(ID, InPFC=adj.P.Val < 0.05)], 
          by = "ID"
  ) %>% 
  .[, table(InApx, InPFC)]
f <- fisher.test(t)

# Show the table
knitr::kable(t) %>%
kable_styling(bootstrap_options = c("hover"),
              full_width = FALSE)

# Show fisher test result
f %>%
tidy() %>%
kable() %>%
kable_styling(bootstrap_options = c("hover"))
```

Appendix - OFB

```{r, include=TRUE}
t <- merge( 
          apx[, list(ID, InApx=adj.P.Val < 0.05)],
          ofbx[, list(ID, InOFB=adj.P.Val < 0.05)], 
          by = "ID"
  ) %>% 
  .[, table(InApx, InOFB)]
f <- fisher.test(t)

# Show the table
knitr::kable(t) %>%
kable_styling(bootstrap_options = c("hover"),
              full_width = FALSE)

# Show fisher test result
f %>%
tidy() %>%
kable() %>%
kable_styling(bootstrap_options = c("hover"))
```

## At the level of genes

Consider a gene affected if it has at least one significant DMC.

```{r, include=TRUE}
A <- apx[ , sum(adj.P.Val < 0.05, na.rm=TRUE), list(Gene=toupper(Gene))]
B <- pfcx[ , sum(adj.P.Val < 0.05, na.rm=TRUE), list(Gene=toupper(Gene))]
C <- ofbx[ , sum(adj.P.Val < 0.05, na.rm=TRUE), list(Gene=toupper(Gene))]
```

Appendix with olfactory bulb:

```{r, include=TRUE}
t <- merge(A, C, by = "Gene") %>%  .[, table(V1.x > 0, V1.y > 0)]
f <- fisher.test(t)

# Show the table
knitr::kable(t) %>%
kable_styling(bootstrap_options = c("hover"),
							full_width = FALSE)

# Show fisher test result
f %>%
tidy() %>%
kable() %>%
kable_styling(bootstrap_options = c("hover"))
```

Appendix with prefrontal cortex neurons

```{r, include=TRUE}
t <- merge(A, B, by = "Gene") %>% .[, table(V1.x > 0, V1.y > 0)]
f <- fisher.test(t)

# Show the table
knitr::kable(t) %>%
kable_styling(bootstrap_options = c("hover"),
							full_width = FALSE)

# Show fisher test result
f %>%
tidy() %>%
kable() %>%
kable_styling(bootstrap_options = c("hover"))
```

Overlap of differentially methylated ALP genes in PFC replicate cohort and apppendix

```{r, include=TRUE}
D <- pfc2x[ , sum(adj.P.Val < 0.05, na.rm=TRUE), list(Gene=toupper(Gene))]


t <- merge(A, D, by = "Gene") %>% .[, table(V1.x > 0, V1.y > 0)] 
f <- fisher.test(t)

# Show the table
knitr::kable(t) %>%
kable_styling(bootstrap_options = c("hover"),
							full_width = FALSE)

# Show fisher test result
f %>%
tidy() %>%
kable() %>%
kable_styling(bootstrap_options = c("hover"))
```


# Hallmarks of PD

```{r}


computeORs <- function(dt, cores = parallel::detectCores()) {
	require(doParallel)
	registerDoParallel(cores = cores)
  dt <- 
    foreach (gene = unique(dt$Gene), .combine = rbind) %dopar% {
      require(data.table)
      tryCatch({
        t <- dt[, table(Gene == gene, Significant == TRUE)]
        f <- fisher.test(t, alternative = "greater")
        data.table(OR = f$estimate, P = f$p.value, Gene = gene)    
      }, error = function(e) {
        message(e)
        data.table(OR = NA, P = NA, Gene = gene)
      })
    }
  dt
}

epfcr <- cache(foo=computeORs, fname="epfcr.RDS", 
               dt = pfc2x[, list(Gene, Significant = adj.P.Val < 0.05)])
epfc  <- cache(foo=computeORs, fname="epfc.RDS",
               dt = pfcx[, list(Gene, Significant = adj.P.Val < 0.05)])
eofb  <- cache(foo=computeORs, fname="eofb.RDS",
               dt = ofbx[, list(Gene, Significant = adj.P.Val < 0.05)])
eapp  <- cache(foo=computeORs, fname="eapp.RDS",
               dt = apx[, list(Gene, Significant = adj.P.Val < 0.05)])
hallmarks <- aggregateRanks(
  list(
    # epfcr[order(P), Gene],
    epfc[order(P), Gene],
    eofb[order(P), Gene],
    eapp[order(P), Gene]
  )
)  
setDT(hallmarks)
setnames(hallmarks, "Name", "Gene")
hallmarks <- 
merge(hallmarks, epfcr[, list(Gene, PFCII.OR=OR, PFCII.P=P)], by = "Gene") %>% 
		merge(., epfc[, list(Gene, PFC.OR=OR, PFC.P=P)], by = "Gene") %>%
		merge(., eofb[, list(Gene, OFB.OR=OR, OFB.P=P)], by = "Gene") %>%
		merge(., eapp[, list(Gene, APP.OR=OR, APP.P=P)], by = "Gene")

if (!dir.exists("www")) dir.create("www")
write.table(hallmarks[order(Score)], file = "www/PD_Hallmarks.csv", sep = ",", row.names = FALSE)

```

```{r, include=TRUE}
hallmarks[order(Score)][1:40] %>%
kable %>%
kable_styling(bootstrap_options="hover")
```

```{r}
url <- "./www/PD_Hallmarks.csv"
stopifnot(file.exists(url))
```

- **[Additional file 8](`r url`) ALP genes consistently epigenetically disrupted across the PD appendix and brain.**


# Overlap of appendix and brain

```{r}
genome <- "hg19"
annots <- c('hg19_cpgs', 'hg19_basicgenes', 'hg19_genes_intergenic')
annotations_hg19 <- build_annotations(genome = genome, annotations = annots)

myFisherTest <- function(t) {
  require(data.table)
  require(dplyr)
  tryCatch({
    exp <- sum(t[2,])/sum(t)
    obs <- t[2,2] / sum(t[,2])
    f <- t %>% fisher.test
    data.table(
      `Expected, %` = exp,
      `Obseverd, %` = obs,
      OR            = f$estimate, 
      Lo            = f$conf.int[1], 
      Hi            = f$conf.int[2], 
      P             = f$p.value,
      `P < 0.05`    = gtools::stars.pval(f$p.value)
    )
  }, error = function(e) { 
    message(e)
    return(NULL)
  })
}


##################################
# load psychEncode data
##################################
loadPsychEncodeChIP <- function() {
  require("bedr")
  stopifnot(check.binary("bedtools"))
  
  # AC marks
  dAC <- fread("../input/Enh_Prom/EpiMap_H3K27ac_Neu_sort.txt") %>%
          setnames(c("Chr", "Start", "End", "Name"))
  dAC <- 
    bedr(engine = "bedtools", input = list(i = dAC), method = "merge", 
        check.zero.based = FALSE, check.chr = FALSE, 
        check.valid = FALSE, check.merge = FALSE, check.sort = FALSE, 
        verbose = TRUE)  %>%
    setDT %>%
    setnames(c("Chr", "Start", "End")) %>%
    setkey(Chr, Start, End) %>%
    .[, ID := paste0(Chr, ":", Start, "-", End)]
  
  # ME marks
  dME <- fread("../input/Enh_Prom/EpiMap_H3K4me3_Neu_sort.txt") %>%
          setnames(c("Chr", "Start", "End", "Name"))
  dME <- 
    bedr(engine = "bedtools", input = list(i = dME), method = "merge", 
        check.zero.based = FALSE, check.chr = FALSE, 
        check.valid = FALSE, check.merge = FALSE, check.sort = FALSE, 
        verbose = TRUE)  %>%
    setDT %>%
    setnames(c("Chr", "Start", "End")) %>%
    setkey(Chr, Start, End) %>%
    .[, ID := paste0(Chr, ":", Start, "-", End)]
  return(list(dAC, dME))  
}

##################################
# load appendix ChIP data
##################################
loadAppendixChIP <- function() {
  d <- Sys.glob("../input/PD_Padlock_Appendix_ChIP/")
  fAC <- file.path(d, "ACvsinput_peaks.narrowPeak_IDR0.05_conservative_final.bed")
  fME <- file.path(d, "MEvsinput_peaks.narrowPeak_IDR0.05_conservative_final.bed")

  # AC marks
  dAC <- fread(fAC) %>%
          setnames(c("Chr", "Start", "End", "Name", "Score", "Strand")) %>% 
          .[, list(Chr, Start, End)] %>%
          setkey(Chr, Start, End) %>%
          .[, ID := paste0(Chr, ":", Start, "-", End)]
  
  # ME marks
  dME <- fread(fME) %>%
          setnames(c("Chr", "Start", "End", "Name", "Score", "Strand")) %>% 
          .[, list(Chr, Start, End)] %>%
          setkey(Chr, Start, End) %>%
          .[, ID := paste0(Chr, ":", Start, "-", End)]
  return(list(dAC, dME))  
}


##################################
# annotate fits with marks from ChIP
##################################
annotateChIP <- function(myfits, dAC, dME) {
  ids <- foverlaps(
    myfits[, list(Chr, Start = SNP, End = SNP, ID)], 
    dAC, 
    by.x = c("Chr", "Start", "End"),
    by.y = c("Chr", "Start", "End")) %>%
    .[!is.na(Start), ]
  myfits[match(ids$i.ID, ID), AC := ids$ID]

  ids <- foverlaps(
    myfits[, list(Chr, Start = SNP, End = SNP, ID)], 
    dME, 
    by.x = c("Chr", "Start", "End"),
    by.y = c("Chr", "Start", "End")) %>%
    .[!is.na(Start), ]
  myfits[match(ids$i.ID, ID), ME := ids$ID]
  
  # Now mark enhancers and promoters
  myfits %>%
    # Promoter is AC only
    .[, Promoter := ifelse( !is.na(AC) & is.na(ME), AC, NA)] %>%
    # Poised enhancer is ME only
    .[, PoisedEnhancer := ifelse( is.na(AC) & !is.na(ME), ME, NA)] %>%
    # Active enhancer is when both ME and AC are present
    .[, ActiveEnhancer := ifelse( !is.na(ME) & !is.na(AC), paste(AC, ME), NA)] %>%
    # Drop unused cols
    .[, AC := NULL] %>%
    .[, ME := NULL]
  return(myfits)
}  
  
##################################
# annotate fits with genomic elements
##################################

annotateGenomic <- function(myfits, annotations) {
  require(GenomicRanges)
  require(annotatr)
  require(data.table)
  fitsgr <- with(myfits, GRanges(seqnames=Chr, IRanges(SNP, SNP)))
  if (!is.null(myfits$str))
    strand(fitsgr) <- myfits$str
  values(fitsgr) <- myfits[, list(ID)]
  annot <- annotate_regions(
    regions = fitsgr,
    annotations = annotations,
    ignore.strand = TRUE,
    quiet = FALSE)
  annot <- data.frame(annot) %>% setDT

  types <- annot[, unique(annot.type)]
  for (type in types) {
    X <- annot[annot.type == type, 
               paste(
                 ifelse(is.na(annot.symbol[1]), "", annot.symbol[1]), 
                 annot.id[1]
                ), ID]
    setnames(X, "V1", type)
    myfits <- merge(myfits, 
          X,
          all.x = TRUE, 
          by = "ID")
  }
  myfits
}

##################################
# annotate all fits
##################################

g(brainME, brainAC) %=% loadPsychEncodeChIP()
g(apxME, apxAC) %=% loadAppendixChIP()

# Find overlaps of brain and apx chip
extendRegions <- function(dt, exby = 2500) {
  dt <- dt[, list(Chr, Start = Start - exby, End = End + exby, Col = 1)]
  dt <- bedr(engine = "bedtools", 
             input = list(i = dt), 
             method = "merge", 
             params = "-c 1 -o count",
             check.zero.based = FALSE, 
             check.chr = FALSE, 
             check.valid = FALSE, 
             check.merge = FALSE, 
             check.sort = FALSE, 
             verbose = TRUE)
  setDT(dt)
  return(dt)
}
giveClusterIDs <- function(X) {
  curid <- 0
  X <- X %>% 
    diff %>%
    c(0, .)
  foreach (x = X) %do% {
    if (x == 0) {
      curid
    } else  {
      curid <- curid + 1
      curid
    } 
  } %>% unlist
}


# aggregate brain and appendix chip
dtME <- rbind(
    extendRegions(brainME, exby = 500),
    extendRegions(apxME, exby = 500)) %>% 
  setkey(Chr, Start, End)
dtME <- extendRegions(dtME, exby = 0)
dtME[, Cluster := giveClusterIDs(Col > 1), Chr]
dtME <- dtME[Col > 1, list(Start = min(Start), End = max(End)), list(Chr, Cluster)] %>%
  dplyr::select(-Cluster)
dtME[, ID := paste0(Chr, ":", Start, "-", End)]
setkey(dtME, Chr, Start, End)

dtAC <- rbind(
    extendRegions(brainAC),
    extendRegions(apxAC)) %>% 
  setkey(Chr, Start, End)
dtAC <- extendRegions(dtAC, exby = 0)
dtAC[, Cluster := giveClusterIDs(Col > 1), Chr]
dtAC <- dtAC[Col > 1, list(Start = min(Start), End = max(End)), list(Chr, Cluster)] %>%
  dplyr::select(-Cluster)
dtAC[, ID := paste0(Chr, ":", Start, "-", End)]
setkey(dtAC, Chr, Start, End)

data <- list(APX = apx, `PFC` = pfcx, `PFC II` = pfc2x, OFB = ofbx)
# d <- data[[1]]
annotated <- withCluster(foreach(d = data ) %dopar% {
  require(data.table)
  require(dplyr)
  dAnno <- copy(d)
  dAnno <- annotateGenomic(dAnno, annotations_hg19)
  dAnno <- annotateChIP(dAnno, dtAC, dtME)
  setnames(dAnno, 
           c("Promoter", "PoisedEnhancer", "ActiveEnhancer"),
           c("Pro", "PE", "AE")
           )
  dAnno
}, nNodes = 4)
names(annotated) <- names(data)


##################################
# Compute overlaps
##################################
n <- ncol(annotated[[1]])
dimensions <- colnames(annotated[[1]])[15:n]
dt <- withCluster(
  foreach( brain = names(annotated)[-1], .combine = rbind) %:% 
  foreach( dim   = dimensions, .combine = rbind) %dopar% {
    require(data.table)
    require(dplyr)
    message(brain, ": ", dim)
    A <- annotated[[1]]
    B <- annotated[[brain]]
  
    # idsA <- A[order(P.Value)][1:10000, ID]
    # idsB <- B[order(P.Value)][1:10000, ID]
    idsA <- A[order(P.Value)][1:5000, ID]
    idsB <- B[order(P.Value)][1:5000, ID]

    tryCatch({
      merge(
        A[!is.na(get(dim)), any(ID %in% idsA), get(dim)],
        B[!is.na(get(dim)), any(ID %in% idsB), get(dim)],
        by = "get"
      ) %>%
      .[, table(V1.x, V1.y)] %>%
      myFisherTest() %>%
        .[, DS := brain] %>%
        .[, Dimension := dim]    
    }, error = function(e) {
      message(e)
      return(NULL)
    })
  }, 
  nNodes = 5)
dt[, Type := "All"]

# Same analysis for top hyper-methylated probes
dtHyper <- withCluster(
  foreach( brain = names(annotated)[-1], .combine = rbind) %:% 
  foreach( dim   = dimensions, .combine = rbind) %dopar% {
    require(data.table)
    require(dplyr)
    message(brain, ": ", dim)
    A <- annotated[[1]]
    B <- annotated[[brain]]
  
    # idsA <- A[order(P.Value)][1:10000, ID]
    # idsB <- B[order(P.Value)][1:10000, ID]
    o <- function(s, fc, p) { 
      order(-1 * sign(s) * sign(fc) * log(p), decreasing=TRUE)
    }
    idsA <- A[o(logFC, P.Value, s=1)][1:5000, ID]
    idsB <- B[o(logFC, P.Value, s=1)][1:5000, ID]

    tryCatch({
      merge(
        A[!is.na(get(dim)), any(ID %in% idsA), get(dim)],
        B[!is.na(get(dim)), any(ID %in% idsB), get(dim)],
        by = "get"
      ) %>%
      .[, table(V1.x, V1.y)] %>%
      myFisherTest() %>%
        .[, DS := brain] %>%
        .[, Dimension := dim]    
    }, error = function(e) {
      message(e)
      return(NULL)
    })
  }, 
  nNodes = 5)
dtHyper[, Type := "Hyper"]

# Same analysis for top hypo-methylated probes
dtHypo <- withCluster(
  foreach( brain = names(annotated)[-1], .combine = rbind) %:% 
  foreach( dim   = dimensions, .combine = rbind) %dopar% {
    require(data.table)
    require(dplyr)
    message(brain, ": ", dim)
    A <- annotated[[1]]
    B <- annotated[[brain]]
  
    # idsA <- A[order(P.Value)][1:10000, ID]
    # idsB <- B[order(P.Value)][1:10000, ID]
    o <- function(s, fc, p) { 
      order(-1 * sign(s) * sign(fc) * log(p), decreasing=TRUE)
    }
    idsA <- A[o(logFC, P.Value, s=-1)][1:5000, ID]
    idsB <- B[o(logFC, P.Value, s=-1)][1:5000, ID]

    tryCatch({
      merge(
        A[!is.na(get(dim)), any(ID %in% idsA), get(dim)],
        B[!is.na(get(dim)), any(ID %in% idsB), get(dim)],
        by = "get"
      ) %>%
      .[, table(V1.x, V1.y)] %>%
      myFisherTest() %>%
        .[, DS := brain] %>%
        .[, Dimension := dim]    
    }, error = function(e) {
      message(e)
      return(NULL)
    })
  }, 
  nNodes = 5)
dtHypo[, Type := "Hypo"]


##################################
# Plot
##################################
plot_pC <- function(dt) {
  pd <- copy(dt)
  pd[grepl("hg19_genes", Dimension), Group := "1.Genes"]
  pd[grepl("hg19_cpg", Dimension), Group := "2.CpG Islands"]
  # pd[grepl("Apx", Dimension), Group := "4.Appendix\nChIP-seq"]
  # pd[grepl("Brain", Dimension), Group := "3.PsychEncode\nChIP-seq"]
  pd[is.na(Group), Group := "3.Appendix + PsychEncode\nChIP-seq"]
  pd[, Dimension := gsub("hg19_|cpg_|genes_", "", Dimension) %>% Hmisc::capitalize()]
  pd[, Dimension := gsub("Apx|Brain", "", Dimension)]
  pd[, Dimension := gsub("PE", "Poised Enh", Dimension)]
  pd[, Dimension := gsub("Pro$", "Active Promoter", Dimension)]
  pd[, Dimension := gsub("AE", "Active Enh", Dimension)]
  pd[, Dimension := gsub("Inter$", "Open Sea", Dimension)]
  pd[, OR := log(OR)]
  pd[, Lo := log(Lo)]
  pd[, Hi := log(Hi)]
  pd[, Shape := ifelse(P < 0.05, "P < 0.05", "P > 0.05")]
  pd[, Stars := ""]
  pd[P < 0.05, Stars := "*"]
  pd[P < 0.01, Stars := "**"]
  pd[P < 0.001, Stars := "***"]

  # Deal with ordering
  pd$Group <- factor(pd$Group)
  levels(pd$Group) <- levels(pd$Group) %>% gsub("[1-4]\\.", "", .)

  pd$Dimension <- factor(pd$Dimension, levels = 
                c("Intergenic", "1to5kb", "Promoters", "5UTRs", "Exons", "Introns", "3UTRs",
                "Active Enh", "Poised Enh", "Active Promoter",
                "Open Sea", "Shelves", "Shores", "Islands"))
  pd$DS <- factor(pd$DS, levels = c("APX", "PFC", "PFC II", "OFB"))
  mycolors <- c(APX="#7570b3", OFB="#e7298a", `PFC II`="#66a61e", PFC="#e6ab02")

  ggplot(pd, aes(Dimension, OR, group=DS, color=DS,
                 ymin=Lo, ymax=Hi, shape = Shape)) + 
    geom_point(
      size = 1.5,
      position=position_dodge(width=0.75)
    ) + 
    geom_errorbar(
      width=0.5, size = 0.25, color = "grey",
      position=position_dodge(width=0.75)
    ) + 
    geom_hline(yintercept = 0, color = "grey50", size = 0.25) +
    geom_text(
      aes(label = Stars, y = Hi), color = "black",
      parse = FALSE, size = 2.2, hjust = -0.5, vjust = 0.75,
      position=position_dodge(width=0.75)
      ) +
    #ylim(-0.5, 2.1) + 
    facet_wrap(~Group, ncol=4, scales="free_y") + 
    ylab("Odds ratio, log") +
    scale_shape_manual("", values = c(16, 1), guide = FALSE) + 
    scale_color_manual("Dataset", values = mycolors, drop=FALSE) + 
    theme_publication() +
    coord_flip() +
    guides(shape = FALSE) + 
    theme(
      axis.title.y = element_blank())

}

theme_publication <- function(font = 7) {
  theme(
    text = element_text(family = "Helvetica", size = font, color = "black"),
    plot.title = element_text(size = font, color = "black"),
    
    panel.background = element_rect(fill = "transparent", colour = NA),
    panel.border = element_blank(),
    plot.background = element_rect(fill = "transparent", colour = NA),
    
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    
    axis.line = element_line(color = "black"),
    axis.text = element_text(size = font, color = "black"), 
    axis.ticks = element_line(size = 0.3, color = "black"),
    axis.title = element_text(size = font, color = "black"),
    
    # strip.background = element_rect(color = "white", fill = "white"),
    strip.text.x = element_text(size = font, color = "black"),
    strip.text.y = element_text(size = font, color = "black"),
    
    legend.background = element_rect(
      fill = alpha("white", 0),
      color = alpha("white", 0)
    ),
    legend.key = element_rect(color = NA, fill = NA),
    legend.key.size = unit(0.2, "cm"),    
    legend.text = element_text(size = font)
  )
}

pC <- plot_pC(dt)
pCHyper <- plot_pC(dtHyper)+ggtitle("Hyper-methylation")
pCHypo <- plot_pC(dtHypo)+ggtitle("Hypo-methylation")
saveRDS(pC, "pC.RDS")
```

```{r, include=TRUE, fig.width=7.2, fig.height=4, fig.cap="**Figure S19. Genomic elements exhibiting similar changes in DNA methylation in the PD brain and PD appendix stratified by the direction of methylation change.**"}
require("cowplot")
ggdraw() +
  draw_plot(pCHyper + theme(legend.position = "none"),
            x = 0, y = 0.55, width = 1, height = 0.45) +
  draw_plot(pCHypo + theme(legend.position = "bottom"),
            x = 0, y = 0, width = 1, height = 0.55)
```
