---
title: "Results Proteomics"
output:
  html_document:
    toc: true
    toc_depth: 3
    toc_float: 
      collapsed: false
    includes:
      in_header: ../include/in_header.html
      before_body: ../include/before_body.html
      after_body: ../include/after_body.html
runtime: shiny_prerendered
---
<!-- # (c)  Juozas Gordevičius -->

```{r, include=FALSE}
if (!require(weights)) {
	install.packages("weights")
}
require(broom)
require(knitr)
source("../code/common.R")
```

```{r}
# Load other appendix fits
padlApx <- fread("../Appendix_PDvsControls_Padlock/www/m6_Appendix_PDvsControls_Padlock.csv")
rnasApx <- fread("../Appendix_PDvsControls_RNAseq/www/r1_Appendix_PDvsControls_RNAseq.csv")
padlPfc <- fread("../Brain_PFCRep_Padlock_withGLU/www/m6_Brain_PFCRep_Padlock_withGLU.csv")
padlPfc2 <- fread("../Brain_PFC_Padlock_CGonly/www/m6_Brain_PFC_Padlock_CGonly.csv")

# Load the proteomics fits
protApx <- fread("www/190319-Labrie-APP_Group1vs2_Quan.csv")
protPfc <- fread("www/190319-Labrie-PFCX_Group1vs2_Quan.csv")


# extract relevant data
protApx[, Gene := Description %>% gsub("^.*GN=", "", .) %>% gsub(" PE.*$", "", .)]
protPfc[, Gene := Description %>% gsub("^.*GN=", "", .) %>% gsub(" PE.*$", "", .)]
protApx <- 
  protApx[, list(
    Accession, 
    Gene, 
    FC = `Abundance Ratio (log2): (2) / (1)`,
    Q  = `Abundance Ratio Adj. P-Value: (2) / (1)`,
    A1 = `Abundances (Normalized): F1: Sample, Group 1, 1, 1`,
    A2 = `Abundances (Normalized): F2: Sample, Group 1, 1, 2`,
    A3 = `Abundances (Normalized): F6: Sample, Group 1, 1, 3`,
    A4 = `Abundances (Normalized): F3: Sample, Group 2, 2, 1`,
    A5 = `Abundances (Normalized): F4: Sample, Group 2, 2, 2`,
    A6 = `Abundances (Normalized): F5: Sample, Group 2, 2, 3`
  )]
protPfc <- 
  protPfc[, list(
    Accession, 
    Gene, 
    FC = `Abundance Ratio (log2): (2) / (1)`,
    Q  = `Abundance Ratio Adj. P-Value: (2) / (1)`,
    A1 = `Abundances (Normalized): F1: Sample, Group 1, 1, 1`,
    A2 = `Abundances (Normalized): F2: Sample, Group 1, 1, 2`,
    A3 = `Abundances (Normalized): F6: Sample, Group 1, 1, 3`,
    A4 = `Abundances (Normalized): F3: Sample, Group 2, 2, 1`,
    A5 = `Abundances (Normalized): F4: Sample, Group 2, 2, 2`,
    A6 = `Abundances (Normalized): F5: Sample, Group 2, 2, 3`
  )]

# filter reliable readings
isReliableReading <- function(gA, gB) {
  sum(!is.na(gA)) >= 2 & sum(!is.na(gB)) >= 2
}
protApx[, Reliable := isReliableReading(c(A1, A2, A3), c(A4, A5, A6)), Accession]
protPfc[, Reliable := isReliableReading(c(A1, A2, A3), c(A4, A5, A6)), Accession]

genesFromALP <- padlPfc[, unique(Gene)]

```

# Appendix 

```{r, include=TRUE, results='asis'}
n <- protApx[, .N]
glue("- Number of proteins investigated {n}")
n <- protApx[Reliable == TRUE & abs(FC) > 0.2, .N]
glue("- Number of altered proteins {n}")
```


Dominant direction of change

```{r, include=TRUE}
t <-
	protApx[, list(
						Changed = ifelse(Reliable == TRUE & abs(FC) > 0.2, "Changed", "NS"), 
					 	Direction = ifelse(FC < 0, "Down", "Up")
					 )] %>%
	.[, table(Changed, Direction)] %>%
	.[c("NS", "Changed"), c("Down", "Up")]

# Show the table
knitr::kable(t) %>%
kable_styling(bootstrap_options = c("hover"),
							full_width = FALSE)

# Show fisher test result
t %>% fisher.test %>% 
tidy() %>%
kable() %>%
kable_styling(bootstrap_options = c("hover"))
```

```{r}
if (!dir.exists("www")) dir.create("www")
protApx[Reliable == TRUE] %>%
  .[, DiffProtein := abs(FC) > 0.2] %>%
  .[, IsALPProtein := Gene %in% genesFromALP] %>%
  merge(
    padlApx[, list(EpigeneticallyAltered = sum(adj.P.Val < 0.05, na.rm=TRUE)>0), Gene],
    by = "Gene", all.x = TRUE) %>%
  merge(
    rnasApx[, list(DiffExpressed = adj.P.Val < 0.05), list(Gene = SYMBOL)],
    by = "Gene", all.x = TRUE)  %>%
  select(-Reliable) %>% 
  fwrite(file = "www/protApx.csv")
# url <- "./www/190319-Labrie-APP_Group1vs2_Quan.csv"
url <- "./www/protApx.csv"
stopifnot(file.exists(url))
```

- **[Additional file 9](`r url`) Quantitative proteomics analysis of the PD appendix.**



# Prefrontal cortex neurons 

```{r, include=TRUE, results='asis'}
n <- protPfc[, .N]
glue("- Number of proteins investigated {n}")
n <- protPfc[Reliable == TRUE & abs(FC) > 0.2, .N]
glue("- Number of altered proteins {n}")
```


Dominant direction of change

```{r, include=TRUE}
t <-
	protPfc[, list(
						Changed = ifelse(Reliable == TRUE & abs(FC) > 0.2, "Changed", "NS"), 
					 	Direction = ifelse(FC < 0, "Down", "Up")
					 )] %>%
	.[, table(Changed, Direction)] %>%
	.[c("NS", "Changed"), c("Down", "Up")]

# Show the table
knitr::kable(t) %>%
kable_styling(bootstrap_options = c("hover"),
							full_width = FALSE)

# Show fisher test result
t %>% fisher.test %>% 
tidy() %>%
kable() %>%
kable_styling(bootstrap_options = c("hover"))
```

```{r}
protPfc[Reliable == TRUE] %>%
  .[, DiffProtein := abs(FC) > 0.2] %>%
  merge(
    padlPfc[, list(EpigeneticallyAlteredInPFC = sum(adj.P.Val < 0.05, na.rm=TRUE)>0), Gene],
    by = "Gene", all.x = TRUE) %>%
  merge(
    padlPfc2[, list(EpigeneticallyAlteredInPFC2 = sum(adj.P.Val < 0.05, na.rm=TRUE)>0), Gene],
    by = "Gene", all.x = TRUE) %>%
  select(-Reliable) %>% 
  fwrite(file = "www/protPfc.csv")
url <- "./www/protPfc.csv"
stopifnot(file.exists(url))
```

- **[Additional file 10](`r url`) Quantitative proteomic analysis of the PD prefrontal cortex.**

# Correlation of appendix and PFC proteome fold changes

```{r, include=TRUE}
merge(
  protApx[, list(Gene, FC)],
  protPfc[, list(Gene, FC)],
  by = "Gene") %>%
.[, cor.test(FC.x, FC.y)] %>%
tidy() %>%
kable %>% 
kable_styling(bootstrap_options="hover")
```

The proteins with epigenetic disregulation in PD appendix

```{r}
dt <-
	padlApx[, list(
	  NoDMCs = sum(adj.P.Val < 0.05, na.rm=TRUE),
	  NoDMCs_Hyper = sum(adj.P.Val < 0.05 & logFC > 0, na.rm=TRUE),
	  NoDMCs_Hypo  = sum(adj.P.Val < 0.05 & logFC < 0, na.rm=TRUE)),
	  Gene] %>%
	merge(., protApx[Reliable == TRUE, list(Accession, Gene, FC)], by = "Gene") %>%
	.[NoDMCs > 0 & abs(FC) > 0.2] %>%
	merge(., protPfc[Reliable == TRUE & abs(FC) > 0.2, list(Gene, FC)], by = "Gene", all.x = TRUE) %>%
	setnames(c("FC.x", "FC.y"), c("FC.Appendix", "FC.PFC"))
```

```{r, include=TRUE}
dt %>%
kable(caption="**Table S1. Fold change and percent change of ALP proteins of epigenetically dysregulated genes in the PD appendix.**") %>%
kable_styling(bootstrap_options="hover")
```

Correlation of fold changes in the above table

```{r, include=TRUE}
dt %>%
.[, cor.test(FC.Appendix, FC.PFC, use="p")] %>%
tidy %>%
kable %>%
kable_styling(bootstrap_options="hover")
```

# Proteins altered in both apx and brain

## Appendix 

The fraction of genes epigenetically altered among
ALP genes altered in proteomic analysis:

```{r, include=TRUE}
genesFromALP <- padlPfc[, unique(Gene)]

# The ALP genes that have had a change in proteomics data
alpGenesAlteredInProteomicsApx <- protApx[Reliable == TRUE & abs(FC) > 0.2 & Gene %in% genesFromALP, Gene]

# The ALP genes that have epigentic change
alpGenesModifiedInAppendix <- padlApx[adj.P.Val < 0.05, unique(Gene)]

# fraction of the proteins that have a change in epigenetic data?
mean(alpGenesAlteredInProteomicsApx %in% alpGenesModifiedInAppendix)
```

## PFC

The fraction of genes altered in both proteomic and epigenetic analyses among
ALP genes altered in proteomic analysis:

```{r, include=TRUE}
# The ALP genes that have had a change in proteomics data
alpGenesAlteredInProteomicsPFC <- protPfc[Reliable == TRUE & abs(FC) > 0.2 & Gene %in% genesFromALP, Gene]

# fraction of these genes have a change in epigenetic data?
alpGenesModifiedInPFC <- unique(c(padlPfc[adj.P.Val < 0.05, unique(Gene)], padlPfc2[adj.P.Val < 0.05, unique(Gene)]))

mean(alpGenesAlteredInProteomicsPFC %in% alpGenesModifiedInPFC)
```


## Both


Of the ALP genes that are showing protein changes in the appendix AND brain how many are epigenetically altered in both appendix and brain?

```{r, include=TRUE}
# The ALP genes that have had a change in proteomics data
alpGenesAlteredInProteomics <- 
  intersect(
    protApx[Reliable == TRUE & abs(FC) > 0.2 & Gene %in% genesFromALP, Gene],
    protPfc[Reliable == TRUE & abs(FC) > 0.2 & Gene %in% genesFromALP, Gene])

# fraction of these genes have a change in epigenetic data?
alpGenesModified <- 
  intersect(
    padlApx[adj.P.Val < 0.05, unique(Gene)],
    union(
      padlPfc[adj.P.Val < 0.05, unique(Gene)],
      padlPfc2[adj.P.Val < 0.05, unique(Gene)]
    )
  )

mean(alpGenesAlteredInProteomics %in% alpGenesModified)
```


```{r}
# Downloadable list
allGenes <- unique(c(alpGenesAlteredInProteomicsApx, alpGenesAlteredInProteomicsPFC, alpGenesAlteredInProteomics))
res <- data.table(Gene = allGenes)
res[, ProteomicChangeInApx := Gene %in% alpGenesAlteredInProteomicsApx]
res[, EpigeneticChangeInApx := Gene %in% alpGenesModifiedInAppendix]
res[, ProteomicChangeInPFC := Gene %in% alpGenesAlteredInProteomicsPFC]
res[, EpigeneticChangeInPFC := Gene %in% alpGenesModifiedInPFC]
res[, ProteomicChangeInBoth := Gene %in% alpGenesAlteredInProteomics]
res[, EpigeneticChangeInBoth := Gene %in% alpGenesModified]
```

```{r, include=TRUE}
res %>%
.[order(Gene)] %>%
kable(caption="**Table S2. ALP proteins with differential abundance in either appendix or prefrontal cortex neurons and their corresponding epigenetic alterations.**") %>%
kable_styling(bootstrap_options="hover")
```

```{r}
# if (!dir.exists("www")) dir.create("www")
# fwrite(res, file = "www/protein2epigenome.csv")
# # url <- "./www/190319-Labrie-APP_Group1vs2_Quan.csv"
# url <- "./www/protein2epigenome.csv"
# stopifnot(file.exists(url))
# - **[Additional file NNN](`r url`) Overlap of ALP gene proteomic and epigenomic changes.**
```


# iTRAQ data

Lysosome-enriched fraction

```{r}
pathways <- readRDS("../Discover_Pathways/humanPathways.RDS")
lys <- fread("./www/iTRAQ_lysozome_human_NoNormalization_proteins.csv")
key <- fread("./www/iTRAQ_lysozome_key.tsv")


# Compute differences
# lys
dt <- 
  suppressWarnings(
    lys[,
      list(`Protein ID`, `Gene ID`, Coverage = `Coverage (%)`, 
        Unique = `#Unique`,
        `3681`=as.numeric(`Intensity Control 3681(TMT10-126)`), 
        `3649`=as.numeric(`Intensity Control 3649(TMT10-127N)`), 
        `3623`=as.numeric(`Intensity PD 3623(TMT10-127C)`), 
        `3823`=as.numeric(`Intensity Control 3823(TMT10-128N)`), 
        `3604`=as.numeric(`Intensity PD 3604(TMT10-128C)`), 
        `3825`=as.numeric(`Intensity PD 3825(TMT10-129N)`), 
        `3531`=as.numeric(`Intensity PD 3531(TMT10-129C)`), 
        `3645`=as.numeric(`Intensity PD 3645(TMT10-130N)`),
        `97-10-A163a`=as.numeric(`Intensity Control 97-10-A163A(TMT10-130C)`), 
        `08-01-A021a`=as.numeric(`Intensity Control 08-01-A021a(TMT10-131)`)
      )
    ])

dt <- 
  dt %>%
    .[Unique > 2] %>%
    reshape2::melt(id.vars = 1:4) %>% 
    setDT %>% 
    .[, value := log(value + 1)] %>%
    .[, value := scale(value), variable]
dt <- merge(dt, key, by.x = 'variable', by.y = 'case number')
```

The number of proteins interrogated across all samples: 
```{r, include=TRUE} 
dt[, length(unique(`Gene ID`))]
```

The fraction of ALP genes interrogated in this analysis: 
```{r, include=TRUE} 
genesFromALP %in% dt[, unique(`Gene ID`)] %>% mean
```

The fraction of lysosome pathway genes interrogated:
```{r, include=TRUE} 
unique(pathways[[3]]$genes) %in% dt[, unique(`Gene ID`)]  %>% mean
```


```{r}
# Plot of value distribution after scale
dt %>%
  ggplot(aes(variable, value, color = `summary diagnosis`)) + 
  geom_violin()

# PCA 
pca <- dt %>%
  reshape2::acast(`Protein ID` ~ variable, value.var = 'value') %>%
  as.data.frame %>%
  t %>%
  prcomp
pairs(pca$x[, 1:3], col = key[, `summary diagnosis`] %>% as.factor)
# One sample may be an ooutlier (C5), but let's keep it for now
```

A robust linear model implementing a robust t-test for comparison between cases and controls

```{r}
# Compute linear model fits
X <- reshape2::acast(dt, `Protein ID` ~ variable, value.var = 'value')
D <- model.matrix( ~ `summary diagnosis` + `age (years)` + sex + `postmortem interval (h)`,
   data = key[match(colnames(X), `case number`)])

fit <- 
  lmFit(X, D, method="robust", maxit=100) %>% 
  eBayes() %>%
  topTable(coef=2, number=Inf, sort.by = "none") %>%
  as.data.table(keep.rownames=TRUE)
setnames(fit, "rn", "Protein ID")
fitLys <- 
  merge(
    fit, 
    lys[!duplicated(`Gene ID`), list(`Protein ID` = as.character(`Protein ID`), `Gene ID`)], 
    by = "Protein ID")
```

Number of significant hits: 

```{r, include=TRUE}
fitLys[, list( `Q < 0.05` = sum(adj.P.Val < 0.05),
              `Q < 0.2`  = sum(adj.P.Val < 0.2),
              `P < 0.05` = sum(P.Value < 0.05),
              .N)]
```


Number of differentially abundant ALP genes that are also epigenetically affected

```{r, include=TRUE}
epiAffected <- padlApx[adj.P.Val < 0.05, unique(Gene)]
fitLys[adj.P.Val < 0.05] %>%
  .[`Gene ID` %in% epiAffected]  
```



Correlation of fold changes in label-free and iTRAQ data using
proteins that were deemed significant in the previous dataset datasets.


```{r, include=TRUE}
require(weights)
merge(
  protApx[Reliable == TRUE & abs(FC) > 0.2, list(Gene, FC)],
  fitLys[!duplicated(`Gene ID`), list(Gene = `Gene ID`, FC = logFC, P=P.Value)],
  by = "Gene") %>%
.[, wtd.cor(FC.x, FC.y, weight=-log(P))] %>%
kable() %>%
kable_styling(bootstrap_options="hover")
# 5.425238e-16
```

Overlap of affected proteins in label-free and iTRAQ-labeled data

```{r, include=TRUE}
merge(
  protApx[Reliable == TRUE, list(Gene, FC)],
  fitLys[!duplicated(`Gene ID`), list(Gene = `Gene ID`, FC = logFC, P=P.Value, Q=adj.P.Val)],
  by = "Gene") %>%
.[, table( abs(FC.x) > 0.2, Q < 0.05 )] %>% 
fisher.test %>%
broom::tidy() %>%
kable() %>%
kable_styling(bootstrap_options="hover")
```

```{r}
url <- "./www/itraqApx.csv"
dt %>% 
  reshape2::acast(`Protein ID` ~ variable + `summary diagnosis`, value.var = "value") %>%
  as.data.table(keep.rownames=TRUE) %>%
  setnames("rn", "Protein ID") %>%
  merge(fitLys, by = "Protein ID") %>%
  fwrite(url)
stopifnot(file.exists(url))
```

- **[Additional file 11](`r url`) iTRAQ-labeled proteomics analysis of the PD appendix.**

