---
title: "etc"
output:
  html_document:
    toc: true
    toc_depth: 3
    toc_float: 
      collapsed: false
    includes:
      in_header: ../include/in_header.html
      before_body: ../include/before_body.html
      after_body: ../include/after_body.html
runtime: shiny_prerendered
---
<!-- # (c)  Juozas Gordevičius -->

```{r, include=FALSE}
source("../code/common.R")
pacman::p_load("cowplot")
pacman::p_load("pheatmap")
pacman::p_load("ggrepel")
pacman::p_load("xlsx")

# Data primers
if (!dir.exists("www")) 
	dir.create('www')

```

# Summary of all results at ALP gene level

```{r}
pfcr <- "../Brain_PFCRep_Padlock_withGLU/www/m6_Brain_PFCRep_Padlock_withGLU.csv" %>% Sys.glob %>% fread
pfc  <- "../Brain_PFC_Padlock_CGonly/www/m6*.csv" %>% Sys.glob %>% fread
ofb  <- "../Brain_OFB_Padlock_CGonly/www/m6*.csv" %>% Sys.glob %>% fread
app  <- "../Appendix_PDvsControls_Padlock/www/m6_*.csv" %>% Sys.glob %>% fread
apprna <- "../Appendix_PDvsControls_RNAseq/www/r1_*.csv" %>% Sys.glob %>% fread
aapp <- fread("../Appendix_AgeAcceleration_Padlock/www/m6_Appendix_AgeAcceleration_Padlock.csv")
apfc <- fread("../Brain_AgeAcceleration_Padlock/www/m6_Brain_AgeAcceleration_Padlock.csv")
aofb <- fread("../Brain_AgeAcceleration_OFB_Padlock_CGonly/www/m6_Brain_AgeAcceleration_OFB_Padlock_CGonly.csv")

# Mice
mcp <- Sys.glob("../Mice_CecalPatch_Padlock/www/m6_*.csv") %>% fread
mdss <-Sys.glob("../Mice_DSS*/www/m6_*.csv") %>% fread
  

geneList <- c("GBA","NUCKS1","SLC41A1","SIPA1L2","TMEM163","CCNT2","STK39","CHMP2B","MCCC1","TMEM175","DGKQ","FAM200B","CD38","FAM47E","SNCA","HLA-DRB6","HLA-DQA1","KLHL7","NUPL2","GPNMB","MICU3","BAG3","DLG2","MIR4697","LRRK2","OGFOD2","GCH1","TMEM229B","VPS13C","ZNF646","KAT8","ARHGAP27","CRHR1","SPPL2C","MAPT","STH","KANSL1","SYT4","LSM7","DDRGK1","ITPKB","IL1R2","SCN3A","SATB1","NCKIPSD","CDC71","ALAS1","TLR9","DNAH1","BAP1","PHF7","NISCH","STAB1","ITIH3","ITIH4","ANK2","CAMK2D","ELOVL7","ZNF184","CTSB","SORBS3","PDLIM2","C8orf58","BIN3","SH3GL2","FAM171A1","GALC","COQ7","TOX3","ATP6V0A1","PSMC3IP","TUBG2")


wmean <- function(x, w) {
  i <- is.na(x) | is.na(w)
  x <- x[!i]; w <- w[!i]
  weighted.mean(x, w, na.rm=TRUE)
}

fdr <- function(x) { p.adjust(x, "fdr")}

# Failsafe Fisher's test
myFisherTest <- function(t, ...) {
  require(data.table)
  require(dplyr)
  tryCatch({
    exp <- sum(t[2,])/sum(t)
    obs <- t[2,2] / sum(t[,2])
    f <- t %>% fisher.test(...)
    data.table(
      `Expected, %` = exp,
      `Obseverd, %` = obs,
      OR            = f$estimate, 
      Lo            = f$conf.int[1], 
      Hi            = f$conf.int[2], 
      P             = f$p.value,
      `P < 0.05`    = gtools::stars.pval(f$p.value)
    )
  }, error = function(e) { 
    message(e)
    return(NULL)
  })
}


# Classify genes into hyper- and hypo- modified
classifyGene <- function(logfc, p, q, qth = 0.05, retNAforNoDMC = FALSE) {
  nhyper <- sum(q < qth & logfc > 0, na.rm=TRUE)
  nhypo <- sum(q < qth & logfc < 0, na.rm=TRUE)
  if (nhyper > nhypo) {
    return("Hyper-")
  } else if (nhyper < nhypo) {
    return("Hypo-")
  } else {
    meanfc <- weighted.mean(logfc, -log(p), na.rm=TRUE)
    hasdmc <- sum(q < qth, na.rm=TRUE) > 0
    if (hasdmc <= 0 & retNAforNoDMC == TRUE)
      return("NS")
    if (sign(meanfc) > 0) 
      return("Hyper-")
    else 
      return("Hypo-")
  }
}


# Compute enrichment
computeORs <- function(dt, cores = parallel::detectCores()) {
	require(doParallel)
	registerDoParallel(cores = cores)
  dt <- 
    foreach (gene = unique(dt$Gene), .combine = rbind) %dopar% {
      require(data.table)
      tryCatch({
        t <- dt[, table(Gene == gene, Significant == TRUE)]
        f <- fisher.test(t, alternative = "greater")
        data.table(OR = f$estimate, P = f$p.value, Gene = gene)    
      }, error = function(e) {
        message(e)
        data.table(OR = NA, P = NA, Gene = gene)
      })
    }
  dt
}
```

```{r}
# Compute the enrichment and sign of each gene and in each condition

# PFC replicate dataset
epfcr <- 
	readRDS("../Results_Brain/epfcr.RDS") %>%
	merge( 
		pfc[, classifyGene(logFC, P.Value, adj.P.Val, qth=0.05), Gene],
		by = "Gene") %>%
	.[, OR := ifelse(V1 == "Hyper-", OR, -OR)] %>%
	.[, list(Gene, `PFC Replicte, OR` = OR, `PFC Replicate, p` = P)]

# PFC
epfc <- 
	readRDS("../Results_Brain/epfc.RDS") %>%
	merge( 
		pfcr[, classifyGene(logFC, P.Value, adj.P.Val, qth=0.05), Gene],
		by = "Gene") %>%
	.[, OR := ifelse(V1 == "Hyper-", OR, -OR)] %>%
	.[, list(Gene, `PFC, OR` = OR, `PFC, p` = P)]

# OFB
eofb <- 
	readRDS("../Results_Brain/eofb.RDS") %>%
	merge( 
		ofb[, classifyGene(logFC, P.Value, adj.P.Val, qth=0.05), Gene],
		by = "Gene") %>%
	.[, OR := ifelse(V1 == "Hyper-", OR, -OR)] %>%
	.[, list(Gene, `OFB, OR` = OR, `OFB, p` = P)]

# Appendix
eapp <- 
	readRDS("../Results_Brain/eapp.RDS") %>%
	merge( 
		app[, classifyGene(logFC, P.Value, adj.P.Val, qth=0.05), Gene],
		by = "Gene") %>%
	.[, OR := ifelse(V1 == "Hyper-", OR, -OR)] %>%
	.[, list(Gene, `Appendix, OR` = OR, `Appendix, p` = P)]


# Appendix aging
x <- 
  aapp[, list(Gene, logFC = C.DiagnosisControl.Age, P = P.DiagnosisControl.Age, Q = fdr(P.DiagnosisControl.Age))]
eapp_ACtrl <- 
	cache(foo=computeORs, fname="eapp_aging_ctrl.RDS",
          dt = aapp[, list(Gene, Significant = fdr(P.DiagnosisControl.Age) < 0.05)]) %>%
	merge( 
		x[, classifyGene(logFC, P, Q, qth=0.05), Gene],
		by = "Gene") %>%
	.[, OR := ifelse(V1 == "Hyper-", OR, -OR)] %>%
	.[, list(Gene, `Aging Control Appendix, OR` = OR, `Aging Control Appendix, p` = P)]


x <- 
  aapp[, list(Gene, logFC = C.DiagnosisPD.LBD.Age, P = P.DiagnosisPD.LBD.Age, Q = fdr(P.DiagnosisPD.LBD.Age))]
eapp_APD <-
	cache(foo=computeORs, fname="eapp_aging_pd.RDS",
          dt = aapp[, list(Gene, Significant = fdr(P.DiagnosisPD.LBD.Age) < 0.05)]) %>%
	merge( 
		x[, classifyGene(logFC, P, Q, qth=0.05), Gene],
		by = "Gene") %>%
	.[, OR := ifelse(V1 == "Hyper-", OR, -OR)] %>%
	.[, list(Gene, `Aging PD Appendix, OR` = OR, `Aging PD Appendix, p` = P)]

# PFC aging
x <- 
  apfc[, list(Gene, logFC = C.GroupCTRL.Age, P = P.GroupCTRL.Age, Q = fdr(P.GroupCTRL.Age))]
epfc_ACtrl <- 
	cache(foo=computeORs, fname="epfc_aging_ctrl.RDS",
          dt = apfc[, list(Gene, Significant = fdr(P.GroupCTRL.Age) < 0.05)]) %>%
	merge( 
		x[, classifyGene(logFC, P, Q, qth=0.05), Gene],
		by = "Gene") %>%
	.[, OR := ifelse(V1 == "Hyper-", OR, -OR)] %>%
	.[, list(Gene, `Aging Control PFC, OR` = OR, `Aging Control PFC, p` = P)]

x <- 
  apfc[, list(Gene, logFC = C.GroupPD.Age, P = P.GroupPD.Age, Q = fdr(P.GroupPD.Age))]
epfc_APD <- 
	cache(foo=computeORs, fname="epfc_aging_pdl.RDS",
        dt = apfc[, list(Gene, Significant = fdr(P.GroupPD.Age) < 0.05)]) %>%
	merge( 
		x[, classifyGene(logFC, P, Q, qth=0.05), Gene],
		by = "Gene") %>%
	.[, OR := ifelse(V1 == "Hyper-", OR, -OR)] %>%
	.[, list(Gene, `Aging PD PFC, OR` = OR, `Aging PD PFC, p` = P)]

# OFB aging
x <- 
  aofb[, list(Gene, logFC = C.GroupCTRL.Age, P = P.GroupCTRL.Age, Q = fdr(P.GroupCTRL.Age))]
eofb_ACtrl <- cache(foo=computeORs, fname="eofb_aging_ctrl.RDS",
                    dt = aofb[, list(Gene, Significant = fdr(P.GroupCTRL.Age) < 0.05)]) %>%
	merge( 
		x[, classifyGene(logFC, P, Q, qth=0.05), Gene],
		by = "Gene") %>%
	.[, OR := ifelse(V1 == "Hyper-", OR, -OR)] %>%
	.[, list(Gene, `Aging Control OFB, OR` = OR, `Aging Control OFB, p` = P)]


x <- 
  aofb[, list(Gene, logFC = C.GroupPD.Age, P = P.GroupPD.Age, Q = fdr(P.GroupPD.Age))]
eofb_APD <- 
	cache(foo=computeORs, fname="eofb_aging_pdl.RDS",
        dt = aofb[, list(Gene, Significant = fdr(P.GroupPD.Age) < 0.05)]) %>%
	merge( 
		x[, classifyGene(logFC, P, Q, qth=0.05), Gene],
		by = "Gene") %>%
	.[, OR := ifelse(V1 == "Hyper-", OR, -OR)] %>%
	.[, list(Gene, `Aging PD OFB, OR` = OR, `Aging PD OFB, p` = P)]


# Mice
gmap <- fread("../etc/padlock_genes_human2mouse.csv") %>% 
  .[, list(V1, V2)]
setnames(gmap, gmap[1,] %>% as.character)
gmap <- gmap[-1,]


x <- mdss[, list(Gene, logFC=`C.GTwt_DSS - GTwt_Water`, P=`P.GTwt_DSS - GTwt_Water`)] %>%
  .[, Q := fdr(P)]
edss_wtdss_wtwater <- 
  cache(foo=computeORs, fname="edss_wtdss_wtwater.RDS",
        dt = x[, list(Gene, Significant = Q < 0.05)]) %>%
	merge( 
		x[, classifyGene(logFC, P, Q, qth=0.05), Gene],
		by = "Gene") %>%
	.[, OR := ifelse(V1 == "Hyper-", OR, -OR)] %>%
	.[, list(Gene, `WT_DSS-WT_Water, OR` = OR, `WT_DSS-WT_Water, p` = P)]


x <- mdss[, list(Gene, logFC=`C.GTtg_DSS - GTtg_Water`, P=`P.GTtg_DSS - GTtg_Water`)] %>%
  .[, Q := fdr(P)]
edss_tgdss_tgwater <- 
  cache(foo=computeORs, fname="edss_tgdss_tgwater.RDS",
        dt = x[, list(Gene, Significant = Q < 0.05)]) %>%
	merge( 
		x[, classifyGene(logFC, P, Q, qth=0.05), Gene],
		by = "Gene") %>%
	.[, OR := ifelse(V1 == "Hyper-", OR, -OR)] %>%
	.[, list(Gene, `A30P_DSS-A30P_Water, OR` = OR, `A30P_DSS-A30P_Water, p` = P)]

x <- mdss[, list(Gene, logFC=`C.GTtg_DSS - GTwt_Water`, P=`P.GTtg_DSS - GTwt_Water`)] %>%
  .[, Q := fdr(P)]
edss_tgdss_wtwater <- 
  cache(foo=computeORs, fname="edss_tgdss_wtwater.RDS",
        dt = x[, list(Gene, Significant = Q < 0.05)]) %>%
	merge( 
		x[, classifyGene(logFC, P, Q, qth=0.05), Gene],
		by = "Gene") %>%
	.[, OR := ifelse(V1 == "Hyper-", OR, -OR)] %>%
	.[, list(Gene, `A30P_DSS - WT_Water, OR` = OR, `A30P_DSS - WT_Water, p` = P)]


x <- mdss[, list(Gene, logFC=`C.GTtg_Water - GTwt_Water`, P=`P.GTtg_Water - GTwt_Water`)] %>%
  .[, Q := fdr(P)]
edss_tgwater_wtwater <- 
  cache(foo=computeORs, fname="edss_tgwater_wtwater.RDS",
        dt = x[, list(Gene, Significant = Q < 0.05)]) %>%
	merge( 
		x[, classifyGene(logFC, P, Q, qth=0.05), Gene],
		by = "Gene") %>%
	.[, OR := ifelse(V1 == "Hyper-", OR, -OR)] %>%
	.[, list(Gene, `A30P_Water - WT_Water, OR` = OR, `A30P_Water - WT_Water, p` = P)]

x <- mcp[, list(Gene, logFC, P=P.Value, Q=adj.P.Val)]
emcp <- 
  cache(foo=computeORs, fname="emcp.RDS",
        dt = x[, list(Gene, Significant = Q < 0.05)]) %>%
 	merge( 
		x[, classifyGene(logFC, P, Q, qth=0.05), Gene],
		by = "Gene") %>%
	.[, OR := ifelse(V1 == "Hyper-", OR, -OR)] %>%
	.[, list(Gene, `rAAV a-syn, OR` = OR, `rAAV a-syn, p` = P)]

# Combine human datasets into one

human <- 
	merge(epfcr, epfc, by = "Gene", all = TRUE) %>%
	merge(eofb, by = "Gene", all = TRUE) %>%
	merge(eapp, by = "Gene", all = TRUE) %>%
	merge(eapp_ACtrl, by = "Gene", all = TRUE) %>%
	merge(eapp_APD, by = "Gene", all = TRUE) %>%
	merge(epfc_ACtrl, by = "Gene", all = TRUE) %>%
	merge(epfc_APD, by = "Gene", all = TRUE) %>%
	merge(eofb_ACtrl, by = "Gene", all = TRUE) %>%
	merge(eofb_APD, by = "Gene", all = TRUE)

# Combine mouse datasets
mouse <-
	merge(edss_wtdss_wtwater, edss_tgdss_tgwater, by = "Gene", all = TRUE) %>%
	merge(edss_tgdss_wtwater, by = "Gene", all = TRUE) %>%
	merge(edss_tgwater_wtwater, by = "Gene", all = TRUE) %>%
	merge(emcp, by = "Gene", all = TRUE) %>%
	merge(gmap, by.x = "Gene", by.y = "MGI_mm10") %>%
	setnames("Gene", "MGI_mm10")

# Now merge human and mouse
dt <- merge(human, mouse,
	by.x = "Gene", by.y = "HGNC_hg19", all=TRUE) 

# Mark GWAS genes
dt[, IsGWASGene := Gene %in% geneList]


fwrite(dt, "www/geneEnrichment.csv")
url <- "./www/geneEnrichment.csv"
```

- **[Additional File 18](`r url`) Enrichment of genes with significantly differentially modified cytosines and the dominant direction of modification.**



# Other additional files

```{r}
studies <- list.files(path = "../", pattern = "*Padlock*|*RNAseq*")
studies <- setdiff(studies, "Appendix_AgeAcceleration_Padlock_AgeMatched")
res <-
foreach (study = studies) %do% {
	message("Loading study: ", study)
	path <- file.path("..", study, "www")
	name <- paste0("m6_methylation.RDS")
	if (file.exists(file.path(path, name))) {
		# padlock study
		x <- readRDS(file.path(path, name))
		x$key[, Study := study]
	} else {
		name <- name <- paste0("r1_rnaseq.RDS")
		if (file.exists(file.path(path, name))) {
			# rnaseq study
			x <- readRDS(file.path(path, name))
			x$key[, Study := study]
		} else {
			message(study, ": no file?")
		}
	}
}
names(res) <- studies


file <- "www/sampleKey.xlsx"
unlink(file)
for (study in studies) {
  message(study)
	write.xlsx(res[[study]], file, sheetName=study, 
	  col.names=TRUE, row.names=FALSE, append=TRUE)
}

url <- "./www/sampleKey.xlsx"
```

- **[Additional File 19](`r url`) Demographic and clinical information for human samples.**

<!--  -->

```{r}
file.copy("../etc/Supplementary Data_primers.xlsx", "www/")
```


```{r}
url <- "./www/Supplementary Data_primers.xlsx"
stopifnot(file.exists(url))
```

- **[Additional File 20](`r url`) Padlock probe libraries and study primers.**














