---
title: "etc"
output:
  html_document:
    toc: true
    toc_depth: 3
    toc_float: 
      collapsed: false
    includes:
      in_header: ../include/in_header.html
      before_body: ../include/before_body.html
      after_body: ../include/after_body.html
runtime: shiny_prerendered
---
<!-- # (c)  Juozas GordeviÄius -->

```{r, include=FALSE}
source("../code/common.R")
require("cowplot")
require("xlsx")
```

# Heatmap of all results

```{r}
pfcr <- "../Brain_PFCRep_Padlock_withGLU/www/m6_Brain_PFCRep_Padlock_withGLU.csv" %>% Sys.glob %>% fread
pfc  <- "../Brain_PFC_Padlock_CGonly/www/m6*.csv" %>% Sys.glob %>% fread
ofb  <- "../Brain_OFB_Padlock_CGonly/www/m6*.csv" %>% Sys.glob %>% fread
app  <- "../Appendix_PDvsControls_Padlock/www/m6_*.csv" %>% Sys.glob %>% fread
apprna <- "../Appendix_PDvsControls_RNAseq/www/r1_*.csv" %>% Sys.glob %>% fread
aapp <- fread("../Appendix_AgeAcceleration_Padlock/www/m6_Appendix_AgeAcceleration_Padlock.csv")
apfc <- fread("../Brain_AgeAcceleration_Padlock/www/m6_Brain_AgeAcceleration_Padlock.csv")
aofb <- fread("../Brain_AgeAcceleration_OFB_Padlock_CGonly/www/m6_Brain_AgeAcceleration_OFB_Padlock_CGonly.csv")

# Mice
mcp <- Sys.glob("../Mice_CecalPatch_Padlock/www/m6_*.csv") %>% fread
mdss <-Sys.glob("../Mice_DSS*/www/m6_*.csv") %>% fread
  

geneList <- c("GBA","NUCKS1","SLC41A1","SIPA1L2","TMEM163","CCNT2","STK39","CHMP2B","MCCC1","TMEM175","DGKQ","FAM200B","CD38","FAM47E","SNCA","HLA-DRB6","HLA-DQA1","KLHL7","NUPL2","GPNMB","MICU3","BAG3","DLG2","MIR4697","LRRK2","OGFOD2","GCH1","TMEM229B","VPS13C","ZNF646","KAT8","ARHGAP27","CRHR1","SPPL2C","MAPT","STH","KANSL1","SYT4","LSM7","DDRGK1","ITPKB","IL1R2","SCN3A","SATB1","NCKIPSD","CDC71","ALAS1","TLR9","DNAH1","BAP1","PHF7","NISCH","STAB1","ITIH3","ITIH4","ANK2","CAMK2D","ELOVL7","ZNF184","CTSB","SORBS3","PDLIM2","C8orf58","BIN3","SH3GL2","FAM171A1","GALC","COQ7","TOX3","ATP6V0A1","PSMC3IP","TUBG2")


wmean <- function(x, w) {
  i <- is.na(x) | is.na(w)
  x <- x[!i]; w <- w[!i]
  weighted.mean(x, w, na.rm=TRUE)
}

fdr <- function(x) { p.adjust(x, "fdr")}

# Failsafe Fisher's test
myFisherTest <- function(t, ...) {
  require(data.table)
  require(dplyr)
  tryCatch({
    exp <- sum(t[2,])/sum(t)
    obs <- t[2,2] / sum(t[,2])
    f <- t %>% fisher.test(...)
    data.table(
      `Expected, %` = exp,
      `Obseverd, %` = obs,
      OR            = f$estimate, 
      Lo            = f$conf.int[1], 
      Hi            = f$conf.int[2], 
      P             = f$p.value,
      `P < 0.05`    = gtools::stars.pval(f$p.value)
    )
  }, error = function(e) { 
    message(e)
    return(NULL)
  })
}


# Classify genes into hyper- and hypo- modified
classifyGene <- function(logfc, p, q, qth = 0.05, retNAforNoDMC = FALSE) {
  nhyper <- sum(q < qth & logfc > 0, na.rm=TRUE)
  nhypo <- sum(q < qth & logfc < 0, na.rm=TRUE)
  if (nhyper > nhypo) {
    return("Hyper-")
  } else if (nhyper < nhypo) {
    return("Hypo-")
  } else {
    meanfc <- weighted.mean(logfc, -log(p), na.rm=TRUE)
    hasdmc <- sum(q < qth, na.rm=TRUE) > 0
    if (hasdmc <= 0 & retNAforNoDMC == TRUE)
      return("NS")
    if (sign(meanfc) > 0) 
      return("Hyper-")
    else 
      return("Hypo-")
  }
}


# Compute enrichment
computeORs <- function(dt, cores = parallel::detectCores()) {
	require(doParallel)
	registerDoParallel(cores = cores)
  dt <- 
    foreach (gene = unique(dt$Gene), .combine = rbind) %dopar% {
      require(data.table)
      tryCatch({
        t <- dt[, table(Gene == gene, Significant == TRUE)]
        f <- fisher.test(t, alternative = "greater")
        data.table(OR = f$estimate, P = f$p.value, Gene = gene)    
      }, error = function(e) {
        message(e)
        data.table(OR = NA, P = NA, Gene = gene)
      })
    }
  dt
}
```

```{r}
# Compute the enrichment and sign of each gene and in each condition
# Case/control analyses
epfcr <- readRDS("../Results_Brain/epfcr.RDS")
epfc <- readRDS("../Results_Brain/epfc.RDS")
eofb <- readRDS("../Results_Brain/eofb.RDS")
eapp <- readRDS("../Results_Brain/eapp.RDS")


# Appendix aging
eapp_ACtrl <- cache(foo=computeORs, fname="eapp_aging_ctrl.RDS",
                    dt = aapp[, list(Gene, Significant = fdr(P.DiagnosisControl.Age) < 0.05)])
eapp_APD <- cache(foo=computeORs, fname="eapp_aging_pd.RDS",
                    dt = aapp[, list(Gene, Significant = fdr(P.DiagnosisPD.LBD.Age) < 0.05)])

# PFC aging
epfc_ACtrl <- cache(foo=computeORs, fname="epfc_aging_ctrl.RDS",
                    dt = apfc[, list(Gene, Significant = fdr(P.GroupCTRL.Age) < 0.05)])
epfc_APD <- cache(foo=computeORs, fname="epfc_aging_pdl.RDS",
                    dt = apfc[, list(Gene, Significant = fdr(P.GroupPD.Age) < 0.05)])

# OFB aging
eofb_ACtrl <- cache(foo=computeORs, fname="eofb_aging_ctrl.RDS",
                    dt = aofb[, list(Gene, Significant = fdr(P.GroupCTRL.Age) < 0.05)])
eofb_APD <- cache(foo=computeORs, fname="eofb_aging_pdl.RDS",
                    dt = aofb[, list(Gene, Significant = fdr(P.GroupPD.Age) < 0.05)])


# Mice
gmap <- fread("../etc/padlock_genes_human2mouse.csv") %>% 
  .[, list(V1, V2)]
setnames(gmap, gmap[1,] %>% as.character)
gmap <- gmap[-1,]


x <- mdss[, list(Gene, logFC=`C.GTwt_DSS - GTwt_Water`, P=`P.GTwt_DSS - GTwt_Water`)] %>%
  .[, Q := fdr(P)] %>% 
  .[, list(Gene, Significant = Q < 0.05)]
edss_wtdss_wtwater <- 
  cache(foo=computeORs, fname="edss_wtdss_wtwater.RDS",
        dt = x)

x <- mdss[, list(Gene, logFC=`C.GTtg_DSS - GTtg_Water`, P=`P.GTtg_DSS - GTtg_Water`)] %>%
  .[, Q := fdr(P)] %>% 
  .[, list(Gene, Significant = Q < 0.05)]
edss_tgdss_tgwater <- 
  cache(foo=computeORs, fname="edss_tgdss_tgwater.RDS",
        dt = x)


x <- mdss[, list(Gene, logFC=`C.GTtg_DSS - GTwt_Water`, P=`P.GTtg_DSS - GTwt_Water`)] %>%
  .[, Q := fdr(P)] %>% 
  .[, list(Gene, Significant = Q < 0.05)]
edss_tgdss_wtwater <- 
  cache(foo=computeORs, fname="edss_tgdss_wtwater.RDS",
        dt = x)

x <- mdss[, list(Gene, logFC=`C.GTtg_Water - GTwt_Water`, P=`P.GTtg_Water - GTwt_Water`)] %>%
  .[, Q := fdr(P)] %>% 
  .[, list(Gene, Significant = Q < 0.05)]
edss_tgwater_wtwater <- 
  cache(foo=computeORs, fname="edss_tgwater_wtwater.RDS",
        dt = x)

x <- mcp[, list(Gene, logFC, P=P.Value, Q=adj.P.Val)] %>%
  .[, list(Gene, Significant = Q < 0.05)]
emcp <- 
  cache(foo=computeORs, fname="emcp.RDS",
        dt = x)

```

```{r}
pacman::p_load(pheatmap)
col_pfcr <- 
  epfcr %>%
  merge(pfcr[, list(Dir = classifyGene(logFC, P.Value, adj.P.Val)), Gene],
        by = "Gene") %>%
  .[, Sign := ifelse(Dir == "Hyper-", 1, -1)] %>%
  .[, list(Gene, PFCR = -1 * Sign * log10(P))]
col_pfc <- 
  epfc %>%
  merge(pfc[, list(Dir = classifyGene(logFC, P.Value, adj.P.Val)), Gene],
        by = "Gene") %>%
  .[, Sign := ifelse(Dir == "Hyper-", 1, -1)] %>%
  .[, list(Gene, PFC = -1 * Sign * log10(P))]
col_ofb <- 
  eofb %>%
  merge(ofb[, list(Dir = classifyGene(logFC, P.Value, adj.P.Val)), Gene],
        by = "Gene") %>%
  .[, Sign := ifelse(Dir == "Hyper-", 1, -1)] %>%
  .[, list(Gene, OFB = -1 * Sign * log10(P))]
col_apx <- 
  eapp %>%
  merge(app[, list(Dir = classifyGene(logFC, P.Value, adj.P.Val)), Gene],
        by = "Gene") %>%
  .[, Sign := ifelse(Dir == "Hyper-", 1, -1)] %>%
  .[, list(Gene, APX = -1 * Sign * log10(P))]

col_apx_rna <-
  apprna[, list(Gene = SYMBOL, APX_RNA = -1 * sign(logFC) * log10(P.Value))]

# Apx aging
x <- 
  aapp[, list(Gene, logFC = C.DiagnosisControl.Age, P = P.DiagnosisControl.Age, Q = fdr(P.DiagnosisControl.Age))]
col_apxaging_ctrl <-
  eapp_ACtrl %>%
  merge(x[, list(Dir = classifyGene(logFC, P, Q)), Gene],
        by = "Gene") %>%
  .[, Sign := ifelse(Dir == "Hyper-", 1, -1)] %>%
  .[, list(Gene, APXA_Ctrl = -1 * Sign * log10(P))]

x <- 
  aapp[, list(Gene, logFC = C.DiagnosisPD.LBD.Age, P = P.DiagnosisPD.LBD.Age, Q = fdr(P.DiagnosisPD.LBD.Age))]
col_apxaging_pd <-
  eapp_APD %>%
  merge(x[, list(Dir = classifyGene(logFC, P, Q)), Gene],
        by = "Gene") %>%
  .[, Sign := ifelse(Dir == "Hyper-", 1, -1)] %>%
  .[, list(Gene, APXA_PD = -1 * Sign * log10(P))]

# PFC aging
x <- 
  apfc[, list(Gene, logFC = C.GroupCTRL.Age, P = P.GroupCTRL.Age, Q = fdr(P.GroupCTRL.Age))]
col_pfcaging_ctrl <-
  epfc_ACtrl %>%
  merge(x[, list(Dir = classifyGene(logFC, P, Q)), Gene],
        by = "Gene") %>%
  .[, Sign := ifelse(Dir == "Hyper-", 1, -1)] %>%
  .[, list(Gene, PFCA_Ctrl = -1 * Sign * log10(P))]

x <- 
  apfc[, list(Gene, logFC = C.GroupPD.Age, P = P.GroupPD.Age, Q = fdr(P.GroupPD.Age))]
col_pfcaging_pd <-
  epfc_APD %>%
  merge(x[, list(Dir = classifyGene(logFC, P, Q)), Gene],
        by = "Gene") %>%
  .[, Sign := ifelse(Dir == "Hyper-", 1, -1)] %>%
  .[, list(Gene, PFCA_PD = -1 * Sign * log10(P))]

# OFB aging
x <- 
  aofb[, list(Gene, logFC = C.GroupCTRL.Age, P = P.GroupCTRL.Age, Q = fdr(P.GroupCTRL.Age))]
col_ofbaging_ctrl <-
  eofb_ACtrl %>%
  merge(x[, list(Dir = classifyGene(logFC, P, Q)), Gene],
        by = "Gene") %>%
  .[, Sign := ifelse(Dir == "Hyper-", 1, -1)] %>%
  .[, list(Gene, OFBA_Ctrl = -1 * Sign * log10(P))]

x <- 
  aofb[, list(Gene, logFC = C.GroupPD.Age, P = P.GroupPD.Age, Q = fdr(P.GroupPD.Age))]
col_ofbaging_pd <-
  eofb_APD %>%
  merge(x[, list(Dir = classifyGene(logFC, P, Q)), Gene],
        by = "Gene") %>%
  .[, Sign := ifelse(Dir == "Hyper-", 1, -1)] %>%
  .[, list(Gene, OFBA_PD = -1 * Sign * log10(P))]


# Mice
x <- 
  mcp[, list(Gene, logFC, P = P.Value, Q = adj.P.Val)]
col_mcp <-
  emcp %>%
  merge(x[, list(Dir = classifyGene(logFC, P, Q)), Gene],
        by = "Gene") %>%
  merge(gmap, by.x = "Gene", by.y = "MGI_mm10") %>%
  .[, list(Gene = HGNC_hg19, Dir, P)] %>%
  .[, list(Dir = Dir[which.min(P)], P = P[which.min(P)]), Gene] %>%
  .[, Sign := ifelse(Dir == "Hyper-", 1, -1)] %>%
  .[, list(Gene, `Cecal patch` = -1 * Sign * log10(P))]

x <- 
  mdss[, list(Gene, logFC = `C.GTtg_DSS - GTwt_Water`, P = `P.GTtg_DSS - GTwt_Water`)] %>%
  .[, Q := fdr(P)]
col_dss_tgdss_wtwater <-
  edss_tgdss_wtwater %>%
  merge(x[, list(Dir = classifyGene(logFC, P, Q)), Gene],
        by = "Gene") %>%
  merge(gmap, by.x = "Gene", by.y = "MGI_mm10") %>%
  .[, list(Gene = HGNC_hg19, Dir, P)] %>%
  .[, list(Dir = Dir[which.min(P)], P = P[which.min(P)]), Gene] %>%
  .[, Sign := ifelse(Dir == "Hyper-", 1, -1)] %>%
  .[, list(Gene, `A30P_DSS - WT_Water` = -1 * Sign * log10(P))]

x <- 
  mdss[, list(Gene, logFC = `C.GTwt_DSS - GTwt_Water`, P = `P.GTwt_DSS - GTwt_Water`)] %>%
  .[, Q := fdr(P)]
col_dss_wtdss_wtwater <-
  edss_wtdss_wtwater %>%
  merge(x[, list(Dir = classifyGene(logFC, P, Q)), Gene],
        by = "Gene") %>%
  merge(gmap, by.x = "Gene", by.y = "MGI_mm10") %>%
  .[, list(Gene = HGNC_hg19, Dir, P)] %>%
  .[, list(Dir = Dir[which.min(P)], P = P[which.min(P)]), Gene] %>%
  .[, Sign := ifelse(Dir == "Hyper-", 1, -1)] %>%
  .[, list(Gene, `WT_DSS - WT_Water` = -1 * Sign * log10(P))]

x <- 
  mdss[, list(Gene, logFC = `C.GTtg_DSS - GTtg_Water`, P = `P.GTtg_DSS - GTtg_Water`)] %>%
  .[, Q := fdr(P)]
col_dss_tgdss_tgwater <-
  edss_tgdss_tgwater %>%
  merge(x[, list(Dir = classifyGene(logFC, P, Q)), Gene],
        by = "Gene") %>%
  merge(gmap, by.x = "Gene", by.y = "MGI_mm10") %>%
  .[, list(Gene = HGNC_hg19, Dir, P)] %>%
  .[, list(Dir = Dir[which.min(P)], P = P[which.min(P)]), Gene] %>%
  .[, Sign := ifelse(Dir == "Hyper-", 1, -1)] %>%
  .[, list(Gene, `A30P_DSS - A30P_Water` = -1 * Sign * log10(P))]

x <- 
  mdss[, list(Gene, logFC = `C.GTtg_Water - GTwt_Water`, P = `P.GTtg_Water - GTwt_Water`)] %>%
  .[, Q := fdr(P)]
col_dss_tgwater_wtwater <-
  edss_tgwater_wtwater %>%
  merge(x[, list(Dir = classifyGene(logFC, P, Q)), Gene],
        by = "Gene") %>%
  merge(gmap, by.x = "Gene", by.y = "MGI_mm10") %>%
  .[, list(Gene = HGNC_hg19, Dir, P)] %>%
  .[, list(Dir = Dir[which.min(P)], P = P[which.min(P)]), Gene] %>%
  .[, Sign := ifelse(Dir == "Hyper-", 1, -1)] %>%
  .[, list(Gene, `A30P_Water - WT_Water` = -1 * Sign * log10(P))]


pd <- 
  merge(col_pfcr, col_pfc, by = "Gene", all=TRUE) %>%
  merge(col_ofb, by = "Gene", all=TRUE) %>%
  merge(col_apx, by = "Gene", all=TRUE) %>%
  merge(col_apx_rna, by = "Gene", all.x = TRUE) %>%
  merge(col_apxaging_ctrl, by = "Gene", all=TRUE) %>%
  merge(col_apxaging_pd, by = "Gene", all=TRUE) %>%
  merge(col_pfcaging_ctrl, by = "Gene", all=TRUE) %>%
  merge(col_pfcaging_pd, by = "Gene", all=TRUE) %>%
  merge(col_ofbaging_ctrl, by = "Gene", all=TRUE) %>%
  merge(col_ofbaging_pd, by = "Gene", all=TRUE) %>%
  merge(col_mcp, by = "Gene", all=TRUE) %>%
  merge(col_dss_tgdss_tgwater, by = "Gene", all=TRUE) %>%
  merge(col_dss_tgdss_wtwater, by = "Gene", all=TRUE) %>%
  merge(col_dss_tgwater_wtwater, by = "Gene", all=TRUE) %>%
  merge(col_dss_wtdss_wtwater, by = "Gene", all=TRUE)
pd <- 
  pd[, list(
          Gene,
          `Appendix PD/Control` = APX,
          `Appendix PD/Control RNA-seq` = APX_RNA,
          `PFC PD/Control` = PFCR,
          `PFC II PD/Control` = PFC,
          `OFB PD/Control` = OFB,
          `Appendix Control Aging` = APXA_Ctrl,
          `Appendix PD Aging` = APXA_PD,
          `PFC Control Aging` = PFCA_Ctrl,
          `PFC PD Aging` = PFCA_PD,
          `OFB Control Aging` = OFBA_Ctrl,
          `OFB PD Aging` = OFBA_PD,
          `Cecal patch`,
          `A30P_DSS - A30P_Water`,
          `A30P_DSS - WT_Water`,
          `A30P_Water - WT_Water`,
          `WT_DSS - WT_Water`
  )]
dim(pd)

# Transform NAs into 0s
pd[is.na(pd)] <- 0

# Convert PD to a matrix
pd <- as.matrix(pd, rownames="Gene")

# Remove columns that are entirely made of 0
keep <- apply(pd, 1, function(x) max(abs(x))) %>% `>`(-log10(0.05))
pd <- pd[keep, ]
dim(pd)

pacman::p_load(RColorBrewer)
pheatmap(pd, cluster_cols = FALSE,
         color = rev(brewer.pal(n = 11, name = "RdBu")),
         breaks = seq(-12, 12, length.out = 12),
         gaps_col = c(2, 5, 11),
         fontsize = 5,
         angle_col = 45,
         border_color = "grey80"
         )




```


A three-color heatmap with direction and has-dmc mark.

```{r}

col_pfcr <- 
  pfcr[, 
       list(
         PFCR = classifyGene(logFC, P.Value, adj.P.Val, retNAforNoDMC = TRUE)), 
       Gene]

col_pfc <- 
  pfc[, list(PFC = classifyGene(logFC, P.Value, adj.P.Val, retNAforNoDMC = TRUE)), Gene]

col_ofb <- 
  ofb[, list(OFB = classifyGene(logFC, P.Value, adj.P.Val, retNAforNoDMC = TRUE)), Gene]


col_apx <- 
  app[, list(APX = classifyGene(logFC, P.Value, adj.P.Val, retNAforNoDMC = TRUE)), Gene]

col_apx_rna <-
  apprna[, list(Gene = SYMBOL, Sign = adj.P.Val < 0.05, Dir = ifelse(sign(logFC) > 0, "Hyper-", "Hypo-"))] %>%
  .[, list(Gene, APX_RNA = ifelse(Sign==TRUE, Dir, "NS"))]

# Apx aging
x <- 
  aapp[, list(Gene, logFC = C.DiagnosisControl.Age, P = P.DiagnosisControl.Age, Q = fdr(P.DiagnosisControl.Age))]
col_apxaging_ctrl <-
  x[, list(APXA_Ctrl = classifyGene(logFC, P, Q, retNAforNoDMC = TRUE)), Gene]

x <- 
  aapp[, list(Gene, logFC = C.DiagnosisPD.LBD.Age, P = P.DiagnosisPD.LBD.Age, Q = fdr(P.DiagnosisPD.LBD.Age))]
col_apxaging_pd <-
  x[, list(APXA_PD = classifyGene(logFC, P, Q, retNAforNoDMC = TRUE)), Gene]

# PFC aging
x <- 
  apfc[, list(Gene, logFC = C.GroupCTRL.Age, P = P.GroupCTRL.Age, Q = fdr(P.GroupCTRL.Age))]
col_pfcaging_ctrl <-
  x[, list(PFCA_Ctrl = classifyGene(logFC, P, Q, retNAforNoDMC = TRUE)), Gene]

x <- 
  apfc[, list(Gene, logFC = C.GroupPD.Age, P = P.GroupPD.Age, Q = fdr(P.GroupPD.Age))]
col_pfcaging_pd <-
  x[, list(PFCA_PD = classifyGene(logFC, P, Q, retNAforNoDMC = TRUE)), Gene]

# OFB aging
x <- 
  aofb[, list(Gene, logFC = C.GroupCTRL.Age, P = P.GroupCTRL.Age, Q = fdr(P.GroupCTRL.Age))]
col_ofbaging_ctrl <-
  x[, list(OFBA_Ctrl = classifyGene(logFC, P, Q, retNAforNoDMC = TRUE)), Gene]

x <- 
  aofb[, list(Gene, logFC = C.GroupPD.Age, P = P.GroupPD.Age, Q = fdr(P.GroupPD.Age))]
col_ofbaging_pd <-
  x[, list(OFBA_PD = classifyGene(logFC, P, Q, retNAforNoDMC = TRUE)), Gene]


# Mice
x <- 
  mcp[, list(Gene, logFC, P = P.Value, Q = adj.P.Val)]
col_mcp <-
  x[, list(`Cecal patch` = classifyGene(logFC, P, Q, retNAforNoDMC = TRUE)), Gene] %>%
  merge(gmap, by.x = "Gene", by.y = "MGI_mm10") %>%
  .[, 3:2, with=FALSE] %>%
  setnames("HGNC_hg19", "Gene") %>%
  .[!duplicated(Gene)]

x <- 
  mdss[, list(Gene, logFC = `C.GTtg_DSS - GTwt_Water`, P = `P.GTtg_DSS - GTwt_Water`)] %>%
  .[, Q := fdr(P)]
col_dss_tgdss_wtwater <-
  x[, list(`A30P_DSS - WT_Water` = classifyGene(logFC, P, Q, retNAforNoDMC = TRUE)), Gene] %>%
  merge(gmap, by.x = "Gene", by.y = "MGI_mm10") %>%
  .[, 3:2, with=FALSE] %>%
  setnames("HGNC_hg19", "Gene") %>%
  .[!duplicated(Gene)]

x <- 
  mdss[, list(Gene, logFC = `C.GTwt_DSS - GTwt_Water`, P = `P.GTwt_DSS - GTwt_Water`)] %>%
  .[, Q := fdr(P)]
col_dss_wtdss_wtwater <-
  x[, list(`WT_DSS - WT_Water` = classifyGene(logFC, P, Q, retNAforNoDMC = TRUE)), Gene] %>%
  merge(gmap, by.x = "Gene", by.y = "MGI_mm10") %>%
  .[, 3:2, with=FALSE] %>%
  setnames("HGNC_hg19", "Gene") %>%
  .[!duplicated(Gene)]

x <- 
  mdss[, list(Gene, logFC = `C.GTtg_DSS - GTtg_Water`, P = `P.GTtg_DSS - GTtg_Water`)] %>%
  .[, Q := fdr(P)]
col_dss_tgdss_tgwater <-
  x[, list(`A30P_DSS - A30P_Water` = classifyGene(logFC, P, Q, retNAforNoDMC = TRUE)), Gene] %>%
  merge(gmap, by.x = "Gene", by.y = "MGI_mm10") %>%
  .[, 3:2, with=FALSE] %>%
  setnames("HGNC_hg19", "Gene") %>%
  .[!duplicated(Gene)]

x <- 
  mdss[, list(Gene, logFC = `C.GTtg_Water - GTwt_Water`, P = `P.GTtg_Water - GTwt_Water`)] %>%
  .[, Q := fdr(P)]
col_dss_tgwater_wtwater <-
  x[, list(`A30P_Water - WT_Water` = classifyGene(logFC, P, Q, retNAforNoDMC = TRUE)), Gene] %>%
  merge(gmap, by.x = "Gene", by.y = "MGI_mm10") %>%
  .[, 3:2, with=FALSE] %>%
  setnames("HGNC_hg19", "Gene") %>%
  .[!duplicated(Gene)]


pd <- 
  merge(col_pfcr, col_pfc, by = "Gene", all=TRUE) %>%
  merge(col_ofb, by = "Gene", all=TRUE) %>%
  merge(col_apx, by = "Gene", all=TRUE) %>%
  merge(col_apx_rna, by = "Gene", all.x = TRUE) %>%
  merge(col_apxaging_ctrl, by = "Gene", all=TRUE) %>%
  merge(col_apxaging_pd, by = "Gene", all=TRUE) %>%
  merge(col_pfcaging_ctrl, by = "Gene", all=TRUE) %>%
  merge(col_pfcaging_pd, by = "Gene", all=TRUE) %>%
  merge(col_ofbaging_ctrl, by = "Gene", all=TRUE) %>%
  merge(col_ofbaging_pd, by = "Gene", all=TRUE) %>%
  merge(col_mcp, by = "Gene", all=TRUE) %>%
  merge(col_dss_tgdss_tgwater, by = "Gene", all=TRUE) %>%
  merge(col_dss_tgdss_wtwater, by = "Gene", all=TRUE) %>%
  merge(col_dss_tgwater_wtwater, by = "Gene", all=TRUE) %>%
  merge(col_dss_wtdss_wtwater, by = "Gene", all=TRUE)


pd <- 
  pd[, list(
          Gene,
          `Appendix PD/Control` = APX,
          #`Appendix PD/Control RNA-seq` = APX_RNA,
          `PFC PD/Control` = PFCR,
          `PFC II PD/Control` = PFC,
          `OFB PD/Control` = OFB,
          `Appendix Control Aging` = APXA_Ctrl,
          `Appendix PD Aging` = APXA_PD,
          `PFC Control Aging` = PFCA_Ctrl,
          `PFC PD Aging` = PFCA_PD,
          `OFB Control Aging` = OFBA_Ctrl,
          `OFB PD Aging` = OFBA_PD,
          `Cecal patch`,
          `A30P_DSS - A30P_Water`,
          `A30P_DSS - WT_Water`,
          `WT_DSS - WT_Water`,
          `A30P_Water - WT_Water`
  )]


# Convert PD to a matrix
pd <- as.matrix(pd, rownames="Gene")
# Transform NAs into 0s
pd[is.na(pd)] <- "NS"
```


Plot for human tissues only

```{r}
mypd <- 
  pd[, c(
      "Appendix PD/Control",
      "PFC PD/Control",
      "PFC II PD/Control",
      "OFB PD/Control",
      "Appendix Control Aging",
      "Appendix PD Aging",
      "PFC Control Aging",
      "PFC PD Aging",
      "OFB Control Aging",
      "OFB PD Aging")]
# Remove columns that are entirely made of NS
keep <- apply(mypd, 1, function(x) any(x != "NS"))
mypd <- mypd[keep, ]
mypd[mypd == "NS"] <- 0
mypd[mypd == "Hypo-"] <- -1
mypd[mypd == "Hyper-"] <- 1
n <- rownames(mypd)
mypd <- apply(mypd, 2, as.numeric)
rownames(mypd) <- n
rownames(mypd)[ !rownames(mypd) %in% geneList] <- ""
rownames(mypd)[ rownames(mypd) == "BAG3"] <- "              BAG3"


require(ggrepel)
pheatmap(mypd,
         cluster_cols = FALSE,
         cluster_rows = TRUE,
         color = c("#3E78AF", "grey90", "#599D3E"),
         gaps_col = c(4),
         fontsize = 5,
         angle_col = 45,
         border_color = NA
         ) 

```

Plot for human appendix and mice

```{r}
mypd <- 
  pd[, c(
      "Appendix PD/Control",
      "Cecal patch",
      "A30P_DSS - A30P_Water",
      "A30P_DSS - WT_Water",
      "WT_DSS - WT_Water",
      "A30P_Water - WT_Water"
)]
# Remove columns that are entirely made of NS
keep <- apply(mypd, 1, function(x) any(x != "NS"))
mypd <- mypd[keep, ]
mypd[mypd == "NS"] <- 0
mypd[mypd == "Hypo-"] <- -1
mypd[mypd == "Hyper-"] <- 1
n <- rownames(mypd)
mypd <- apply(mypd, 2, as.numeric)
rownames(mypd) <- n
rownames(mypd)[ !rownames(mypd) %in% geneList] <- ""
rownames(mypd)[ rownames(mypd) == "BAG3"] <- "              BAG3"


require(ggrepel)
pheatmap(mypd,
         cluster_cols = FALSE,
         cluster_rows = TRUE,
         color = c("#3E78AF", "grey90", "#599D3E"),
         fontsize = 5,
         angle_col = 45,
         border_color = NA
         ) 

```

Export everything into a csv file

```{r}
pd %>% 
  data.table(keep.rownames = TRUE) %>% 
  setnames("rn", "Gene") %>% 
  fwrite(
    quote = FALSE,
    file="allResultsHeatmap.csv")
```


# Other additional files

```{r}
studies <- list.files(path = "../", pattern = "*Padlock*|*RNAseq*")

res <-
foreach (study = studies) %do% {
	message("Loading study: ", study)
	path <- file.path("..", study, "www")
	name <- paste0("m6_methylation.RDS")
	if (file.exists(file.path(path, name))) {
		# padlock study
		x <- readRDS(file.path(path, name))
		x$key[, Study := study]
	} else {
		name <- name <- paste0("r1_rnaseq.RDS")
		if (file.exists(file.path(path, name))) {
			# rnaseq study
			x <- readRDS(file.path(path, name))
			x$key[, Study := study]
		} else {
			message(study, ": no file?")
		}
	}
}
names(res) <- studies


file <- "www/sampleKey.xlsx"
unlink(file)
for (study in studies) {
	write.xlsx(res[[study]], file, sheetName=study, 
	  col.names=TRUE, row.names=FALSE, append=TRUE)
}

url <- "./www/sampleKey.xlsx"
```

- **[Additional File 17](`r url`) Demographic and clinical information for human samples.**

<!--  -->

```{r}
# Data primers
if (!dir.exists("www")) 
	dir.create('www')

file.copy("../etc/Supplementary Data_primers.xlsx", "www/")
```


```{r}
url <- "./www/Supplementary Data_primers.xlsx"
stopifnot(file.exists(url))
```

- **[Additional File 18](`r url`) Padlock probe libraries and study primers.**














