<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Results Appendix</title>

<link href='../libs/fluidbox/css/fluidbox.css' rel='stylesheet' type='text/css'>
<link href='https://fonts.googleapis.com/css?family=Source+Code+Pro' rel='stylesheet' type='text/css'>
<link href='../include/css/style.css' rel='stylesheet' type='text/css'>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#section-TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#section-TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="section-TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <a class="navbar-brand" href="../IndexPage/">
        <img src="../include/gene.png" height="28" width="28">
      </a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="../IndexPage/">Home</a></li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Data processing<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Appendix_PDvsControls_Padlock/>Appendix PD vs Controls (bisulfite)</a></li><li><a href=../Appendix_PDvsControls_RNAseq/>Appendix PD vs Controls (RNA-seq)</a></li><li><a href=../Brain_OFB_Padlock_CGonly/>Olfactory bulb PD vs Controls (bisulfite)</a></li><li><a href=../Brain_PFCRep_Padlock_withGLU/>Prefrontal cortex PD vs Controls (bisulfite)</a></li><li><a href=../Brain_PFCRep_Padlock_withGLU_Braak/>Prefrontal cortex PD vs Controls stagewise (bisulfite)</a></li><li><a href=../Brain_PFC_Padlock_CGonly/>Prefrontal cortex replicate PD vs Controls (bisulfite)</a></li><li><a href=../Appendix_AgeAcceleration_Padlock/>Appendix Aging (bisulfite)</a></li><li><a href=../Appendix_AgeAcceleration_RNAseq/>Appendix Aging (RNA-seq)</a></li><li><a href=../Brain_AgeAcceleration_Padlock/>Prefrontal cortex aging (bisulfite)</a></li><li><a href=../Mice_CecalPatch_Padlock/>Mice cecal catch (bisulfite)</a></li><li><a href=../Mice_DSS_Padlock/>Mice DSS (bisulfite)</a></li>
          </ul>
        </li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Results<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Results_Appendix/>Appendix</a></li><li><a href=../Results_Brain/>Brain</a></li><li><a href=../Results_Proteomics/>Proteomics</a></li><li><a href=../Results_Aging/>Aging</a></li><li><a href=../Results_Mice/>Mice</a></li><li><a href=../Results_etc/>etc</a></li>
          </ul>
        </li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Figures<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Figure1/>Figure 1</a></li><li><a href=../Figure2/>Figure 2</a></li><li><a href=../Figure3/>Figure 3</a></li><li><a href=../Figure4/>Figure 4</a></li><li><a href=../Figure5/>Figure 5</a></li>
          </ul>
        </li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="section-header">



<h1 class="title toc-ignore">Results Appendix</h1>

</div>


<!-- # (c)  Juozas Gordevičius -->
<div id="section-appendix-pd-vs-controls-padlocks" class="section level1">
<h1>Appendix PD vs Controls, Padlocks</h1>
<pre><code>## Number of loci 182024</code></pre>
<pre><code>## Cases and controls:</code></pre>
<pre><code>## Diagnosis
## Control  PD/LBD 
##      19      24</code></pre>
<pre><code>## Number of significant loci 928</code></pre>
<pre><code>## Number of genes affected 326</code></pre>
<p>Direction enrichment</p>
<pre><code>##        
##            -1     1
##   FALSE 87943 93124
##   TRUE    376   552</code></pre>
<pre><code>## # A tibble: 1 x 6
##   estimate    p.value conf.low conf.high method                      alternative
##      &lt;dbl&gt;      &lt;dbl&gt;    &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;                       &lt;chr&gt;      
## 1     1.39 0.00000105     1.21      1.59 Fisher&#39;s Exact Test for Co… two.sided</code></pre>
<ul>
<li><a href="../Appendix_PDvsControls_Padlock/www/m6_Appendix_PDvsControls_Padlock.csv">Additional File 2</a> Robust linear regression fits on appendix PD/control DNA methylation data</li>
</ul>
<div id="section-enrichment-near-lysosomal-gene-starts" class="section level2">
<h2>Enrichment near lysosomal gene starts</h2>
<p>For each CG site we measure distance from padlock target gene start site. We test the enrichment of DMCs among CGs that are within distance <span class="math inline">\(x\)</span> from a target gene.</p>
<ul>
<li><p>The target genes were stratified into ALP, PD GWAS implicated genes and Imprinted control genes</p></li>
<li><p>DMCs with FDR q &lt; 0.05 were considered for ALP and GWAS genes but nominal p value &lt; 0.05 was used to define DMC of imprinted genes</p></li>
<li><p>Fisher test was performed comparing the number of DMCs (yes/no) falling within certain distance of targeted gene (yes/no)</p></li>
</ul>
<p><img src="index_files/figure-html/unnamed-chunk-8-1.png" width="100%" /></p>
<pre><code>##     estimate      p.value  conf.low conf.high
## 1: 2.1165879 3.665610e-08 1.6361292 2.7064676
## 2: 0.5803111 1.093513e-02 0.3558832 0.8974586
## 3: 1.6009940 7.760824e-03 1.1216186 2.2306684
## 4: 1.9065444 2.023606e-09 1.5537669 2.3214737
## 5: 0.6821717 2.051770e-02 0.4791228 0.9446170
##                                method alternative Bracket
## 1: Fisher&#39;s Exact Test for Count Data   two.sided       0
## 2: Fisher&#39;s Exact Test for Count Data   two.sided       5
## 3: Fisher&#39;s Exact Test for Count Data   two.sided       0
## 4: Fisher&#39;s Exact Test for Count Data   two.sided       0
## 5: Fisher&#39;s Exact Test for Count Data   two.sided       5
##                      Target            Q
## 1:           Lysosome genes 5.498414e-07
## 2:           Lysosome genes 8.201345e-02
## 3: Other targeted ALP genes 7.760824e-02
## 4:                      All 6.070819e-08
## 5:                      All 1.231062e-01</code></pre>
</div>
<div id="section-genomic-enrichment" class="section level2">
<h2>Genomic enrichment</h2>
<pre><code>##     Expected, % Obseverd, %        OR        Lo        Hi            P P &lt; 0.05
##  1:  0.58119179  0.57650862 0.9808703 0.8592082 1.1205660 7.896205e-01         
##  2:  0.64307811  0.42564655 0.4093252 0.3583334 0.4672773 1.773217e-41      ***
##  3:  0.09078821  0.05495690 0.5810935 0.4290376 0.7713622 5.817118e-05      ***
##  4:  0.22464353  0.32435345 1.6618031 1.4429236 1.9105317 2.970903e-12      ***
##  5:  0.04233633  0.20043103 5.7856458 4.8920968 6.8176576 3.198655e-70      ***
##  6:  0.02511607  0.09051724 3.9167461 3.0855907 4.9162060 1.956929e-23      ***
##  7:  0.15933405  0.20581897 1.3697844 1.1614894 1.6090920 1.840789e-04      ***
##  8:  0.05352894  0.03879310 0.7125332 0.4955073 0.9948064 4.789977e-02        *
##  9:  0.22692931  0.14008621 0.5535825 0.4561456 0.6673310 3.270535e-11      ***
## 10:  0.14586115  0.19827586 1.4513385 1.2276046 1.7086580 1.371840e-05      ***
## 11:  0.07596912  0.22413793 3.5521664 3.0264545 4.1549804 1.710097e-45      ***
##                  Category Type
##  1:    hg19_genes_introns  All
##  2:        hg19_cpg_inter  All
##  3:      hg19_cpg_shelves  All
##  4:       hg19_cpg_shores  All
##  5:      hg19_cpg_islands  All
##  6:      hg19_genes_5UTRs  All
##  7:      hg19_genes_exons  All
##  8:      hg19_genes_3UTRs  All
##  9: hg19_genes_intergenic  All
## 10:     hg19_genes_1to5kb  All
## 11:  hg19_genes_promoters  All</code></pre>
<pre><code>##    Expected, % Obseverd, %        OR        Lo         Hi             P
## 1:  0.06178192  0.24030172 4.8799913 4.1740179  5.6887812  1.242559e-69
## 2:  0.06178192  0.38768116 9.7821017 8.1942692 11.6531103 7.446640e-111
## 3:  0.06178192  0.02393617 0.3719065 0.1686697  0.7140582  1.156806e-03
## 4:  0.10495343  0.07866379 0.7270894 0.5641960  0.9246258  8.402415e-03
## 5:  0.10495343  0.07065217 0.6476122 0.4549934  0.8979160  6.638033e-03
## 6:  0.10495343  0.09042553 0.8475247 0.5772500  1.2079775  3.997174e-01
## 7:  0.17272453  0.22413793 1.3862033 1.1816744  1.6204897  5.977400e-05
## 8:  0.17272453  0.31521739 2.2114258 1.8363643  2.6544362  2.730163e-16
## 9:  0.17272453  0.09042553 0.4755940 0.3239653  0.6777756  7.848752e-06
##    P &lt; 0.05       Category   Type
## 1:      ***       Promoter    All
## 2:      ***       Promoter Hyper-
## 3:       **       Promoter  Hypo-
## 4:       ** PoisedEnhancer    All
## 5:       ** PoisedEnhancer Hyper-
## 6:          PoisedEnhancer  Hypo-
## 7:      *** ActiveEnhancer    All
## 8:      *** ActiveEnhancer Hyper-
## 9:      *** ActiveEnhancer  Hypo-</code></pre>
<div class="figure">
<img src="index_files/figure-html/unnamed-chunk-12-1.png" alt="Enrichment of genomic elements in apx pd vs controls" width="100%" />
<p class="caption">
Enrichment of genomic elements in apx pd vs controls
</p>
</div>
</div>
<div id="section-enrichment-of-alp-pathways-with-dmcs" class="section level2">
<h2>Enrichment of ALP pathways with DMCs</h2>
<pre><code>##      estimate      p.value  conf.low conf.high
##  1: 0.9573666 0.9207028622 0.6156104  1.424297
##  2: 0.7026202 0.3007716576 0.3484172  1.268370
##  3: 1.3385645 0.2718588807 0.7239546  2.277442
##  4: 0.9890237 0.9331831932 0.8338224  1.167865
##  5: 0.8286218 0.1150447767 0.6534123  1.040933
##  6: 1.2437198 0.0869078585 0.9646624  1.589275
##  7: 1.2474292 0.0009557178 1.0919620  1.426230
##  8: 1.3332721 0.0009795409 1.1201590  1.589722
##  9: 1.1315301 0.2546174665 0.9180836  1.396792
## 10: 1.0164594 0.8875260281 0.5383671  1.750191
## 11: 1.1863592 0.5800249195 0.5389238  2.272600
## 12: 0.7688460 0.8237682844 0.2082274  1.987499
## 13: 1.0339789 0.6516878184 0.8895633  1.198707
## 14: 0.8964805 0.2839678986 0.7309250  1.093487
## 15: 1.2546253 0.0448162125 0.9976683  1.569468
##                                 method alternative       Term
##  1: Fisher&#39;s Exact Test for Count Data   two.sided GO:0061912
##  2: Fisher&#39;s Exact Test for Count Data   two.sided GO:0061912
##  3: Fisher&#39;s Exact Test for Count Data   two.sided GO:0061912
##  4: Fisher&#39;s Exact Test for Count Data   two.sided GO:0016236
##  5: Fisher&#39;s Exact Test for Count Data   two.sided GO:0016236
##  6: Fisher&#39;s Exact Test for Count Data   two.sided GO:0016236
##  7: Fisher&#39;s Exact Test for Count Data   two.sided GO:0005764
##  8: Fisher&#39;s Exact Test for Count Data   two.sided GO:0005764
##  9: Fisher&#39;s Exact Test for Count Data   two.sided GO:0005764
## 10: Fisher&#39;s Exact Test for Count Data   two.sided GO:0061684
## 11: Fisher&#39;s Exact Test for Count Data   two.sided GO:0061684
## 12: Fisher&#39;s Exact Test for Count Data   two.sided GO:0061684
## 13: Fisher&#39;s Exact Test for Count Data   two.sided GO:0006914
## 14: Fisher&#39;s Exact Test for Count Data   two.sided GO:0006914
## 15: Fisher&#39;s Exact Test for Count Data   two.sided GO:0006914
##                             Name   Type    Dir
##  1:          Selective autophagy Target    All
##  2:          Selective autophagy Target Hyper-
##  3:          Selective autophagy Target  Hypo-
##  4:               Macroautophagy Target    All
##  5:               Macroautophagy Target Hyper-
##  6:               Macroautophagy Target  Hypo-
##  7:                    Lysosomes Target    All
##  8:                    Lysosomes Target Hyper-
##  9:                    Lysosomes Target  Hypo-
## 10: Chaperone-mediated autophagy Target    All
## 11: Chaperone-mediated autophagy Target Hyper-
## 12: Chaperone-mediated autophagy Target  Hypo-
## 13:                    Autophagy Target    All
## 14:                    Autophagy Target Hyper-
## 15:                    Autophagy Target  Hypo-</code></pre>
<div class="figure">
<img src="index_files/figure-html/unnamed-chunk-15-1.png" alt="Enrichment of ALP pathways in apx pd vs controls" width="100%" />
<p class="caption">
Enrichment of ALP pathways in apx pd vs controls
</p>
</div>
</div>
</div>
<div id="section-appendix-pd-vs-controls-rnaseq" class="section level1">
<h1>Appendix PD vs Controls, RNAseq</h1>
<pre><code>## Number of loci 14336</code></pre>
<pre><code>## Cases and controls:</code></pre>
<pre><code>## Diagnosis
## Control  PD/LBD 
##      16      12</code></pre>
<pre><code>## Number of genes affected 246</code></pre>
<p>The ALP genes tend to be down-regulated</p>
<pre><code>## # A tibble: 1 x 6
##   estimate p.value conf.low conf.high method                         alternative
##      &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;                          &lt;chr&gt;      
## 1     2.95 0.00231     1.42      5.51 Fisher&#39;s Exact Test for Count… two.sided</code></pre>
<ul>
<li><a href="../Appendix_PDvsControls_RNAseq/www/r1_Appendix_PDvsControls_RNAseq.csv">Additional File 3</a> Appendix PD vs Controls RNA-seq fits</li>
</ul>
<div id="section-enrichment-of-alp-pathways-with-degs" class="section level2">
<h2>Enrichment of ALP pathways with DEGs</h2>
<pre><code>##      estimate     p.value    conf.low conf.high
##  1: 0.2848690 0.080679607 0.033372196 1.0965730
##  2: 0.3081378 0.368730846 0.007621483 1.8150409
##  3: 0.2951236 0.374055200 0.007300026 1.7382019
##  4: 0.9416623 0.790929641 0.641319232 1.3472913
##  5: 1.3929568 0.114900362 0.879114644 2.1197582
##  6: 0.5460653 0.054887335 0.268267354 0.9980949
##  7: 0.9727938 0.900573679 0.749552545 1.2483315
##  8: 0.5497994 0.004529585 0.339886314 0.8464226
##  9: 1.4234856 0.018528123 1.046359291 1.9038069
## 10: 0.0000000 0.150377146 0.000000000 1.6714127
## 11: 0.0000000 0.619674066 0.000000000 3.7045653
## 12: 0.0000000 0.620857640 0.000000000 3.5479623
## 13: 0.9315208 0.677397132 0.692787061 1.2338169
## 14: 1.2338310 0.217137881 0.850756088 1.7428100
## 15: 0.6656126 0.075169380 0.410627153 1.0272375
##                                 method alternative       Term
##  1: Fisher&#39;s Exact Test for Count Data   two.sided GO:0061912
##  2: Fisher&#39;s Exact Test for Count Data   two.sided GO:0061912
##  3: Fisher&#39;s Exact Test for Count Data   two.sided GO:0061912
##  4: Fisher&#39;s Exact Test for Count Data   two.sided GO:0016236
##  5: Fisher&#39;s Exact Test for Count Data   two.sided GO:0016236
##  6: Fisher&#39;s Exact Test for Count Data   two.sided GO:0016236
##  7: Fisher&#39;s Exact Test for Count Data   two.sided GO:0005764
##  8: Fisher&#39;s Exact Test for Count Data   two.sided GO:0005764
##  9: Fisher&#39;s Exact Test for Count Data   two.sided GO:0005764
## 10: Fisher&#39;s Exact Test for Count Data   two.sided GO:0061684
## 11: Fisher&#39;s Exact Test for Count Data   two.sided GO:0061684
## 12: Fisher&#39;s Exact Test for Count Data   two.sided GO:0061684
## 13: Fisher&#39;s Exact Test for Count Data   two.sided GO:0006914
## 14: Fisher&#39;s Exact Test for Count Data   two.sided GO:0006914
## 15: Fisher&#39;s Exact Test for Count Data   two.sided GO:0006914
##                             Name Type    Dir
##  1:          Selective autophagy  RNA    All
##  2:          Selective autophagy  RNA Hyper-
##  3:          Selective autophagy  RNA  Hypo-
##  4:               Macroautophagy  RNA    All
##  5:               Macroautophagy  RNA Hyper-
##  6:               Macroautophagy  RNA  Hypo-
##  7:                    Lysosomes  RNA    All
##  8:                    Lysosomes  RNA Hyper-
##  9:                    Lysosomes  RNA  Hypo-
## 10: Chaperone-mediated autophagy  RNA    All
## 11: Chaperone-mediated autophagy  RNA Hyper-
## 12: Chaperone-mediated autophagy  RNA  Hypo-
## 13:                    Autophagy  RNA    All
## 14:                    Autophagy  RNA Hyper-
## 15:                    Autophagy  RNA  Hypo-</code></pre>
</div>
<div id="section-correlation-of-fold-changes" class="section level2">
<h2>Correlation of fold changes</h2>
<pre><code>##   correlation   std.err  t.value      p.value
## Y   0.2084524 0.0477231 4.367956 1.581823e-05</code></pre>
<!--html_preserve-->
<script type="application/shiny-prerendered" data-context="dependencies">
{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["1.11.3"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jquery"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.18"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["bootstrap"]},{"type":"character","attributes":{},"value":["3.3.5"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/bootstrap"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["viewport"]}},"value":[{"type":"character","attributes":{},"value":["width=device-width, initial-scale=1"]}]},{"type":"character","attributes":{},"value":["js/bootstrap.min.js","shim/html5shiv.min.js","shim/respond.min.js"]},{"type":"character","attributes":{},"value":["css/bootstrap.min.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.18"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["1.11.3"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jquery"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.18"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jqueryui"]},{"type":"character","attributes":{},"value":["1.11.4"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jqueryui"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery-ui.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.18"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tocify"]},{"type":"character","attributes":{},"value":["1.9.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/tocify"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.tocify.js"]},{"type":"character","attributes":{},"value":["jquery.tocify.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.18"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["1.11.3"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jquery"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.18"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["navigation"]},{"type":"character","attributes":{},"value":["1.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/navigation-1.1"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tabsets.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.18"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["highlightjs"]},{"type":"character","attributes":{},"value":["9.12.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/highlightjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["highlight.js"]},{"type":"character","attributes":{},"value":["default.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.18"]}]}]}
</script>
<!--/html_preserve-->
<!--html_preserve-->
<script type="application/shiny-prerendered" data-context="execution_dependencies">
{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["packages"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["packages","version"]},"class":{"type":"character","attributes":{},"value":["data.frame"]},"row.names":{"type":"integer","attributes":{},"value":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183]}},"value":[{"type":"character","attributes":{},"value":["acepack","AnnotationDbi","AnnotationHub","annotatr","askpass","assertthat","backports","base","base64enc","Biobase","BiocFileCache","BiocGenerics","BiocManager","BiocParallel","BiocVersion","biomaRt","Biostrings","bit","bit64","bitops","blob","boot","broom","BSgenome","checkmate","cli","cluster","codetools","colorspace","compiler","cowplot","crayon","curl","data.table","datasets","DBI","dbplyr","DelayedArray","DEoptimR","digest","doParallel","doSNOW","dplyr","dynamicTreeCut","evaluate","fansi","farver","fastcluster","fastmap","fit.models","foreach","foreign","Formula","gdata","generics","GenomeInfoDb","GenomeInfoDbData","GenomicAlignments","GenomicFeatures","GenomicRanges","GetoptLong","ggplot2","ggpubr","ggsignif","GlobalOptions","glue","GO.db","graphics","grDevices","grid","gridExtra","gtable","gtools","highr","Hmisc","hms","htmlTable","htmltools","htmlwidgets","httpuv","httr","impute","interactiveDisplayBase","IRanges","iterators","itertools","jomo","jpeg","jsonlite","kableExtra","knitr","labeling","later","latex2exp","lattice","latticeExtra","lazyeval","lifecycle","limma","lme4","magrittr","MASS","Matrix","matrixStats","memoise","methods","mice","mime","minqa","mitml","munsell","mvtnorm","nlme","nloptr","nnet","openssl","org.Hs.eg.db","pan","parallel","pcaPP","pillar","pkgconfig","plotly","plyr","png","preprocessCore","prettyunits","progress","promises","purrr","R6","rappdirs","RColorBrewer","Rcpp","RCurl","readr","regioneR","reshape2","rGREAT","rjson","rlang","rmarkdown","robust","robustbase","rpart","rrcov","Rsamtools","RSQLite","rstudioapi","rtracklayer","rvest","S4Vectors","scales","shiny","snow","splines","stats","stats4","stringi","stringr","SummarizedExperiment","survival","tibble","tidyr","tidyselect","tools","TxDb.Hsapiens.UCSC.hg19.knownGene","utf8","utils","vctrs","viridisLite","webshot","weights","WGCNA","withr","xfun","XML","xml2","xtable","XVector","yaml","zeallot","zlibbioc"]},{"type":"character","attributes":{},"value":["1.4.1","1.48.0","2.18.0","1.12.0","1.1","0.2.1","1.1.5","3.6.1","0.1-3","2.46.0","1.10.0","0.32.0","1.30.10","1.20.0","3.10.1","2.42.0","2.54.0","1.1-15","0.9-7","1.0-6","1.2.0","1.3-24","0.5.3","1.54.0","1.9.4","2.0.1","2.1.0","0.2-16","1.4-1","3.6.1","1.0.0","1.3.4","4.3","1.12.8","3.6.1","1.1.0","1.4.2","0.12.0","1.0-8","0.6.23","1.0.15","1.0.18","0.8.3","1.63-1","0.14","0.4.1","2.0.2","1.1.25","1.0.1","0.5-14","1.4.7","0.8-74","1.2-3","2.18.0","0.0.2","1.22.0","1.2.2","1.22.0","1.38.0","1.38.0","0.1.8","3.2.1","0.2.4","0.6.0","0.1.1","1.3.1","3.10.0","3.6.1","3.6.1","3.6.1","2.3","0.3.0","3.8.1","0.8","4.3-0","0.5.3","1.13.3","0.4.0","1.5.1","1.5.2","1.4.1","1.60.0","1.24.0","2.20.0","1.0.12","0.1-3","2.6-10","0.1-8.1","1.6","1.1.0","1.26","0.3","1.0.0","0.4.0","0.20-38","0.6-29","0.2.2","0.1.0","3.42.0","1.1-21","1.5","7.3-51.5","1.2-18","0.55.0","1.1.0","3.6.1","3.7.0","0.8","1.2.4","0.3-7","0.5.0","1.0-12","3.1-143","1.2.1","7.3-12","1.4.1","3.10.0","1.6","3.6.1","1.9-73","1.4.3","2.0.3","4.9.1","1.8.5","0.1-7","1.48.0","1.1.0","1.2.2","1.1.0","0.3.3","2.4.1","0.3.1","1.1-2","1.0.3","1.95-4.12","1.3.1","1.18.0","1.4.3","1.18.0","0.2.20","0.4.2","1.18","0.4-18.2","0.93-5","4.1-15","1.4-9","2.2.0","2.1.5","0.10","1.46.0","0.3.5","0.24.0","1.1.0","1.4.0","0.4-3","3.6.1","3.6.1","3.6.1","1.4.5","1.4.0","1.16.0","3.1-8","2.1.3","1.0.0","0.2.5","3.6.1","3.2.2","1.1.4","3.6.1","0.2.1","0.3.0","0.5.2","1.0","1.68","2.1.2","0.12","3.98-1.20","1.2.2","1.8-4","0.26.0","2.2.0","0.1.0","1.32.0"]}]}]}
</script>
<!--/html_preserve-->
</div>
</div>

<script type="text/javascript" src="../libs/fluidbox/jquery.fluidbox.min.js"></script>
<script type="text/javascript" src="../include/js/enable-fluidbox.js"></script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("section-TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#section-TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
