<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Appendix Age Acceleration, Padlocks</title>

<link href='../libs/fluidbox/css/fluidbox.css' rel='stylesheet' type='text/css'>
<link href='https://fonts.googleapis.com/css?family=Source+Code+Pro' rel='stylesheet' type='text/css'>
<link href='../include/css/style.css' rel='stylesheet' type='text/css'>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#section-TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#section-TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="section-TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <a class="navbar-brand" href="../IndexPage/">
        <img src="../include/gene.png" height="28" width="28">
      </a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="../IndexPage/">Home</a></li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Data processing<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Appendix_PDvsControls_Padlock/>Appendix PD vs Controls (bisulfite)</a></li><li><a href=../Appendix_PDvsControls_RNAseq/>Appendix PD vs Controls (RNA-seq)</a></li><li><a href=../Brain_OFB_Padlock_CGonly/>Olfactory bulb PD vs Controls (bisulfite)</a></li><li><a href=../Brain_PFCRep_Padlock_withGLU/>Prefrontal cortex PD vs Controls (bisulfite)</a></li><li><a href=../Brain_PFCRep_Padlock_withGLU_Braak/>Prefrontal cortex PD vs Controls stagewise (bisulfite)</a></li><li><a href=../Brain_PFC_Padlock_CGonly/>Prefrontal cortex replicate PD vs Controls (bisulfite)</a></li><li><a href=../Appendix_AgeAcceleration_Padlock/>Appendix Aging (bisulfite)</a></li><li><a href=../Appendix_AgeAcceleration_RNAseq/>Appendix Aging (RNA-seq)</a></li><li><a href=../Brain_AgeAcceleration_Padlock/>Prefrontal cortex aging (bisulfite)</a></li><li><a href=../Brain_AgeAcceleration_OFB_Padlock_CGonly/>Olfactory bulb aging (bisulfite)</a></li><li><a href=../Mice_CecalPatch_Padlock/>Mice cecal catch (bisulfite)</a></li><li><a href=../Mice_DSS_Padlock/>Mice DSS (bisulfite)</a></li>
          </ul>
        </li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Results<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Results_Appendix/>Appendix</a></li><li><a href=../Results_Brain/>Brain</a></li><li><a href=../Results_Proteomics/>Proteomics</a></li><li><a href=../Results_Aging/>Aging</a></li><li><a href=../Results_Mice/>Mice</a></li><li><a href=../Results_etc/>etc</a></li>
          </ul>
        </li><li class="dropdown">
  <a href="authoring" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Figures<span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li><a href=../Figure1/>Figure 1</a></li><li><a href=../Figure2/>Figure 2</a></li><li><a href=../Figure3/>Figure 3</a></li><li><a href=../Figure4/>Figure 4</a></li><li><a href=../Figure5/>Figure 5</a></li>
          </ul>
        </li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="section-header">



<h1 class="title toc-ignore">Appendix Age Acceleration, Padlocks</h1>

</div>


<!-- # (c)  Juozas Gordevičius -->
<div id="section-parameters" class="section level1">
<h1>Parameters</h1>
<pre><code>## {
##   &quot;host&quot;: &quot;node010&quot;,
##   &quot;nPerm&quot;: 40,
##   &quot;workers&quot;: 41,
##   &quot;outliers&quot;: {
##     &quot;thresholdSD&quot;: 3,
##     &quot;noOfPCs&quot;: 2,
##     &quot;removeOutliers&quot;: true,
##     &quot;shape&quot;: &quot;Diagnosis&quot;,
##     &quot;perCohort&quot;: true
##   },
##   &quot;clustering&quot;: {
##     &quot;groups&quot;: [&quot;Sex&quot;, &quot;Batch&quot;, &quot;substr(ID, 1, 1)&quot;],
##     &quot;addClustersToModel&quot;: 0
##   },
##   &quot;normalization&quot;: {
##     &quot;normalize&quot;: true,
##     &quot;plotColor&quot;: &quot;Batch&quot;
##   },
##   &quot;imputation&quot;: {
##     &quot;imputeBetas&quot;: false,
##     &quot;imputeCounts&quot;: false
##   },
##   &quot;model&quot;: {
##     &quot;formula&quot;: &quot;~ 0 + Diagnosis + Diagnosis:Age + Cohort + Sex + PMI&quot;,
##     &quot;formula0&quot;: &quot;~ 0 + Diagnosis + Age + Cohort + Sex + PMI&quot;,
##     &quot;contrasts&quot;: [&quot;DiagnosisControl.Age&quot;, &quot;DiagnosisPD.LBD.Age&quot;, &quot;DiagnosisControl.Age - DiagnosisPD.LBD.Age&quot;],
##     &quot;svRange&quot;: [1, 2, 3, 4, 5, 6]
##   }
## }</code></pre>
</div>
<div id="section-data" class="section level1">
<h1>Data</h1>
<p>Combined together all appendix samples</p>
<p>Resulting data structure:</p>
<pre><code>##        beta counts countsU countsM genome sample modelData
## [1,] 185400 185400  185400  185400 185400     77        77
## [2,]     77     77      77      77      6      7         5</code></pre>
<!-- Run the processing pipeline -->
</div>
<div id="section-analysis" class="section level1">
<h1>Analysis</h1>
<div id="section-outlier-identification" class="section level2">
<h2>Outlier identification</h2>
<p><img src="index_files/figure-html/outliers_plotPCAPairs-1.png" width="100%" /></p>
<table class="table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Diagnosis
</th>
<th style="text-align:right;">
Age
</th>
<th style="text-align:left;">
Sex
</th>
<th style="text-align:right;">
PMI
</th>
<th style="text-align:left;">
Batch
</th>
<th style="text-align:left;">
Cohort
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
E1
</td>
<td style="text-align:left;">
Control
</td>
<td style="text-align:right;">
77
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:right;">
0.45
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
Aging
</td>
</tr>
<tr>
<td style="text-align:left;">
X3616
</td>
<td style="text-align:left;">
PD/LBD
</td>
<td style="text-align:right;">
87
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:right;">
18.00
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
PD
</td>
</tr>
</tbody>
</table>
<blockquote>
<p>Outliers have been removed.</p>
</blockquote>
<p>[1] “untouched tables: genome”</p>
<!-- 
  @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
  Clustering before normalization 
-->
</div>
<div id="section-sample-clustering" class="section level2">
<h2>Sample clustering</h2>
<p>Hierarchical clustering using correlations as similarity measure and Ward linkage.</p>
<p><img src="index_files/figure-html/unnamed-chunk-10-1.png" width="100%" /></p>
<!-- 
  @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
  Normalization
-->
</div>
<div id="section-normalization" class="section level2">
<h2>Normalization</h2>
<p>Signal density before normalization</p>
<p><img src="index_files/figure-html/unnamed-chunk-13-1.png" width="100%" /></p>
<!-- After normalization -->
<p>NU and M matrices were quantile normalized separately, total counts and beta values were obtained from normalized matrices.</p>
<p>Signals after normalization</p>
<p><img src="index_files/figure-html/unnamed-chunk-14-1.png" width="100%" />PCA and outliers after normalization</p>
<p><img src="index_files/figure-html/unnamed-chunk-14-2.png" width="100%" /></p>
</div>
<div id="section-covariates" class="section level2">
<h2>Covariates</h2>
<p>Covariates used in subsequent models</p>
<pre><code>## [1] &quot;Age&quot;       &quot;Sex&quot;       &quot;PMI&quot;       &quot;Diagnosis&quot; &quot;Cohort&quot;</code></pre>
<p>Are any of the covariates related to the variable of interest? (Assuming that the first variable in modelData is the one of interest).</p>
<pre><code>## 
## Call:
## glm(formula = formula, data = data$modelData)
## 
## Deviance Residuals: 
##     Min       1Q   Median       3Q      Max  
## -38.059   -7.747    0.780    9.109   24.212  
## 
## Coefficients:
##                 Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)     58.81305    3.11547  18.878   &lt;2e-16 ***
## SexF            -2.73216    3.54267  -0.771   0.4432    
## PMI             -0.02973    0.20635  -0.144   0.8859    
## DiagnosisPD/LBD  7.97708    4.77356   1.671   0.0992 .  
## Cohort1         13.79746    5.51749   2.501   0.0148 *  
## Cohort2         15.19058    6.96096   2.182   0.0325 *  
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for gaussian family taken to be 221.8034)
## 
##     Null deviance: 22419  on 74  degrees of freedom
## Residual deviance: 15304  on 69  degrees of freedom
## AIC: 625.72
## 
## Number of Fisher Scoring iterations: 2</code></pre>
</div>
</div>
<div id="section-robust-lmfit-v2" class="section level1">
<h1>Robust lmFit v2</h1>
<p>50% stable, non extreme change, M values, robust limma eBayes</p>
<pre class="r"><code>methylation &lt;- data$beta
prefix &lt;- &quot;m6_&quot;
modelparams &lt;- list(
    allowStable = 0.5,
    useMvalues  = TRUE,
    fixInfinity = 0.001, # Fraction of change for 0% and 100% methylated locations
    downloadDataMatrix = TRUE,
    accelerationVariable = &quot;Diagnosis&quot; # Name of variable to test differential acceleration
  )</code></pre>
<div id="section-removal-of-non-informative-loci" class="section level2">
<h2>Removal of non-informative loci</h2>
<p>Fraction of CGs that should be removed depending on threshold. Currently set threshold is 50%</p>
<blockquote>
<p>Removing 4249 stable loci.</p>
</blockquote>
</div>
<div id="section-conversion-of-beta-values-to-m-values" class="section level2">
<h2>Conversion of Beta values to M values</h2>
<blockquote>
<p>Beta values were converted to M values.</p>
</blockquote>
<p><img src="index_files/figure-html/unnamed-chunk-21-1.png" width="100%" /></p>
</div>
<div id="section-data-download" class="section level2">
<h2>Data download</h2>
<blockquote>
<p>Download <a href="www/m6_methylation.RDS">data</a> file.</p>
</blockquote>
</div>
<div id="section-data-pcs-vs-covariates" class="section level2">
<h2>Data PCs vs Covariates</h2>
<p>Fit covariates to the PCs using limma</p>
<pre><code>## Removing intercept from test coefficients</code></pre>
<table class="table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Age
</th>
<th style="text-align:right;">
SexF
</th>
<th style="text-align:right;">
PMI
</th>
<th style="text-align:right;">
DiagnosisPD.LBD
</th>
<th style="text-align:right;">
Cohort1
</th>
<th style="text-align:right;">
Cohort2
</th>
<th style="text-align:right;">
AveExpr
</th>
<th style="text-align:right;">
F
</th>
<th style="text-align:right;">
P.Value
</th>
<th style="text-align:right;">
adj.P.Val
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
PC2
</td>
<td style="text-align:right;">
-0.1039909
</td>
<td style="text-align:right;">
-8.4068769
</td>
<td style="text-align:right;">
0.8573093
</td>
<td style="text-align:right;">
-49.622599
</td>
<td style="text-align:right;">
1.881541
</td>
<td style="text-align:right;">
-180.803231
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
28.7962534
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
PC1
</td>
<td style="text-align:right;">
1.4054156
</td>
<td style="text-align:right;">
-12.5510868
</td>
<td style="text-align:right;">
2.8543855
</td>
<td style="text-align:right;">
41.301081
</td>
<td style="text-align:right;">
67.783059
</td>
<td style="text-align:right;">
-47.599903
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
6.7168283
</td>
<td style="text-align:right;">
0.0000096
</td>
<td style="text-align:right;">
0.0000480
</td>
</tr>
<tr>
<td style="text-align:left;">
PC3
</td>
<td style="text-align:right;">
1.3295081
</td>
<td style="text-align:right;">
-7.6836794
</td>
<td style="text-align:right;">
-1.3073497
</td>
<td style="text-align:right;">
-22.379140
</td>
<td style="text-align:right;">
32.802034
</td>
<td style="text-align:right;">
-10.906930
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.9814404
</td>
<td style="text-align:right;">
0.0785560
</td>
<td style="text-align:right;">
0.2618534
</td>
</tr>
<tr>
<td style="text-align:left;">
PC7
</td>
<td style="text-align:right;">
0.3710669
</td>
<td style="text-align:right;">
29.0310551
</td>
<td style="text-align:right;">
1.0351822
</td>
<td style="text-align:right;">
-7.540886
</td>
<td style="text-align:right;">
-15.772934
</td>
<td style="text-align:right;">
1.979755
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.2540060
</td>
<td style="text-align:right;">
0.2886930
</td>
<td style="text-align:right;">
0.6158646
</td>
</tr>
<tr>
<td style="text-align:left;">
PC9
</td>
<td style="text-align:right;">
-0.5537953
</td>
<td style="text-align:right;">
-15.9334120
</td>
<td style="text-align:right;">
-0.1339604
</td>
<td style="text-align:right;">
1.377232
</td>
<td style="text-align:right;">
16.050137
</td>
<td style="text-align:right;">
51.720257
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.2148259
</td>
<td style="text-align:right;">
0.3079323
</td>
<td style="text-align:right;">
0.6158646
</td>
</tr>
<tr>
<td style="text-align:left;">
PC10
</td>
<td style="text-align:right;">
0.4113075
</td>
<td style="text-align:right;">
-0.7170407
</td>
<td style="text-align:right;">
-0.7626552
</td>
<td style="text-align:right;">
2.720919
</td>
<td style="text-align:right;">
18.004121
</td>
<td style="text-align:right;">
-18.364818
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.8027938
</td>
<td style="text-align:right;">
0.5707886
</td>
<td style="text-align:right;">
0.9385228
</td>
</tr>
<tr>
<td style="text-align:left;">
PC4
</td>
<td style="text-align:right;">
-0.0612320
</td>
<td style="text-align:right;">
-0.4549554
</td>
<td style="text-align:right;">
0.5964035
</td>
<td style="text-align:right;">
1.802803
</td>
<td style="text-align:right;">
-24.061844
</td>
<td style="text-align:right;">
-52.197111
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.6539184
</td>
<td style="text-align:right;">
0.6868292
</td>
<td style="text-align:right;">
0.9385228
</td>
</tr>
<tr>
<td style="text-align:left;">
PC6
</td>
<td style="text-align:right;">
-0.4495926
</td>
<td style="text-align:right;">
11.4854057
</td>
<td style="text-align:right;">
0.5749265
</td>
<td style="text-align:right;">
4.940078
</td>
<td style="text-align:right;">
-12.083381
</td>
<td style="text-align:right;">
12.789843
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.5616686
</td>
<td style="text-align:right;">
0.7594608
</td>
<td style="text-align:right;">
0.9385228
</td>
</tr>
<tr>
<td style="text-align:left;">
PC8
</td>
<td style="text-align:right;">
-0.0199469
</td>
<td style="text-align:right;">
12.8760727
</td>
<td style="text-align:right;">
-0.5316641
</td>
<td style="text-align:right;">
6.844144
</td>
<td style="text-align:right;">
-5.979546
</td>
<td style="text-align:right;">
-13.795932
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.4474110
</td>
<td style="text-align:right;">
0.8446705
</td>
<td style="text-align:right;">
0.9385228
</td>
</tr>
<tr>
<td style="text-align:left;">
PC5
</td>
<td style="text-align:right;">
0.2398807
</td>
<td style="text-align:right;">
9.8571820
</td>
<td style="text-align:right;">
-0.1324579
</td>
<td style="text-align:right;">
0.732610
</td>
<td style="text-align:right;">
-2.656854
</td>
<td style="text-align:right;">
6.694785
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.1555765
</td>
<td style="text-align:right;">
0.9873742
</td>
<td style="text-align:right;">
0.9873742
</td>
</tr>
</tbody>
</table>
<p>The covariates significantly affecting the PCs using ANOVA:</p>
<p><img src="index_files/figure-html/unnamed-chunk-25-1.png" width="100%" /></p>
</div>
<div id="section-select-the-number-of-svs" class="section level2">
<h2>Select the number of SVs</h2>
<p>All known covariates are regressed from data and PCA of residuals is computed. The number of PCs that explain more than 5% of variability is used as the desired number of SVs.</p>
<pre><code>## Selected number of SVs 0</code></pre>
</div>
<div id="section-probe-level-models" class="section level2">
<h2>Probe-level models</h2>
<div id="section-p-value-distribution" class="section level3">
<h3>P value distribution</h3>
<pre><code>## Warning: Removed 30 rows containing non-finite values (stat_bin).</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-30-1.png" width="100%" /></p>
</div>
<div id="section-volcano-plot" class="section level3">
<h3>Volcano plot</h3>
<p><img src="index_files/figure-html/unnamed-chunk-32-1.png" width="100%" /></p>
</div>
<div id="section-manhattan-plot" class="section level3">
<h3>Manhattan plot</h3>
<p><img src="index_files/figure-html/unnamed-chunk-33-1.png" width="100%" /><a href="www/m6_manhattan_DiagnosisControl.Age.pdf">PDF</a> <img src="index_files/figure-html/unnamed-chunk-33-2.png" width="100%" /><a href="www/m6_manhattan_DiagnosisPD.LBD.Age.pdf">PDF</a> <img src="index_files/figure-html/unnamed-chunk-33-3.png" width="100%" /><a href="www/m6_manhattan_DiagnosisControl.Age-DiagnosisPD.LBD.Age.pdf">PDF</a></p>
</div>
<div id="section-enrichment" class="section level3">
<h3>Enrichment</h3>
<pre><code>## NULL</code></pre>
</div>
<div id="section-download" class="section level3">
<h3>Download</h3>
<p><a href="www/m6_Appendix_AgeAcceleration_Padlock.csv">Download results</a></p>
<!-- Final touches to remove lasting effects -->
</div>
<div id="section-age-acceleration" class="section level3">
<h3>Age acceleration</h3>
<p>An age predictor is trained and evaluated using 10 fold cross-validation on all the samples. Random forest is used as training algorithm.</p>
<pre><code>## `geom_smooth()` using formula &#39;y ~ x&#39;
## `geom_smooth()` using formula &#39;y ~ x&#39;</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-40-1.png" width="100%" /></p>
</div>
</div>
</div>
<div id="section-sessioninfo" class="section level1">
<h1>SessionInfo</h1>
<pre><code>## R version 3.6.1 (2019-07-05)
## Platform: x86_64-conda_cos6-linux-gnu (64-bit)
## Running under: CentOS Linux 7 (Core)
## 
## Matrix products: default
## BLAS/LAPACK: /secondary/projects/labrie/juozas/condaenv/ALP/lib/libopenblasp-r0.3.7.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] ggsci_2.9             randomForest_4.6-14   caret_6.0-86         
##  [4] lattice_0.20-41       bigmemory_4.5.36      latex2exp_0.4.0      
##  [7] doSNOW_1.0.18         snow_0.4-3            itertools_0.1-3      
## [10] iterators_1.0.12      limma_3.42.0          impute_1.60.0        
## [13] WGCNA_1.69            fastcluster_1.1.25    dynamicTreeCut_1.63-1
## [16] glue_1.4.1            foreach_1.5.0         ggpubr_0.3.0         
## [19] plotly_4.9.2.1        ggplot2_3.3.1         dplyr_1.0.0          
## [22] data.table_1.12.8     kableExtra_1.1.0      knitr_1.28           
## [25] shiny_1.4.0.2        
## 
## loaded via a namespace (and not attached):
##   [1] readxl_1.3.1          backports_1.1.8       Hmisc_4.4-0          
##   [4] plyr_1.8.6            lazyeval_0.2.2        splines_3.6.1        
##   [7] digest_0.6.25         htmltools_0.5.0       GO.db_3.10.0         
##  [10] magrittr_1.5          checkmate_2.0.0       memoise_1.1.0        
##  [13] cluster_2.1.0         doParallel_1.0.15     openxlsx_4.1.5       
##  [16] recipes_0.1.12        readr_1.3.1           gower_0.2.1          
##  [19] matrixStats_0.56.0    jpeg_0.1-8.1          colorspace_1.4-1     
##  [22] blob_1.2.1            rvest_0.3.5           ggrepel_0.8.2        
##  [25] haven_2.3.1           xfun_0.14             crayon_1.3.4         
##  [28] jsonlite_1.6.1        bigmemory.sri_0.1.3   survival_3.2-3       
##  [31] gtable_0.3.0          ipred_0.9-9           webshot_0.5.2        
##  [34] car_3.0-8             BiocGenerics_0.32.0   abind_1.4-5          
##  [37] scales_1.1.1          DBI_1.1.0             rstatix_0.5.0        
##  [40] Rcpp_1.0.4.6          rlecuyer_0.3-4        viridisLite_0.3.0    
##  [43] xtable_1.8-4          htmlTable_1.13.3      foreign_0.8-76       
##  [46] bit_1.1-15.2          preprocessCore_1.48.0 Formula_1.2-3        
##  [49] lava_1.6.7            prodlim_2019.11.13    stats4_3.6.1         
##  [52] htmlwidgets_1.5.1     httr_1.4.1            RColorBrewer_1.1-2   
##  [55] acepack_1.4.1         ellipsis_0.3.1        pkgconfig_2.0.3      
##  [58] farver_2.0.3          nnet_7.3-14           reshape2_1.4.4       
##  [61] tidyselect_1.1.0      labeling_0.3          rlang_0.4.6          
##  [64] later_1.1.0.1         AnnotationDbi_1.48.0  munsell_0.5.0        
##  [67] cellranger_1.1.0      tools_3.6.1           generics_0.0.2       
##  [70] RSQLite_2.2.0         broom_0.5.6           evaluate_0.14        
##  [73] stringr_1.4.0         fastmap_1.0.1         yaml_2.2.1           
##  [76] ModelMetrics_1.2.2.2  bit64_0.9-7           zip_2.0.4            
##  [79] purrr_0.3.4           nlme_3.1-148          mime_0.9             
##  [82] xml2_1.3.2            compiler_3.6.1        rstudioapi_0.11      
##  [85] curl_4.3              png_0.1-7             ggsignif_0.6.0       
##  [88] tibble_3.0.1          stringi_1.4.6         highr_0.8            
##  [91] forcats_0.5.0         Matrix_1.2-18         vctrs_0.3.1          
##  [94] pillar_1.4.4          lifecycle_0.2.0       cowplot_1.0.0        
##  [97] httpuv_1.5.4          R6_2.4.1              latticeExtra_0.6-29  
## [100] promises_1.1.1        gridExtra_2.3         rio_0.5.16           
## [103] IRanges_2.20.0        codetools_0.2-16      MASS_7.3-51.6        
## [106] withr_2.2.0           S4Vectors_0.24.0      mgcv_1.8-31          
## [109] hms_0.5.3             grid_3.6.1            rpart_4.1-15         
## [112] timeDate_3043.102     class_7.3-17          tidyr_1.1.0          
## [115] rmarkdown_1.18        carData_3.0-4         pROC_1.16.2          
## [118] lubridate_1.7.9       Biobase_2.46.0        base64enc_0.1-3</code></pre>
<!--html_preserve-->
<script type="application/shiny-prerendered" data-context="dependencies">
{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["1.11.3"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jquery"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.18"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["bootstrap"]},{"type":"character","attributes":{},"value":["3.3.5"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/bootstrap"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["viewport"]}},"value":[{"type":"character","attributes":{},"value":["width=device-width, initial-scale=1"]}]},{"type":"character","attributes":{},"value":["js/bootstrap.min.js","shim/html5shiv.min.js","shim/respond.min.js"]},{"type":"character","attributes":{},"value":["css/bootstrap.min.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.18"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["1.11.3"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jquery"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.18"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jqueryui"]},{"type":"character","attributes":{},"value":["1.11.4"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jqueryui"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery-ui.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.18"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tocify"]},{"type":"character","attributes":{},"value":["1.9.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/tocify"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.tocify.js"]},{"type":"character","attributes":{},"value":["jquery.tocify.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.18"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["1.11.3"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jquery"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.18"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["navigation"]},{"type":"character","attributes":{},"value":["1.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/navigation-1.1"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tabsets.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.18"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["highlightjs"]},{"type":"character","attributes":{},"value":["9.12.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/highlightjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["highlight.js"]},{"type":"character","attributes":{},"value":["default.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.18"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["kePrint"]},{"type":"character","attributes":{},"value":["0.0.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["kePrint-0.0.1"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["kePrint.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["kableExtra"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.1.0"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["kePrint"]},{"type":"character","attributes":{},"value":["0.0.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["kePrint-0.0.1"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["kePrint.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["kableExtra"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["1.1.0"]}]}]}
</script>
<!--/html_preserve-->
<!--html_preserve-->
<script type="application/shiny-prerendered" data-context="execution_dependencies">
{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["packages"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["packages","version"]},"class":{"type":"character","attributes":{},"value":["data.frame"]},"row.names":{"type":"integer","attributes":{},"value":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153]}},"value":[{"type":"character","attributes":{},"value":["abind","acepack","AnnotationDbi","backports","base","base64enc","bigmemory","bigmemory.sri","Biobase","BiocGenerics","bit","bit64","blob","broom","car","carData","caret","cellranger","checkmate","class","cluster","codetools","colorspace","compiler","cowplot","crayon","curl","data.table","datasets","DBI","digest","doParallel","doSNOW","dplyr","dynamicTreeCut","ellipsis","evaluate","farver","fastcluster","fastmap","forcats","foreach","foreign","Formula","generics","ggplot2","ggpubr","ggrepel","ggsci","ggsignif","glue","GO.db","gower","graphics","grDevices","grid","gridExtra","gtable","haven","highr","Hmisc","hms","htmlTable","htmltools","htmlwidgets","httpuv","httr","impute","ipred","IRanges","iterators","itertools","jpeg","jsonlite","kableExtra","knitr","labeling","later","latex2exp","lattice","latticeExtra","lava","lazyeval","lifecycle","limma","lubridate","magrittr","MASS","Matrix","matrixStats","memoise","methods","mgcv","mime","ModelMetrics","munsell","nlme","nnet","openxlsx","parallel","pillar","pkgconfig","plotly","plyr","png","preprocessCore","pROC","prodlim","promises","purrr","R6","randomForest","RColorBrewer","Rcpp","readr","readxl","recipes","reshape2","rio","rlang","rlecuyer","rmarkdown","rpart","RSQLite","rstatix","rstudioapi","rvest","S4Vectors","scales","shiny","snow","splines","stats","stats4","stringi","stringr","survival","tibble","tidyr","tidyselect","timeDate","tools","utils","vctrs","viridisLite","webshot","WGCNA","withr","xfun","xml2","xtable","yaml","zip"]},{"type":"character","attributes":{},"value":["1.4-5","1.4.1","1.48.0","1.1.8","3.6.1","0.1-3","4.5.36","0.1.3","2.46.0","0.32.0","1.1-15.2","0.9-7","1.2.1","0.5.6","3.0-8","3.0-4","6.0-86","1.1.0","2.0.0","7.3-17","2.1.0","0.2-16","1.4-1","3.6.1","1.0.0","1.3.4","4.3","1.12.8","3.6.1","1.1.0","0.6.25","1.0.15","1.0.18","1.0.0","1.63-1","0.3.1","0.14","2.0.3","1.1.25","1.0.1","0.5.0","1.5.0","0.8-76","1.2-3","0.0.2","3.3.1","0.3.0","0.8.2","2.9","0.6.0","1.4.1","3.10.0","0.2.1","3.6.1","3.6.1","3.6.1","2.3","0.3.0","2.3.1","0.8","4.4-0","0.5.3","1.13.3","0.5.0","1.5.1","1.5.4","1.4.1","1.60.0","0.9-9","2.20.0","1.0.12","0.1-3","0.1-8.1","1.6.1","1.1.0","1.28","0.3","1.1.0.1","0.4.0","0.20-41","0.6-29","1.6.7","0.2.2","0.2.0","3.42.0","1.7.9","1.5","7.3-51.6","1.2-18","0.56.0","1.1.0","3.6.1","1.8-31","0.9","1.2.2.2","0.5.0","3.1-148","7.3-14","4.1.5","3.6.1","1.4.4","2.0.3","4.9.2.1","1.8.6","0.1-7","1.48.0","1.16.2","2019.11.13","1.1.1","0.3.4","2.4.1","4.6-14","1.1-2","1.0.4.6","1.3.1","1.3.1","0.1.12","1.4.4","0.5.16","0.4.6","0.3-4","1.18","4.1-15","2.2.0","0.5.0","0.11","0.3.5","0.24.0","1.1.1","1.4.0.2","0.4-3","3.6.1","3.6.1","3.6.1","1.4.6","1.4.0","3.2-3","3.0.1","1.1.0","1.1.0","3043.102","3.6.1","3.6.1","0.3.1","0.3.0","0.5.2","1.69","2.2.0","0.14","1.3.2","1.8-4","2.2.1","2.0.4"]}]}]}
</script>
<!--/html_preserve-->
</div>

<script type="text/javascript" src="../libs/fluidbox/jquery.fluidbox.min.js"></script>
<script type="text/javascript" src="../include/js/enable-fluidbox.js"></script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("section-TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#section-TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
