---
title: "Figure 4"
output:
  html_document:
    includes:
      in_header: ../include/in_header.html
      before_body: ../include/before_body.html
      after_body: ../include/after_body.html
runtime: shiny_prerendered
editor_options: 
  chunk_output_type: console
---
<!-- # (c)  Juozas Gordevičius -->

```{r, include=FALSE}
source("../code/common.R", chdir=TRUE)

require("dplyr")
require("data.table")
require("gtools")
require("foreach")
require("circlize")
require("ggplot2")
require("RobustRankAggreg")
require("glue")
require("cowplot")
require("broom")
require("gridGraphics")
require("magick")
require("sitools")
require("ggrepel")
require("ggdendro")
require("latex2exp")
pacman::p_load("biomaRt")
pacman::p_load("ggforce")
pacman::p_load("Gviz")
pacman::p_load("limma")
theme_publication <- function(font = 7) {
  theme(
    text = element_text(family = "Helvetica", size = font, color = "black"),
    plot.title = element_text(size = font, color = "black"),
    
    panel.background = element_rect(fill = "transparent", colour = NA),
    panel.border = element_blank(),
    plot.background = element_rect(fill = "transparent", colour = NA),
    
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    
    axis.line = element_line(color = "black"),
    axis.text = element_text(size = font, color = "black"), 
    axis.ticks = element_line(size = 0.3, color = "black"),
    axis.title = element_text(size = font, color = "black"),
    
    # strip.background = element_rect(color = "white", fill = "white"),
    strip.text.x = element_text(size = font, color = "black"),
    strip.text.y = element_text(size = font, color = "black"),
    
    legend.background = element_rect(
      fill = alpha("white", 0),
      color = alpha("white", 0)
    ),
    legend.key = element_rect(color = NA, fill = NA),
    legend.key.size = unit(0.2, "cm"),    
    legend.text = element_text(size = font)
  )
}

source("../code/pformat.R")

```

```{r}
# Load aging fits from APX
# path <- "../Appendix_AgingAllControls_Padlock/www/m6_Appendix_AgingAllControls_Padlock.csv"
path <- "../Appendix_AgeAcceleration_Padlock/www/m6_Appendix_AgeAcceleration_Padlock.csv"
apxfit <- fread(path)

# Load aging fits from PFC
# path <- "../Brain_PFCRep_Aging_Padlock/www/m6_Brain_PFCRep_Aging_Padlock.csv"
path <- "../Brain_AgeAcceleration_Padlock/www/m6_Brain_AgeAcceleration_Padlock.csv"
pfcfit <- fread(path)

# Aging trends in appendix
path <- "../Appendix_PDvsControls_Padlock/www/m6_Appendix_PDvsControls_Padlock.csv"
apxpdfit <- fread(path)

path <- "../Brain_PFCRep_Padlock_withGLU/www/m6_Brain_PFCRep_Padlock_withGLU.csv"
pfcpdfit <- fread(path)

```

```{r}
# Direction of aging

# In appendix
colFC <- "C.DiagnosisControl.Age"
colP  <- "P.DiagnosisControl.Age"
t <- apxfit[, table(sign(get(colFC)), p.adjust(get(colP), "fdr") < 0.05)]
f <- fisher.test(t)
pd <- t %>%
  reshape2::melt() %>%
  setDT %>%
  setnames(c("Direction", "Significant", "Count"))
pd[, Color := 0]
pd[Significant == TRUE, Color := Direction %>% as.numeric]
pd[, Direction := 
      factor(Direction, 
        levels = c(-1, 1), 
        labels=c("Hypo-\n methylation", "Hyper-\n methylation"))
  ]
pd[, Significant := 
      factor(Significant, 
        levels=c(FALSE, TRUE), 
        labels=c("Background", "Significant"))
  ]
pd[, Fraction := Count / sum(Count), Significant]
pd[, Type := "Appendix"]

# In PFC
colFC <- "C.GroupCTRL.Age"
colP  <- "P.GroupCTRL.Age"
t2 <- pfcfit[, table(sign(get(colFC)), p.adjust(get(colP), "fdr") < 0.05)]
f2 <- fisher.test(t2)
pd2 <- t2 %>%
  reshape2::melt() %>%
  setDT %>%
  setnames(c("Direction", "Significant", "Count"))
pd2[, Color := 0]
pd2[Significant == TRUE, Color := Direction %>% as.numeric]
pd2[, Direction := 
      factor(Direction, 
        levels = c(-1, 1), 
        labels=c("Hypo-\n methylation", "Hyper-\n methylation"))
  ]
pd2[, Significant := 
      factor(Significant, 
        levels=c(FALSE, TRUE), 
        labels=c("Background", "Significant"))
  ]
pd2[, Fraction := Count / sum(Count), Significant]
pd2[, Type := "PFC"]

or <- f$estimate %>% format(digits = 3)
pv <- sprintf("p = %s", texP(f$p.value))
text1 <- glue("{pv}")
or <- f2$estimate %>% format(digits = 3)
pv <- sprintf("p = %s", texP(f2$p.value))
text2 <- glue("{pv}")
textLayer <- data.table(
  Type = c("Appendix", "PFC"), 
  Text = c(text1, text2), 
  X = c(1.5, 1.5),
  Y = c(0.70, 0.70)
)


# Combine datasets and plot
pA <-
  rbind(pd, pd2) %>%
  .[, Color := Color %>% factor(., levels = c(-1, 1, 0))] %>%
  ggplot(., 
    aes(Direction, Fraction, 
      fill = Color %>% as.factor,
      color = Color %>% as.factor)) + 
    geom_bar(stat = "identity", 
      position = position_dodge(width=0.5), 
      width=0.95) + 
    geom_text(data = textLayer, 
              aes(x=X, y=Y, label = TeX(Text, output="character")), 
              color = "black", size = 2.2,
              inherit.aes = FALSE, parse=TRUE) + 
    facet_grid(Type ~ .) + 
    scale_y_continuous(labels = scales::percent_format(accuracy=1),
      limits = c(0, 0.75)) +
    scale_fill_manual("", values=c("#1f78b4", "#33a02c", "white"), 
      labels = c("Hypo-methylation", "Hyper-methylation", "Not significant")) + 
    scale_color_manual("", values=c("#1f78b4", "#33a02c", "grey60"), 
      labels = c("Hypo-methylation", "Hyper-methylation", "Not significant")) + 
    ylab("Fraction, %") +   
    theme_publication() + 
    theme(axis.title.x = element_blank())
```

```{r}
# Enrichment of AP pathways

pd <- fread("../Results_Aging/www/enrichment_Pathways_DMCs.csv")

pd <- 
pd[ Dir == "All"] %>%
.[, TermName := paste(Name, sep=" ") %>% gsub(" ", "\n", .) %>% as.factor] %>%
.[, P := p.value %>% format(digits = 3, scientific = TRUE) %>% gsub("e", "~x~10^", .)] %>%
.[, P := paste0("\"p =\"~", P)] %>%
.[, Stars := gtools::stars.pval(p.value)] %>%
.[Stars == ".", Stars := ""] %>%
.[, estimate := log(estimate)] %>%
.[, conf.low := log(conf.low)] %>%
.[, conf.high := log(conf.high)]

pB <- 
  ggplot(pd, aes(TermName, estimate, fill = p.value < 0.05)) + 
  geom_bar(stat = "identity", color = "grey50", size = 0.25) + 
  geom_hline(yintercept = 0, color = "grey50", size = 0.25) + 
  geom_text(aes(label = Stars, y = estimate + 0.1 * sign(estimate)), 
    size = 2.2, vjust = 1) +   
  facet_grid(Type ~ ., scales = "free_y") + 
  scale_fill_manual("Enrichment p", labels = c("> 0.05", "< 0.05"), values = c("white", "black")) + 
  scale_y_continuous(limits = c(NA, 0.9)) + 
  ylab("Odds ratio, log") + 
  theme_publication() + 
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = c(0.3, 0.9), 
        legend.direction = "vertical")

# # quartz("", 7.2, 6)
# ggdraw() +
#   draw_plot(pB,
#     x = 0.3, y = 0.545, width = 0.3, height = 0.45) 
```

```{r}
# Aging hallmarks in PFC and appendix
hallmarks <- fread("../Results_Aging/www/Aging_Hallmarks.csv")

pd <- hallmarks[order(Score)] %>% 
  .[1:15] %>%
  .[, list(Gene, Score, EPFC = -log10(PFC.P), EAPX = -log10(APX.P))]
pd[, Rank := 1:.N]
pd[, Width := 1/(Score / sum(Score))]
pd[, Width := Width / sum(Width)]
pd[, Start := cumsum(c(0, Width[-.N]))]
pd[, End := cumsum(Width)]
pd[, Gene := factor(Gene, levels = pd$Gene)]

pC <-
  pd[, list(Gene, Rank, EPFC, EAPX)] %>% 
  melt(id.vars = c("Gene", "Rank")) %>%
  .[, variable := factor(variable, levels = c("EAPX", "EPFC"))] %>% 
  .[value >= -log10(0.01), value := -log10(0.01)] %>%
  ggplot(., aes(variable, Rank, fill = value)) + 
  geom_tile(color = "white") + 
  scale_fill_distiller("Significance,\n-log10(p)", palette = "OrRd", type = "seq", direction = 1) + 
  scale_y_reverse(breaks = 1:15, labels = pd$Gene) + 
  scale_x_discrete(labels = c("Appendix", "PFC")) + 
  theme_publication() + 
  theme(
    axis.line.y = element_blank(),
    axis.title.x = element_blank(),
    axis.title.y = element_blank(),
    axis.text.y = element_text(face = "italic"))

# pC <-
# pd[, list(Gene, Width, EPFC, EAPX)] %>% 
# melt(id.vars = c("Gene", "Width")) %>%
# .[value >= -log10(0.01), value := -log10(0.01)] %>%
# ggplot(., aes(Gene, Width, group = variable, fill = value)) + 
#   geom_bar(stat = "identity", width = 0.8, position = position_dodge(width = 0.9)) +
#   ylab("Ranking score") + 
#   scale_fill_distiller("Significance,\n-log10(p)", palette = "OrRd", type = "seq", direction = 1) + 
#   theme_publication() + 
#   theme(axis.text.x = element_text(angle = 45, hjust = 1))

# pC
```





```{r}
# pD

geneList <- c("GBA","NUCKS1","SLC41A1","SIPA1L2","TMEM163","CCNT2","STK39","CHMP2B","MCCC1","TMEM175","DGKQ","FAM200B","CD38","FAM47E","SNCA","HLA-DRB6","HLA-DQA1","KLHL7","NUPL2","GPNMB","MICU3","BAG3","DLG2","MIR4697","LRRK2","OGFOD2","GCH1","TMEM229B","VPS13C","ZNF646","KAT8","ARHGAP27","CRHR1","SPPL2C","MAPT","STH","KANSL1","SYT4","LSM7","DDRGK1","ITPKB","IL1R2","SCN3A","SATB1","NCKIPSD","CDC71","ALAS1","TLR9","DNAH1","BAP1","PHF7","NISCH","STAB1","ITIH3","ITIH4","ANK2","CAMK2D","ELOVL7","ZNF184","CTSB","SORBS3","PDLIM2","C8orf58","BIN3","SH3GL2","FAM171A1","GALC","COQ7","TOX3","ATP6V0A1","PSMC3IP","TUBG2")


wmean <- function(x, w) {
  i <- is.na(x) | is.na(w)
  x <- x[!i]; w <- w[!i]
  weighted.mean(x, w, na.rm=TRUE)
}

fdr <- function(x) { p.adjust(x, "fdr")}

A <- apxfit %>%
  .[, list(ID, Chr, SNP, Gene, 
    FC = C.DiagnosisControl.Age, 
    P = P.DiagnosisControl.Age, 
    Q = p.adjust(P.DiagnosisControl.Age, "fdr"))
  ] %>%
  .[, list(HealthyAging = wmean(FC, -log(P)), IsHealthyAging = any(Q < 0.05, na.rm=TRUE)), Gene]

B <- apxfit %>%
  .[, list(ID, Chr, SNP, Gene, 
    FC = C.DiagnosisPD.LBD.Age, 
    P = P.DiagnosisPD.LBD.Age, 
    Q = p.adjust(P.DiagnosisPD.LBD.Age, "fdr"))
  ] %>%
  .[, list(PDAging = wmean(FC, -log(P)), IsPDAging = any(Q < 0.05, na.rm=TRUE)), Gene]

C  <- apxpdfit %>%
  .[, list(PD = wmean(logFC, -log(P.Value)), IsPD = any(adj.P.Val < 0.05, na.rm=TRUE)), Gene]
dt <- merge(merge(A, B, by = "Gene"), C, by = "Gene")


df.venn <- data.frame(x = c(0, 0.866, -0.866), 
                      y = c(1, -0.5, -0.5), 
                      x.labels = c(-2, 1.2, -1.2), 
                      y.labels = c(2, -2.2, -2.2),
                      labels = c('PD vs \nControl', 'Age, PD', 'Age, Control'))
vdc <- vennCounts(dt[, list(IsHealthyAging, IsPDAging, IsPD)])
class(vdc) <- 'matrix'
df.vdc <- as.data.frame(vdc)[-1,] %>%
  mutate(x = c(0, 1.2, 0.8, -1.2, -0.8, 0, 0),
         y = c(1.2, -0.6, 0.5, -0.6, 0.5, -1, 0))

pD <- 
  df.venn %>% 
  mutate(Tissue = "Appendix") %>%
  ggplot() +
    geom_circle(aes(x0 = x, y0 = y, r = 1.5, fill = labels), alpha = .3, size = 0.25, colour = 'grey') +
    coord_fixed() +
    facet_grid(Tissue ~ .) +
    theme(legend.position = 'bottom') +
    scale_fill_manual(values = c('#0080FF', '#FF00FF',  '#006400'), guide=FALSE) +
    scale_colour_manual(values = c('#0080FF', '#FF00FF',  '#006400'), guide = FALSE) +
    labs(fill = NULL) +
    annotate("text", x = df.vdc$x, y = df.vdc$y, label = df.vdc$Counts, size = 2.2) + 
    annotate('text', x = df.venn$x.labels, y = df.venn$y.labels, label = df.venn$labels, size = 2.2) + 
    theme_publication() +
    theme(
          axis.line=element_blank(),axis.text.x=element_blank(),
          axis.text.y=element_blank(),axis.ticks=element_blank(),
          axis.title.x=element_blank(),
          axis.title.y=element_blank())

```


```{r}
# pE

A <- pfcfit %>%
  .[, list(ID, Chr, SNP, Gene, 
    FC = C.GroupCTRL.Age, 
    P = P.GroupCTRL.Age, 
    Q = p.adjust(P.GroupCTRL.Age, "fdr"))
  ] %>%
  .[, list(HealthyAging = wmean(FC, -log(P)), IsHealthyAging = any(Q < 0.05, na.rm=TRUE)), Gene]

B <- pfcfit %>%
  .[, list(ID, Chr, SNP, Gene, 
    FC = C.GroupPD.Age, 
    P = P.GroupPD.Age, 
    Q = p.adjust(P.GroupPD.Age, "fdr"))
  ] %>%
  .[, list(PDAging = wmean(FC, -log(P)), IsPDAging = any(Q < 0.05, na.rm=TRUE)), Gene]

C  <- pfcpdfit %>%
  .[, list(PD = wmean(logFC, -log(P.Value)), IsPD = any(adj.P.Val < 0.05, na.rm=TRUE)), Gene]
dt <- merge(merge(A, B, by = "Gene"), C, by = "Gene")
dt[Gene %in% geneList]


df.venn <- data.frame(x = c(0, 0.866, -0.866), 
                      y = c(1, -0.5, -0.5), 
                      x.labels = c(-2, 1.2, -1.2), 
                      y.labels = c(2, -2.2, -2.2),
                      labels = c('PD vs \nControl', 'Age, PD', 'Age, Control'))
vdc <- vennCounts(dt[, list(IsHealthyAging, IsPDAging, IsPD)])
class(vdc) <- 'matrix'
df.vdc <- as.data.frame(vdc)[-1,] %>%
  mutate(x = c(0, 1.2, 0.8, -1.2, -0.8, 0, 0),
         y = c(1.2, -0.6, 0.5, -0.6, 0.5, -1, 0))
pE <- 
  df.venn %>% 
  mutate(Tissue = "PFC") %>%
  ggplot() +
    geom_circle(aes(x0 = x, y0 = y, r = 1.5, fill = labels), alpha = .3, size = 0.25, colour = 'grey') +
    coord_fixed() +
    facet_grid(Tissue ~ .) +
    theme(legend.position = 'bottom') +
    scale_fill_manual(values = c('#0080FF', '#FF00FF',  '#006400'), guide=FALSE) +
    scale_colour_manual(values = c('#0080FF', '#FF00FF',  '#006400'), guide = FALSE) +
    labs(fill = NULL) +
    annotate("text", x = df.vdc$x, y = df.vdc$y, label = df.vdc$Counts, size = 2.2) + 
    annotate('text', x = df.venn$x.labels, y = df.venn$y.labels, label = df.venn$labels, size = 2.2) + 
    theme_publication() +
    theme(
          axis.line=element_blank(),axis.text.x=element_blank(),
          axis.text.y=element_blank(),axis.ticks=element_blank(),
          axis.title.x=element_blank(),
          axis.title.y=element_blank())
```

```{r}
#pF

pd <- 
  apxfit[, list(
    Gene, ID, Chr, SNP,
    Healthy = C.DiagnosisControl.Age,
    HealthySLP = -sign(C.DiagnosisControl.Age) * log(P.DiagnosisControl.Age),
    HealthyQ = fdr(P.DiagnosisControl.Age),
    PD = C.DiagnosisPD.LBD.Age,
    PDSLP = -sign(C.DiagnosisPD.LBD.Age) * log(P.DiagnosisPD.LBD.Age),
    PDQ = fdr(P.DiagnosisPD.LBD.Age)
    )]
pd <- 
  pd[, list(
    Healthy = wmean(Healthy, -log(HealthyQ)),
    PD = wmean(PD, -log(PDQ)),
    IsHealthy = any(HealthyQ < 0.05, na.rm=TRUE),
    IsPD = any(PDQ < 0.05, na.rm=TRUE)
  ), Gene]
pd <- pd[IsHealthy | IsPD]
pd[IsHealthy == FALSE, Healthy := 0]
pd[IsPD == FALSE, PD := 0]

pd[, Label := paste(IsHealthy, IsPD)]
pd[, Label := factor(Label, 
                    levels = c("TRUE FALSE", "TRUE TRUE", "FALSE TRUE"),
                    labels = c("Age effect, Control", "Control & PD aging", "Age effect, PD"))]
pd <- pd[order(Label, PD+Healthy)]
pd[, X := 1:.N]
pd[, Tissue := "Appendix"]

pF <- 
  pd %>%  
  ggplot(aes(x=X, xend=X, y=Healthy, yend=PD)) + 
  geom_segment(size = 0.25, color = "grey85") +
   geom_point(
    aes(x = X, y = Healthy, color = "Age effect, Control"), size = 0.25) +
  geom_point(
    aes(x = X, y = PD, color = "Age effect, PD"), size = 0.25) +
  geom_segment(
      data = function(x) x[Gene %in% geneList],
      size = 0.5, color = "orange") +   
   geom_text_repel(
      data = function(x) x[Gene %in% geneList],
      nudge_y = 0.05, segment.size = 0.25,
      ylim = c(0.07, Inf), force=10,
      size = 2.2, fontface = "italic",
      aes(label = Gene)
      ) + 
  ylab("Age effect,\nmean M-value/year") + 
  facet_grid(Tissue~.) + 
  scale_y_continuous(
      labels = scales::number_format(accuracy = 0.01)) + 
  scale_color_manual("", values = c(`Age effect, Control` = '#0080FF', `Age effect, PD` = '#FF00FF')) + 
  theme_publication() + 
  theme(
      legend.position = c(0.5, 0.95),
      axis.line.x = element_blank(),
      axis.ticks.x = element_blank(),
      axis.text.x = element_blank(),
      axis.title.x = element_blank()
    ) 

# ggdraw() +
#   draw_plot(
#     pF,
#     x = 0.29, y = .4, width = 0.71, height = 0.19)

```

```{r}
#pG

# PFC 
pd <- 
  pfcfit[, list(
    Gene, ID, Chr, SNP,
    Healthy = C.GroupCTRL.Age,
    HealthySLP = -sign(C.GroupCTRL.Age) * log(P.GroupCTRL.Age),
    HealthyQ = fdr(P.GroupCTRL.Age),
    PD = C.GroupPD.Age,
    PDSLP = -sign(C.GroupPD.Age) * log(P.GroupPD.Age),
    PDQ = fdr(P.GroupPD.Age)
    )]
pd <- 
  pd[, list(
    Healthy = wmean(Healthy, -log(HealthyQ)),
    PD = wmean(PD, -log(PDQ)),
    IsHealthy = any(HealthyQ < 0.05, na.rm=TRUE),
    IsPD = any(PDQ < 0.05, na.rm=TRUE)
  ), Gene]
pd <- pd[IsHealthy | IsPD]
pd[IsHealthy == FALSE, Healthy := 0]
pd[IsPD == FALSE, PD := 0]

pd[, Label := paste(IsHealthy, IsPD)]
pd[, Label := factor(Label, 
                    levels = c("TRUE FALSE", "TRUE TRUE", "FALSE TRUE"),
                    labels = c("Age effect, Control", "Control & PD aging", "Age effect, PD"))]
pd <- pd[order(Label, PD+Healthy)]
pd[, X := 1:.N]
pd[, Tissue := "PFC"]



pG <-
  pd %>%  
  ggplot(aes(x=X, xend=X, y=Healthy, yend=PD)) + 
  geom_segment(size = 0.25, color = "grey85") +
   geom_point(
    aes(x = X, y = Healthy), color = '#0080FF', size = 0.25) +
  geom_point(
    aes(x = X, y = PD), color = '#FF00FF', size = 0.25) +
  geom_segment(
      data = function(x) x[Gene %in% geneList],
      size = 0.5, color = "orange") +   
  geom_text_repel(
      data = function(x) x[Gene %in% geneList],
      ylim = c(0.05, Inf), force=10,
      nudge_y = 0.01, segment.size = 0.25,
      size = 2.2,
      aes(label = Gene)
      ) + 
  ylab("Age effect,\nmean M-value/year") +
  xlab("Gene") + 
  scale_y_continuous(
      labels = scales::number_format(accuracy = 0.01)) + 
  facet_grid(Tissue~.) + 
  theme_publication() + 
  theme(
      axis.line.x = element_blank(),
      axis.ticks.x = element_blank(),
      axis.text.x = element_blank()
    ) 

ggdraw() +
  draw_plot(
    pF,
    x = 0.29, y = .4, width = 0.71, height = 0.19) + 
  draw_plot(
    pG,
    x = 0.29, y = .21, width = 0.71, height = 0.20)
```


```{r, include = TRUE, fig.width = 7.2, fig.height = 7.2, dpi = 300}
# X11("", 7.2, 7.2)
# dev.set(2)
# quartz("", 7.2, 7.2, dpi=100)


ggdraw() +
  draw_plot(pA + guides(color =FALSE, fill = FALSE),
    x = 0, y = 0.63, width = 0.3, height = 0.37) +
  draw_plot(pB,
    x = 0.3, y = 0.545, width = 0.3, height = 0.45) + 
  draw_plot(pC,
    x = 0.6, y = 0.64, width = 0.4, height = 0.36) +
  draw_plot(
    pD,
    x = -0.05, y = .4, width = 0.4, height = 0.2) +
  draw_plot(
    pE,
    x = -0.05, y = .21, width = 0.4, height = 0.2) + 
  draw_plot(
    pF,
    x = 0.29, y = .4, width = 0.71, height = 0.19) + 
  draw_plot(
    pG,
    x = 0.29, y = .21, width = 0.71, height = 0.20) + 
  draw_plot_label(
    label = c(letters[1:5]),
    size = 10,
    x = c(0, 0.3, 0.6, 0,    0.28),
    y = c(1, 1,   1,   0.59, 0.59))


ggsave("../Figure4/figure4.png", width = 7.2, height = 7.2, dpi = 300, units = "in")
```

